---
title: "scRNA-seq Ofav Heat-stress Spring 2022"
author: "Anthony Bonacolta"
date: "08/21/2023"
output: html_document
editor_options: 
  markdown: 
    wrap: sentence
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = '/Volumes/GoogleDrive/.shortcut-targets-by-id/1wkxSWfsOfMy-OsjBXEzK1bJHunl3Wvv_/AnthonyOfav/seurat/')
```

# Load libraries

```{r, include=FALSE}
library(dplyr)
library(Seurat)
library(patchwork)
library(viridis)
library(ggplot2)
library(Matrix)
library("scCustomize")

c25 <- c(
  "dodgerblue2", "#E31A1C", # red
  "green4",
  "#6A3D9A", # purple
  "#FF7F00", # orange
  "black", "gold1",
  "skyblue2", "#FB9A99", # lt pink
  "palegreen2",
  "#CAB2D6", # lt purple
  "#FDBF6F", # lt orange
  "gray70", "khaki2",
  "maroon", "orchid1", "deeppink1", "blue1", "steelblue4",
  "darkturquoise", "green1", "yellow4", "yellow3",
  "darkorange4", "brown"
)

library(randomcoloR)
n <- 50
palette <- distinctColorPalette(n) # Use "palette" if you need 50 colors in a figure.
```

# Creat Seurat Objects

## Time Point 0- Treatment Sample
```{r}
T0T.data <- ReadSTARsolo(data.dir = "/Volumes/projects/bonacolta/sc_ofav/Ofav_Exp_Spring2022/T0-T/analysis/newref.Solo.out/Gene/raw")
T0T <- CreateSeuratObject(counts = T0T.data, project = "Ofav-T0-T", min.cells = 3, min.features = 10)
nUMI_high <<- quantile(T0T@meta.data$nCount_RNA,0.99)
T0T <- subset(T0T, subset = nCount_RNA > 200 & nCount_RNA < nUMI_high)
T0T
summary(colSums(T0T))
rm(T0T.data)
```

51681 features across 5611 samples within 1 assay 
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  201.0   272.0   394.0   490.9   618.0  1414.0 

## Time Point 3- Treatment Sample
```{r}
T3T.data <- ReadSTARsolo(data.dir = "/Volumes/projects/bonacolta/sc_ofav/Ofav_Exp_Spring2022/T3-T/analysis/newref.T3.Solo.out/Gene/raw")
T3T <- CreateSeuratObject(counts = T3T.data, project = "Ofav-T3-T", min.cells = 3, min.features = 10)
nUMI_high <<- quantile(T3T@meta.data$nCount_RNA,0.99)
T3T <- subset(T3T, subset = nCount_RNA > 200 & nCount_RNA < nUMI_high)
T3T
summary(colSums(T3T))
rm(T3T.data)
```

An object of class Seurat 
31127 features across 3019 samples within 1 assay 
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  201.0   246.0   323.0   380.9   460.0   961.0 

## Time Point 5- Treatment Sample

```{r}
T5T.data <- ReadSTARsolo(data.dir = "/Volumes/projects/bonacolta/sc_ofav/Ofav_Exp_Spring2022/T5-T/analysis/newref.T5.Solo.out/Gene/raw")
T5T <- CreateSeuratObject(counts = T5T.data, project = "Ofav-T5-T", min.cells = 3, min.features = 10)
nUMI_high <<- quantile(T5T@meta.data$nCount_RNA,0.99)
T5T <- subset(T5T, subset = nCount_RNA > 200 & nCount_RNA < nUMI_high)
summary(colSums(T5T))
T5T
rm(T5T.data)
```

An object of class Seurat 
21191 features across 2390 samples within 1 assay 
Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  201.0   244.0   307.0   343.3   415.0   690.0 

## Time Point 7- Treatment Sample
```{r}
T7T.data <- ReadSTARsolo(data.dir = "/Volumes/projects/bonacolta/sc_ofav/Ofav_Exp_Spring2022/T7-T/analysis/newref.T7.Solo.out/Gene/raw")
T7T <- CreateSeuratObject(counts = T7T.data, project = "Ofav-T7-T", min.cells = 3, min.features = 10)
nUMI_high <<- quantile(T7T@meta.data$nCount_RNA,0.99)
T7T
T7T <- subset(T7T, subset = nCount_RNA > 200 & nCount_RNA < nUMI_high)
summary(colSums(T7T))
T7T
rm(T7T.data)
```

24203 features across 4718 samples within 1 assay 
Active assay: RNA (24203 features, 0 variable features)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  201.0   262.2   370.0   459.0   578.0  1257.0 

# Log-normalize the data & find variable features independently
```{r}
T0T <- NormalizeData(T0T, normalization.method = "LogNormalize", scale.factor = 10000)
T0T <- FindVariableFeatures(T0T, selection.method = "vst")
T3T <- NormalizeData(T3T, normalization.method = "LogNormalize", scale.factor = 10000)
T3T <- FindVariableFeatures(T3T, selection.method = "vst")
T5T <- NormalizeData(T5T, normalization.method = "LogNormalize", scale.factor = 10000)
T5T <- FindVariableFeatures(T5T, selection.method = "vst")
T7T <- NormalizeData(T7T, normalization.method = "LogNormalize", scale.factor = 10000)
T7T <- FindVariableFeatures(T7T, selection.method = "vst")
```

# Perform Integration
```{r}
features <- SelectIntegrationFeatures(object.list = c(T0T, T3T, T5T, T7T))
ofav.anchors <- FindIntegrationAnchors(object.list = c(T0T, T3T, T5T, T7T), anchor.features = features)
ofav.combined <- IntegrateData(anchorset = ofav.anchors)
```

```{r}
rm(T0T)
rm(T3T)
rm(T5T)
rm(T7T)
```

# Viz and Cluster
```{r}
DefaultAssay(ofav.combined) <- "integrated"
all.genes <- rownames(ofav.combined)
ofav.combined<- ScaleData(ofav.combined, features = all.genes)
ofav.combined<- RunPCA(ofav.combined, verbose = FALSE, npcs=20)
ofav.combined <- JackStraw(ofav.combined, num.replicate = 100, dims = 50)
ofav.combined <- ScoreJackStraw(ofav.combined, dims = 1:20)
JackStrawPlot(ofav.combined, dims = 1:20)
ElbowPlot(ofav.combined)
# Looks like 17 is a good cutoff
```

```{r}
ofav.combined<- RunTSNE(ofav.combined, reduction = "pca", check_duplicates = FALSE, dims=1:20)
ofav.combined<- RunUMAP(ofav.combined, dims=1:20)
ofav.combined <- FindNeighbors(ofav.combined, dims = 1:20)
ofav.combined<- FindClusters(ofav.combined, resolution=0.3, algorithm =1, group.singletons = F)
UMAPPlot(ofav.combined)
```

# Now do the same thing with JUST the coral genes used
```{r}
features <- SelectIntegrationFeatures(object.list = c(T0T, T3T, T5T, T7T))
Csparse_validate = "CsparseMatrix_validate"
ofav.anchors <- FindIntegrationAnchors(object.list = c(T0T, T3T, T5T, T7T), anchor.features = features)
ofav.combined <- IntegrateData(anchorset = ofav.anchors)

#Grab only coral genes
hvg <- VariableFeatures(ofav.combined)
var_regex = '^QW917' 
hvg <- grep(var_regex, hvg, invert=F, value=T)

DefaultAssay(ofav.combined) <- "integrated"
ofav.combined.coral <- ScaleData(ofav.combined, features = hvg)
ofav.combined.coral <- RunPCA(ofav.combined.coral, verbose = FALSE, npcs=20)
ofav.combined.coral <- JackStraw(ofav.combined.coral, num.replicate = 100, dims = 50)
ofav.combined.coral <- ScoreJackStraw(ofav.combined.coral, dims = 1:20)
ofav.combined.coral <- RunTSNE(ofav.combined.coral, reduction = "pca", check_duplicates = FALSE, dims=1:20)
ofav.combined.coral <- RunUMAP(ofav.combined.coral, dims=1:20)
ofav.combined.coral <- FindNeighbors(ofav.combined.coral, dims = 1:20)
ofav.combined.coral <- FindClusters(ofav.combined.coral, resolution=0.3, algorithm =1, group.singletons = F)
UMAPPlot(ofav.combined.coral)

pdf("figures/UMAP-onlycoral.pdf", height = 7, width=10)
UMAPPlot(ofav.combined.coral)
dev.off()

```
Structure appears to hold

# Start Here
```{r}
#saveRDS(ofav.combined, "/Volumes/GoogleDrive/.shortcut-targets-by-id/1wkxSWfsOfMy-OsjBXEzK1bJHunl3Wvv_/AnthonyOfav/seurat/ofav.combined.rds")
ofav.combined <- readRDS("/Volumes/GoogleDrive/.shortcut-targets-by-id/1wkxSWfsOfMy-OsjBXEzK1bJHunl3Wvv_/AnthonyOfav/seurat/ofav.combined.rds")
```

## UMIs per cell cluster 
```{r}
DefaultAssay(ofav.combined) <- "RNA"
p1 <- VlnPlot(ofav.combined, features = c("nCount_RNA"), pt.size = 0) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "UMIs per Cell") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/Cluster_UMIs.pdf", height = 7, width=10)
plot(p1)
dev.off()
```


## Genes per cell cluster
```{r}
DefaultAssay(ofav.combined) <- "RNA"
p1 <- VlnPlot(ofav.combined, features = c("nFeature_RNA"), pt.size = 0.0) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Genes per Cell") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/Cluster_genes.pdf", height = 7, width=10)
plot(p1)
dev.off()
```


## Cells per cluster per time-point
## Determine number of cells
```{r}
cellplot <- ggplot(ofav.combined@meta.data, aes(orig.ident, fill=seurat_clusters))+geom_bar(stat="count")+ 
  labs(x= "Sample", y = "Number of Cells", fill="Cluster ID") + theme_bw() +  
  scale_fill_manual(values = c25) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  ggtitle(label="Number of cells per scRNA-seq Sample after filtering", subtitle = "after QC") + scale_y_continuous(limits = c(0,6000), expand = c(0,0))
cellplot
pdf("figures/Cell_counts.pdf", width=7, height=5)
plot(cellplot)
dev.off()
```

### Convert this to percentage
```{r}
cellplot <- ggplot(ofav.combined@meta.data, aes(orig.ident, fill=seurat_clusters))+geom_bar(position="fill")+ 
  labs(x= "Sample", y = "Proportion of Cells", fill="Cluster ID") + theme_bw() +  
  scale_fill_manual(values = c25) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
  ggtitle(label="Number of cells per scRNA-seq Sample after filtering", subtitle = "after QC") + scale_y_continuous(limits = c(0,1), expand = c(0,0))
cellplot
pdf("figures/Cell_percentages.pdf", width=7, height=5)
plot(cellplot)
dev.off()
```

## ID Markers
```{r}
DefaultAssay(ofav.combined) <- "RNA"
all.genes <- rownames(ofav.combined)
ofav.combined<- ScaleData(ofav.combined, features = all.genes)
ofav.markers<- FindAllMarkers(ofav.combined, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
ofav.markers %>% group_by(cluster) %>% top_n(10, avg_log2FC) -> top10
ofav.combined@meta.data[,"seurat_clusters",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters) %>% sample_n(40) -> index
DoHeatmap(ofav.combined, features = top10$gene,label=F)
DoHeatmap(ofav.combined, features = top10$gene,label=F,cells=index$rowname)
ofav.markers <- ofav.markers[ofav.markers$p_val_adj<0.05,]


pdf("figures/heatmapunlabelled.pdf", height = 15, width=10)
ofav.markers %>%
    group_by(cluster) %>%
    top_n(n = 10, wt = avg_log2FC) -> top10
DoHeatmap(ofav.combined, features = top10$gene, cells=index$rowname) + NoLegend()
dev.off()
```

# Plot Symbiont UMIs across all clusters
```{r}
DefaultAssay(ofav.combined) <- "RNA"
# Assuming 'seurat_object' is your Seurat object

# Add cluster information to metadata
ofav.combined$cluster <- Idents(ofav.combined)

# Access the transposed expression matrix
expression_matrix <- t(as.matrix(ofav.combined@assays$RNA@data))

# Calculate total expression by gene and cluster
total_expression <- aggregate(expression_matrix, by = list(ofav.combined$cluster), FUN = sum)

# Rename the columns for clarity
colnames(total_expression) <- c("Cluster", colnames(expression_matrix))

# Print or view the total expression table
print(total_expression)

write.csv(t(total_expression), "expression_matrix.csv")

```
## Calculated outside of r

```{r}
# Install and load the ggplot2 package if you haven't already
# install.packages("ggplot2")
library(ggplot2)
# You can read your CSV file using: data <- read.csv("your_data.csv")
data <- read.csv("cluster_symbionts_expression.csv", header = T, row.names = 1, check.names = F)
# Transpose the data for better visualization
data_transposed <- t(data)

# Convert the transposed data to a data frame
data_df <- as.data.frame(data_transposed)

# Add a new column for Cluster IDs
data_df$Cluster <- factor(rownames(data_df), levels = 0:18)

# Reshape the data for ggplot2
library(tidyr)
data_long <- pivot_longer(data_df, cols = -Cluster, names_to = "Reference", values_to = "Value")

# Create the stacked barplot
library(ggplot2)
symbiont_exp <- ggplot(data_long, aes(x = Cluster, y = Value, fill = Reference)) +
  geom_bar(stat = "identity") +
  labs(title = "% of Gene Expression from each Reference",
       x = "Cluster ID",
       y = "% Gene Expression") +
  theme_bw() +  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = c("coral", "darkgoldenrod1", "darkgoldenrod4", "cornsilk3"), breaks=c('Coral', 'Breviolum', 'Durusdinium', 'Cladocopium'))

pdf("figures/symbiont_expression.pdf", height = 7, width=10)
plot(symbiont_exp)
dev.off()

# Install and load the ggplot2 package if you haven't already
# install.packages("ggplot2")
library(ggplot2)
# You can read your CSV file using: data <- read.csv("your_data.csv")
data <- read.csv("cluster_symbionts_expression2.csv", header = T, row.names = 1, check.names = F)
# Transpose the data for better visualization
data_transposed <- t(data)

# Convert the transposed data to a data frame
data_df <- as.data.frame(data_transposed)

# Add a new column for Cluster IDs
data_df$Cluster <- (rownames(data_df))

# Reshape the data for ggplot2
library(tidyr)
data_long <- pivot_longer(data_df, cols = -Cluster, names_to = "Reference", values_to = "Value")

# Create the stacked barplot
library(ggplot2)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")
symbiont_exp <- ggplot(data_long, aes(x = factor(Cluster, levels = my_levels), y = Value, fill = Reference)) +
  geom_bar(stat = "identity") +
  labs(title = "% of Gene Expression from each Reference",
       x = "Cluster ID",
       y = "% Gene Expression") +
  theme_bw() +  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = c("coral", "darkgoldenrod1", "darkgoldenrod4", "cornsilk3"), breaks=c('Coral', 'Breviolum', 'Durusdinium', 'Cladocopium')) +
  scale_x_discrete(guide = guide_axis(angle = 90))

pdf("figures/symbiont_expression.pdf", height = 7, width=10)
plot(symbiont_exp)
dev.off()
```


## Marker specificity (Xenia Approach)
We define that a gene is enriched in a cell if the expression of the gene in the cell is higher than the average expression of the gene among all cells. 
We further define the specificity of a marker gene in a cluster as the percentage of cells , in which the marker gene is enriched as defined above within the cluster.
```{r}
DefaultAssay(ofav.combined) <- "RNA"
exp <- ofav.combined@assays$RNA@data
exp_spcificity <- exp >Matrix::rowMeans(exp)
temp_list <- list()
for (i in 1:18){
  sub_cluster <- ofav.markers[ofav.markers$cluster ==i,]
  marker_exp <- exp_spcificity[ofav.markers[ofav.markers$cluster ==i,]$gene,]
  spcificity_within_cluster <- Matrix::rowSums(marker_exp[,colnames(marker_exp) %in% rownames(ofav.combined@meta.data[ofav.combined@meta.data$seurat_clusters == i,])])/sum(ofav.combined@meta.data$seurat_clusters == i)
  spcificity_outside_cluster <- Matrix::rowSums(marker_exp[,!colnames(marker_exp) %in% rownames(ofav.combined@meta.data[!ofav.combined@meta.data$seurat_clusters == i,])])/sum(!ofav.combined@meta.data$seurat_clusters == i)
  temp <- data.frame(gene=rownames(marker_exp),spcificity_within_cluster=spcificity_within_cluster,spcificity_outside_cluster=spcificity_outside_cluster)
  sub_cluster <- left_join(sub_cluster, temp)
  temp_list[[i]]<-sub_cluster
}
ofav.markers<- do.call(rbind,temp_list)
#Filter low specificity markers
ofav.markers <- ofav.markers[ofav.markers$spcificity_within_cluster>0.5,]
write.csv(ofav.markers, file ="cluster_markers.csv")
```

## Other Approach as found here: https://hbctraining.github.io/scRNA-seq_online/lessons/09_merged_SC_marker_identification.html
```{r}
library("purrr")
library("multtest")
library("metap")
DefaultAssay(ofav.combined) <- "RNA"

cons_1 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 1,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 1, .)

cons_4 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 4,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 4, .)

cons_5 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 5,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 5, .)

cons_6 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 6,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 6, .)

cons_7 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 7,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 7, .)


cons_9 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 9,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 9, .)

cons_10 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 10,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 10, .)

cons_11 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 11,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 11, .)

cons_12 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 12,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 12, .)

cons_13 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 13,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 13, .)

cons_14 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 14,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 14, .)

cons_15 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 15,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 15, .)

cons_16 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 16,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 16, .)

cons_17 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 17,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 17, .)

cons_18 <- FindConservedMarkers(ofav.combined,
                       ident.1 = 18,
                       grouping.var = "orig.ident",
                       only.pos = TRUE,
		                  min.diff.pct = 0.25,
                     min.pct = 0.1,
		     logfc.threshold = 0.25) %>%
    cbind(cluster_id = 18, .)


cons.markers <- rbind(cons_0,  cons_1,  cons_4,  cons_5,  cons_6,  cons_7,  cons_8,  cons_9,  cons_10,  cons_11,  cons_12,  cons_13,  cons_14,  cons_15,  cons_16,  cons_17,  cons_18)

write.csv(cons.markers, "cluster_markers_conserved.csv")
```

# UMAP with hypothesized cluster IDs
```{r}
levels(Idents(ofav.combined))

ofav.combined.labeled <- RenameIdents(ofav.combined, "0"="Unidentified", "1"="Epidermis", "2"="Breviolum-hosting Cells", "3"="Extracellular Breviolum", "4"="Calicoblastic Cells", "5"="Gastrodermis 1", "6"="Gastrodermis 2", "7"="Gland 1", "8"="Neurons 1", "9"="Immune", "10"="Neurons 2", "11"="Gland 2", "12"="Neurons 3", "13"="Gastrodermis (Muscle-like)", "14"="Durusdinium-hosting Cells", "15"="Digestive Filaments", "16"="Precursors", "17"="Gland 3", "18"="Cnidocytes")


umap_labeled <- DimPlot(ofav.combined.labeled, reduction = "umap", split.by = "orig.ident", label = TRUE, pt.size = 0.35, ncol = 2, cols=c25) + NoLegend()+ labs(title = "UMAP Clustering of Orbicella faveolata Holobiont", subtitle = "with hypothesized cell cluster IDs")
umap_labeled

pdf("figures/UMAPs-temps-labeled.pdf", width=10, height=10)
plot(umap_labeled)
dev.off()

umap_labeled <- DimPlot(ofav.combined.labeled, reduction = "umap", split.by = "orig.ident", label = F, pt.size = 0.35, ncol = 2, cols=c25) + labs(title = "UMAP Clustering of Orbicella faveolata Holobiont", subtitle = "with hypothesized cell cluster IDs")
umap_labeled

pdf("figures/UMAPs-temps-labeled-legend.pdf", width=10, height=7)
plot(umap_labeled)
dev.off()
```

## Fixed UMAP with Ordered
```{r}
levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)

# Define colors for clusters
cluster_colors <- c("azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")

ofav.combined.labeled$cluster_colors <- cluster_colors[Idents(ofav.combined.labeled)]

umap_labeled <- DimPlot(ofav.combined.labeled, reduction = "umap", split.by = "orig.ident", label = F, pt.size = 0.35, ncol = 2) + labs(title = "UMAP Clustering of Orbicella faveolata Holobiont", subtitle = "with hypothesized cell cluster IDs") + scale_color_manual(values = cluster_colors)
umap_labeled

pdf("figures/UMAPs-temps-labeled-legend-ordered.pdf", width=10, height=7)
plot(umap_labeled)
dev.off()

umap_labeled <- DimPlot(ofav.combined.labeled, reduction = "umap", label = F, pt.size = 0.35, ncol = 2) + labs(title = "UMAP Clustering of Orbicella faveolata Holobiont", subtitle = "with hypothesized cell cluster IDs") + scale_color_manual(values = cluster_colors)
umap_labeled

pdf("figures/UMAPs-combined-labeled-legend-ordered.pdf", width=10, height=7)
plot(umap_labeled)
dev.off()

umap_labeled <- DimPlot(ofav.combined.labeled, reduction = "umap", split.by = "orig.ident", label = F, pt.size = 0.35, ncol = 2) + labs(title = "UMAP Clustering of Orbicella faveolata Holobiont", subtitle = "with hypothesized cell cluster IDs") + scale_color_manual(values = cluster_colors)
umap_labeled

pdf("figures/UMAPs-temps-labeled-ordered-legend.pdf", width=10, height=7)
plot(umap_labeled)
dev.off()
```

# Start Here with labelled cell clusters
```{r}
#saveRDS(ofav.combined.labeled, "/Volumes/projects/bonacolta/sc_ofav/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled <- readRDS("/Volumes/projects/bonacolta/sc_ofav/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
```

```{r}
DefaultAssay(ofav.combined.labeled) <- "RNA"
all.genes <- rownames(ofav.combined.labeled)
ofav.markers<- FindAllMarkers(ofav.combined.labeled, only.pos = TRUE, min.pct = 0.25, thresh.use = 0.25)
ofav.markers %>% group_by(cluster) %>% top_n(10, avg_log2FC) -> top10
ofav.combined.labeled@meta.data[,"seurat_clusters",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters) %>% sample_n(40) -> index
hm <- DoHeatmap(ofav.combined.labeled, features = top10$gene,label=F,cells=index$rowname, group.colors = c25)
pdf("figures/heatmap_all.pdf", width=11, height=20)
plot(hm)
dev.off()
```


```{r}
gene_table <- AverageExpression(
  ofav.combined,
  assays = NULL,
  features = NULL,
  return.seurat = FALSE,
  group.by = "seurat_clusters",
  add.ident = NULL, 
  slot = "data",
  verbose = TRUE
)

write.csv(gene_table[["RNA"]], "gene_table2.csv")
```


# GO Analysis (https://www.biostars.org/p/350710/)
```{r}
library(Seurat)
library(tidyverse)
library(magrittr)
library(ReactomePA)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(DOSE)
mk <- FindAllMarkers(ofav.combined.labeled, min.pct = 0.01, return.thresh=0.05, only.pos = T)
mk <- mk[mk$p_val_adj <= 0.05, ]
formatted_data <- list()

# Iterate over clusters and format data for each cluster
for (cluster_id in unique(mk$cluster)) {
  cluster_genes <- mk$gene[mk$cluster == cluster_id]
  cluster_pvalues <- mk$p_val_adj[mk$cluster == cluster_id]
  
  # Create a data frame with gene names as row names and P-values as numeric values
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  
  # Append to the list
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}

# Combine the list into a single named list
genes_per_cluster <- formatted_data
```

```{r}
library("topGO")
gocluster1 <- genes_per_cluster[["Calicoblastic Cells"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkorchid4") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Calicoblastic Cells") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_CalicoblasticCells.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```


```{r}
gocluster1 <- genes_per_cluster[["Digestive Filaments 1"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "red1") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Digestive Filaments 1") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Digestive_Filaments_1.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Cnidocytes"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "plum1") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Cnidocytes") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Cnidocytes.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Digestive Filaments 2"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "red3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Digestive Filaments 2") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Digestive_Filaments_2.pdf", width=25, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Gastrodermis (Muscle-like)"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkolivegreen4") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Gastrodermis (Muscle-like)") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Gastro-musclelike.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Gastrodermis 1"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkolivegreen1") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Gastrodermis 1") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Gastro1.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Gastrodermis 2"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkolivegreen2") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Gastrodermis 2") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Gastro2.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Gastrodermis 3"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkolivegreen3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Gastrodermis 3") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Gastro3.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Gland 1"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkorange1") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Gland 1") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Gland1.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Gland 2"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkorange3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Gland 2") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Gland2.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Immune"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkkhaki") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Immune") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Immune.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Neurons 1"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "deepskyblue3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Neurons 1") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Neurons1.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Neurons 2"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "deepskyblue4") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Neurons 2") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Neurons2.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Neurons 3"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "deepskyblue2") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Neurons 3") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Neurons3.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Precursors"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "bisque3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Precursors") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Precursors.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

## GO Symbionts
```{r}
library("topGO")
gocluster1 <- genes_per_cluster[["In-hospite Breviolum"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol_bmin.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkgoldenrod3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO In-hospite Breviolum Cells (Breviolum expression)") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()
pdf("figures/GO/GO_In-hospite-Breviolum.pdf", width=20, height=10)
plot(GO_plot)
dev.off()
```
```{r}
library("topGO")
gocluster1 <- genes_per_cluster[["Expelled Breviolum"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol_bmin.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkgoldenrod1") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Expelled Breviolum Cells (Breviolum expression)") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()
pdf("figures/GO/GO_Expelled-Breviolum.pdf", width=20, height=10)
plot(GO_plot)
dev.off()
```

```{r}
library("topGO")
gocluster1 <- genes_per_cluster[["In-hospite Durusdinium"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol_dtre.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkgoldenrod4") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO In-hospite Durusdinium Cells (Durusdinium expression)") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()
pdf("figures/GO/GO_Durusdinium_symbexp.pdf", width=20, height=10)
plot(GO_plot)
dev.off()
```

#### Compare breviolums
```{r}
levels(ofav.combined.labeled)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled, idents = c("Expelled Breviolum", "In-hospite Breviolum"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["Expelled Breviolum"]]

text_data <- read.csv(file = "id2symbol_bmin.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkgoldenrod") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Expelled vs In-hospite Breviolum") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Breviolum_compared2eachother.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

#### Compare symbionts
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Breviolum-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T5-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["Breviolum-hosting Cells_Ofav-T7-T"]]

text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkgoldenrod") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("Breviolum-hosting Cells vs Durusdinium-hosting Cells- Healthy") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_BrevVSDurs_compared2eachother_healthy.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```


### Symbiont GOs between Timepoints
```{r}
library(Seurat)
library(tidyverse)
library(magrittr)
library(ReactomePA)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(DOSE)
library("topGO")
ofav.combined.labeled.DE <- subset(x = ofav.combined.labeled, idents = c("Unidentified"), invert = TRUE)
ofav.combined.labeled.DE$super_clusters <- paste(Idents(ofav.combined.labeled.DE), ofav.combined.labeled.DE$orig.ident, sep = "_")
Idents(ofav.combined.labeled.DE) <- "super_clusters"
levels(ofav.combined.labeled.DE)
```

#### In-hospite Breviolum
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["In-hospite Breviolum_Ofav-T3-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "#ADC397") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO In-hospite Breviolum", subtitle =  "Coral Symbiosome Early Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_Brev_early_coral.pdf", width=20, height=10)
plot(GO_plot)
dev.off()

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T5-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["In-hospite Breviolum_Ofav-T5-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "#E5A208") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO In-hospite Breviolum", subtitle =  "Coral Symbiosome Secondary Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_Brev_secondary_coral.pdf", width=20, height=10)
plot(GO_plot)
dev.off()

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T7-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["In-hospite Breviolum_Ofav-T7-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "#F11B00") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO In-hospite Breviolum", subtitle =  "Coral Symbiosome Full Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_Brev_full_coral.pdf", width=20, height=10)
plot(GO_plot)
dev.off()
```





#### Expelled Breviolum
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["Expelled Breviolum_Ofav-T3-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol_bmin.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "#ADC397") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Expelled Breviolum", subtitle =  "Symbiont Early Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_ExBrev_early.pdf", width=20, height=10)
plot(GO_plot)
dev.off()

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T5-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["Expelled Breviolum_Ofav-T5-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol_bmin.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "#E5A208") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Expelled Breviolum", subtitle =  "Symbiont Secondary Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_ExBrev_secondary.pdf", width=20, height=10)
plot(GO_plot)
dev.off()

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T7-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["Expelled Breviolum_Ofav-T0-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol_bmin.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "#F11B00") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Expelled Breviolum", subtitle =  "Symbiont Full Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_ExBrev_full.pdf", width=20, height=10)
plot(GO_plot)
dev.off()
```

## Combine Cell Types for Go Analysis (test here with gastrodermis)- looks good
```{r}
gocluster1_1 <- genes_per_cluster[["Neurons 1"]]
gocluster1_2 <- genes_per_cluster[["Neurons 2"]]
gocluster1_3 <- genes_per_cluster[["Neurons 3"]]
gocluster1 <- rbind(gocluster1_1, gocluster1_2, gocluster1_3)
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "deepskyblue") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Neuron Clusters") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Neurons-combined.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1 <- genes_per_cluster[["Epidermis"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "red") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("Epidermis") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_epidermis.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

```{r}
gocluster1_1 <- genes_per_cluster[["Gland 1"]]
gocluster1_2 <- genes_per_cluster[["Gland 2"]]
gocluster1_3 <- genes_per_cluster[["Gland 3"]]
gocluster1 <- rbind(gocluster1_1, gocluster1_2)
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkorange") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Gland Clusters") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Gland-combined.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```
```{r}
gocluster1_1 <- genes_per_cluster[["Gastrodermis 1"]]
gocluster1_2 <- genes_per_cluster[["Gastrodermis 2"]]
gocluster1 <- rbind(gocluster1_1, gocluster1_2)
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)

selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))

GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkorange") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("Gastrodermis") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/GO/GO_Gastrodermis-combined.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```
# Coral Cell DE Analysis
## Remove Unidentified cells for greater power
```{r}
ofav.combined.labeled.DE <- subset(x = ofav.combined.labeled, idents = c("Unidentified"), invert = TRUE)
```
### DE Analysis (Cluster-level)
```{r}
ofav.combined.labeled.DE$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.DE), ofav.combined.labeled.DE$orig.ident, sep = "_")
unique(ofav.combined.labeled.DE$seurat_clusters.Timepoints)
Idents(ofav.combined.labeled.DE) <- "seurat_clusters.Timepoints"
```
#### Calicoblastic Cells
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T3-T", "Calicoblastic Cells_Ofav-T5-T", "Calicoblastic Cells_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T3-T", "Calicoblastic Cells_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T5-T", "Calicoblastic Cells_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_CalicoblasticCells.csv")
```
#### Cnidocytes
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T3-T", "Cnidocytes_Ofav-T5-T", "Cnidocytes_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T3-T", "Cnidocytes_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T5-T", "Cnidocytes_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Cnidocytes.csv")
```
#### Digestive Filaments 1
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 1_Ofav-T3-T", "Digestive Filaments 1_Ofav-T5-T", "Digestive Filaments 1_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 1_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 1_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 1_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T3-T", "Digestive Filaments 1_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T5-T", "Digestive Filaments 1_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Digestive_Filaments_1.csv")
```
#### Digestive Filaments 2
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T0-T", "Digestive Filaments 2_Ofav-T3-T", "Digestive Filaments 2_Ofav-T5-T", "Digestive Filaments 2_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T0-T", "Digestive Filaments 2_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T0-T", "Digestive Filaments 2_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T0-T", "Digestive Filaments 2_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T3-T", "Digestive Filaments 2_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T5-T", "Digestive Filaments 2_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Digestive_Filaments_2.csv")
```
#### Gastrodermis (Muscle-like)
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_GastroMuscle.csv")
```
#### Gastrodermis 1
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 1_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T3-T", "Gastrodermis 1_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T5-T", "Gastrodermis 1_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Gastrodermis1.csv")
```
#### Gastrodermis 2
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T3-T", "Gastrodermis 2_Ofav-T5-T", "Gastrodermis 2_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T3-T", "Gastrodermis 2_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T5-T", "Gastrodermis 2_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Gastrodermis2.csv")
```
#### Gastrodermis 3
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T0-T", "Gastrodermis 3_Ofav-T3-T", "Gastrodermis 3_Ofav-T5-T", "Gastrodermis 3_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T0-T", "Gastrodermis 3_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T0-T", "Gastrodermis 3_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T0-T", "Gastrodermis 3_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T3-T", "Gastrodermis 3_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T5-T", "Gastrodermis 3_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Gastrodermis3.csv")
```
#### Gland 1
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T0-T", "Gland 1_Ofav-T3-T", "Gland 1_Ofav-T5-T", "Gland 1_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T0-T", "Gland 1_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T0-T", "Gland 1_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T0-T", "Gland 1_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T3-T", "Gland 1_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T5-T", "Gland 1_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Gland1.csv")
```
#### Gland 2
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T0-T", "Gland 2_Ofav-T3-T", "Gland 2_Ofav-T5-T", "Gland 2_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T0-T", "Gland 2_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T0-T", "Gland 2_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T0-T", "Gland 2_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T3-T", "Gland 2_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T5-T", "Gland 2_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Gland2.csv")
```
#### Immune
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T3-T", "Immune_Ofav-T5-T", "Immune_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T3-T", "Immune_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T5-T", "Immune_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Immune.csv")
```
#### Neurons 1
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T0-T", "Neurons 1_Ofav-T3-T", "Neurons 1_Ofav-T5-T", "Neurons 1_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T0-T", "Neurons 1_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T0-T", "Neurons 1_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T0-T", "Neurons 1_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T3-T", "Neurons 1_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T5-T", "Neurons 1_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Neurons1.csv")
```
#### Neurons 2
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T0-T", "Neurons 2_Ofav-T3-T", "Neurons 2_Ofav-T5-T", "Neurons 2_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T0-T", "Neurons 2_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T0-T", "Neurons 2_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T0-T", "Neurons 2_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T3-T", "Neurons 2_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T5-T", "Neurons 2_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Neurons2.csv")
```
#### Neurons 3
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T0-T", "Neurons 3_Ofav-T3-T", "Neurons 3_Ofav-T5-T", "Neurons 3_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T0-T", "Neurons 3_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T0-T", "Neurons 3_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T0-T", "Neurons 3_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T3-T", "Neurons 3_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T5-T", "Neurons 3_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Neurons3.csv")
```
#### Precursors
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T3-T", "Precursors_Ofav-T5-T", "Precursors_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T3-T", "Precursors_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T5-T", "Precursors_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Precursors.csv")
```
#### Breviolum
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T3-T", "In-hospite Breviolum_Ofav-T5-T", "In-hospite Breviolum_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T3-T", "In-hospite Breviolum_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T5-T", "In-hospite Breviolum_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_In-hospite_Breviolum.csv")
```
#### Durusdinium
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T3-T", "In-hospite Durusdinium_Ofav-T5-T", "In-hospite Durusdinium_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T3-T", "In-hospite Durusdinium_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T5-T", "In-hospite Durusdinium_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_In-hospite_Durusdinium.csv")
```
#### Explelled Breviolum
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T3-T", "Expelled Breviolum_Ofav-T5-T", "Expelled Breviolum_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T3-T", "Expelled Breviolum_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T5-T", "Expelled Breviolum_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Expelled_breviolum.csv")
```

#### Symbiosome Comparisons
```{r}
Idents(ofav.combined.labeled.DE)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Breviolum-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T0-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Healthy State"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Breviolum-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T3-T"))
heatstress.response3<- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching Stage"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching Stage"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Breviolum-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T5-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late Bleaching stage "

heatstress.response <- rbind( heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_Symbiont_DEGS/DE_BrevVSDur_combined.csv")
```



## DE All Together for Table Analysis
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T3-T"))
heatstress.response2.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response2.1 <- heatstress.response2.1[heatstress.response2.1$p_val_adj <= 0.05, ]
heatstress.response2.1 <- heatstress.response2.1[order(heatstress.response2.1$p_val_adj), ]

heatstress.response2.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T5-T"))
heatstress.response2.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response2.2 <- heatstress.response2.2[heatstress.response2.2$p_val_adj <= 0.05, ]
heatstress.response2.2 <- heatstress.response2.2[order(heatstress.response2.2$p_val_adj), ]

heatstress.response2.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T7-T"))
heatstress.response2.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response2.3 <- heatstress.response2.3[heatstress.response2.3$p_val_adj <= 0.05, ]
heatstress.response2.3 <- heatstress.response2.3[order(heatstress.response2.3$p_val_adj), ]

heatstress.response2.3$grouping <- "Full Bleaching Marker"

heatstress.response2 <- rbind(heatstress.response2.1, heatstress.response2.2, heatstress.response2.3)

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T3-T"))
heatstress.response3.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response3.1 <- heatstress.response3.1[heatstress.response3.1$p_val_adj <= 0.05, ]
heatstress.response3.1 <- heatstress.response3.1[order(heatstress.response3.1$p_val_adj), ]

heatstress.response3.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T5-T"))
heatstress.response3.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response3.2 <- heatstress.response3.2[heatstress.response3.2$p_val_adj <= 0.05, ]
heatstress.response3.2 <- heatstress.response3.2[order(heatstress.response3.2$p_val_adj), ]

heatstress.response3.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T7-T"))
heatstress.response3.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response3.3 <- heatstress.response3.3[heatstress.response3.3$p_val_adj <= 0.05, ]
heatstress.response3.3 <- heatstress.response3.3[order(heatstress.response3.3$p_val_adj), ]

heatstress.response3.3$grouping <- "Full Bleaching Marker"

heatstress.response3 <- rbind(heatstress.response3.1, heatstress.response3.2, heatstress.response3.3)

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 1_Ofav-T3-T"))
heatstress.response4.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response4.1 <- heatstress.response4.1[heatstress.response4.1$p_val_adj <= 0.05, ]
heatstress.response4.1 <- heatstress.response4.1[order(heatstress.response4.1$p_val_adj), ]

heatstress.response4.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 1_Ofav-T5-T"))
heatstress.response4.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response4.2 <- heatstress.response4.2[heatstress.response4.2$p_val_adj <= 0.05, ]
heatstress.response4.2 <- heatstress.response4.2[order(heatstress.response4.2$p_val_adj), ]

heatstress.response4.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 1_Ofav-T7-T"))
heatstress.response4.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response4.3 <- heatstress.response4.3[heatstress.response4.3$p_val_adj <= 0.05, ]
heatstress.response4.3 <- heatstress.response4.3[order(heatstress.response4.3$p_val_adj), ]

heatstress.response4.3$grouping <- "Full Bleaching Marker"

heatstress.response4 <- rbind(heatstress.response4.1, heatstress.response4.2, heatstress.response4.3)

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T0-T", "Digestive Filaments 2_Ofav-T3-T"))
heatstress.response5.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response5.1 <- heatstress.response5.1[heatstress.response5.1$p_val_adj <= 0.05, ]
heatstress.response5.1 <- heatstress.response5.1[order(heatstress.response5.1$p_val_adj), ]

heatstress.response5.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T0-T", "Digestive Filaments 2_Ofav-T5-T"))
heatstress.response5.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response5.2 <- heatstress.response5.2[heatstress.response5.2$p_val_adj <= 0.05, ]
heatstress.response5.2 <- heatstress.response5.2[order(heatstress.response5.2$p_val_adj), ]

heatstress.response5.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments 2_Ofav-T0-T", "Digestive Filaments 2_Ofav-T7-T"))
heatstress.response5.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response5.3 <- heatstress.response5.3[heatstress.response5.3$p_val_adj <= 0.05, ]
heatstress.response5.3 <- heatstress.response5.3[order(heatstress.response5.3$p_val_adj), ]

heatstress.response5.3$grouping <- "Full Bleaching Marker"

heatstress.response5 <- rbind(heatstress.response5.1, heatstress.response5.2, heatstress.response5.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T3-T"))
heatstress.response6.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response6.1 <- heatstress.response6.1[heatstress.response6.1$p_val_adj <= 0.05, ]
heatstress.response6.1 <- heatstress.response6.1[order(heatstress.response6.1$p_val_adj), ]

heatstress.response6.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T5-T"))
heatstress.response6.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response6.2 <- heatstress.response6.2[heatstress.response6.2$p_val_adj <= 0.05, ]
heatstress.response6.2 <- heatstress.response6.2[order(heatstress.response6.2$p_val_adj), ]

heatstress.response6.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T7-T"))
heatstress.response6.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response6.3 <- heatstress.response6.3[heatstress.response6.3$p_val_adj <= 0.05, ]
heatstress.response6.3 <- heatstress.response6.3[order(heatstress.response6.3$p_val_adj), ]

heatstress.response6.3$grouping <- "Full Bleaching Marker"

heatstress.response6 <- rbind(heatstress.response6.1, heatstress.response6.2, heatstress.response6.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T3-T"))
heatstress.response7.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response7.1 <- heatstress.response7.1[heatstress.response7.1$p_val_adj <= 0.05, ]
heatstress.response7.1 <- heatstress.response7.1[order(heatstress.response7.1$p_val_adj), ]

heatstress.response7.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T5-T"))
heatstress.response7.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response7.2 <- heatstress.response7.2[heatstress.response7.2$p_val_adj <= 0.05, ]
heatstress.response7.2 <- heatstress.response7.2[order(heatstress.response7.2$p_val_adj), ]

heatstress.response7.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T7-T"))
heatstress.response7.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response7.3 <- heatstress.response7.3[heatstress.response7.3$p_val_adj <= 0.05, ]
heatstress.response7.3 <- heatstress.response7.3[order(heatstress.response7.3$p_val_adj), ]

heatstress.response7.3$grouping <- "Full Bleaching Marker"

heatstress.response7 <- rbind(heatstress.response7.1, heatstress.response7.2, heatstress.response7.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T3-T"))
heatstress.response8.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response8.1 <- heatstress.response8.1[heatstress.response8.1$p_val_adj <= 0.05, ]
heatstress.response8.1 <- heatstress.response8.1[order(heatstress.response8.1$p_val_adj), ]

heatstress.response8.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T5-T"))
heatstress.response8.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response8.2 <- heatstress.response8.2[heatstress.response8.2$p_val_adj <= 0.05, ]
heatstress.response8.2 <- heatstress.response8.2[order(heatstress.response8.2$p_val_adj), ]

heatstress.response8.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T7-T"))
heatstress.response8.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response8.3 <- heatstress.response8.3[heatstress.response8.3$p_val_adj <= 0.05, ]
heatstress.response8.3 <- heatstress.response8.3[order(heatstress.response8.3$p_val_adj), ]

heatstress.response8.3$grouping <- "Full Bleaching Marker"

heatstress.response8 <- rbind(heatstress.response8.1, heatstress.response8.2, heatstress.response8.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T0-T", "Gastrodermis 3_Ofav-T3-T"))
heatstress.response9.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response9.1 <- heatstress.response9.1[heatstress.response9.1$p_val_adj <= 0.05, ]
heatstress.response9.1 <- heatstress.response9.1[order(heatstress.response9.1$p_val_adj), ]

heatstress.response9.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T0-T", "Gastrodermis 3_Ofav-T5-T"))
heatstress.response9.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response9.2 <- heatstress.response9.2[heatstress.response9.2$p_val_adj <= 0.05, ]
heatstress.response9.2 <- heatstress.response9.2[order(heatstress.response9.2$p_val_adj), ]

heatstress.response9.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis 3_Ofav-T0-T", "Gastrodermis 3_Ofav-T7-T"))
heatstress.response9.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response9.3 <- heatstress.response9.3[heatstress.response9.3$p_val_adj <= 0.05, ]
heatstress.response9.3 <- heatstress.response9.3[order(heatstress.response9.3$p_val_adj), ]

heatstress.response9.3$grouping <- "Full Bleaching Marker"

heatstress.response9 <- rbind(heatstress.response9.1, heatstress.response9.2, heatstress.response9.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T0-T", "Gland 1_Ofav-T3-T"))
heatstress.response10.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response10.1 <- heatstress.response10.1[heatstress.response10.1$p_val_adj <= 0.05, ]
heatstress.response10.1 <- heatstress.response10.1[order(heatstress.response10.1$p_val_adj), ]

heatstress.response10.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T0-T", "Gland 1_Ofav-T5-T"))
heatstress.response10.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response10.2 <- heatstress.response10.2[heatstress.response10.2$p_val_adj <= 0.05, ]
heatstress.response10.2 <- heatstress.response10.2[order(heatstress.response10.2$p_val_adj), ]

heatstress.response10.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 1_Ofav-T0-T", "Gland 1_Ofav-T7-T"))
heatstress.response10.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response10.3 <- heatstress.response10.3[heatstress.response10.3$p_val_adj <= 0.05, ]
heatstress.response10.3 <- heatstress.response10.3[order(heatstress.response10.3$p_val_adj), ]

heatstress.response10.3$grouping <- "Full Bleaching Marker"

heatstress.response10 <- rbind(heatstress.response10.1, heatstress.response10.2, heatstress.response10.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T0-T", "Gland 2_Ofav-T3-T"))
heatstress.response11.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response11.1 <- heatstress.response11.1[heatstress.response11.1$p_val_adj <= 0.05, ]
heatstress.response11.1 <- heatstress.response11.1[order(heatstress.response11.1$p_val_adj), ]

heatstress.response11.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T0-T", "Gland 2_Ofav-T5-T"))
heatstress.response11.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response11.2 <- heatstress.response11.2[heatstress.response11.2$p_val_adj <= 0.05, ]
heatstress.response11.2 <- heatstress.response11.2[order(heatstress.response11.2$p_val_adj), ]

heatstress.response11.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland 2_Ofav-T0-T", "Gland 2_Ofav-T7-T"))
heatstress.response11.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response11.3 <- heatstress.response11.3[heatstress.response11.3$p_val_adj <= 0.05, ]
heatstress.response11.3 <- heatstress.response11.3[order(heatstress.response11.3$p_val_adj), ]

heatstress.response11.3$grouping <- "Full Bleaching Marker"

heatstress.response11 <- rbind(heatstress.response11.1, heatstress.response11.2, heatstress.response11.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T3-T"))
heatstress.response12.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response12.1 <- heatstress.response12.1[heatstress.response12.1$p_val_adj <= 0.05, ]
heatstress.response12.1 <- heatstress.response12.1[order(heatstress.response12.1$p_val_adj), ]

heatstress.response12.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T5-T"))
heatstress.response12.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response12.2 <- heatstress.response12.2[heatstress.response12.2$p_val_adj <= 0.05, ]
heatstress.response12.2 <- heatstress.response12.2[order(heatstress.response12.2$p_val_adj), ]

heatstress.response12.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T7-T"))
heatstress.response12.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response12.3 <- heatstress.response12.3[heatstress.response12.3$p_val_adj <= 0.05, ]
heatstress.response12.3 <- heatstress.response12.3[order(heatstress.response12.3$p_val_adj), ]

heatstress.response12.3$grouping <- "Full Bleaching Marker"

heatstress.response12 <- rbind(heatstress.response12.1, heatstress.response12.2, heatstress.response12.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T0-T", "Neurons 1_Ofav-T3-T"))
heatstress.response13.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response13.1 <- heatstress.response13.1[heatstress.response13.1$p_val_adj <= 0.05, ]
heatstress.response13.1 <- heatstress.response13.1[order(heatstress.response13.1$p_val_adj), ]

heatstress.response13.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T0-T", "Neurons 1_Ofav-T5-T"))
heatstress.response13.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response13.2 <- heatstress.response13.2[heatstress.response13.2$p_val_adj <= 0.05, ]
heatstress.response13.2 <- heatstress.response13.2[order(heatstress.response13.2$p_val_adj), ]

heatstress.response13.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 1_Ofav-T0-T", "Neurons 1_Ofav-T7-T"))
heatstress.response13.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response13.3 <- heatstress.response13.3[heatstress.response13.3$p_val_adj <= 0.05, ]
heatstress.response13.3 <- heatstress.response13.3[order(heatstress.response13.3$p_val_adj), ]

heatstress.response13.3$grouping <- "Full Bleaching Marker"

heatstress.response13 <- rbind(heatstress.response13.1, heatstress.response13.2, heatstress.response13.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T0-T", "Neurons 2_Ofav-T3-T"))
heatstress.response14.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response14.1 <- heatstress.response14.1[heatstress.response14.1$p_val_adj <= 0.05, ]
heatstress.response14.1 <- heatstress.response14.1[order(heatstress.response14.1$p_val_adj), ]

heatstress.response14.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T0-T", "Neurons 2_Ofav-T5-T"))
heatstress.response14.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response14.2 <- heatstress.response14.2[heatstress.response14.2$p_val_adj <= 0.05, ]
heatstress.response14.2 <- heatstress.response14.2[order(heatstress.response14.2$p_val_adj), ]

heatstress.response14.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 2_Ofav-T0-T", "Neurons 2_Ofav-T7-T"))
heatstress.response14.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response14.3 <- heatstress.response14.3[heatstress.response14.3$p_val_adj <= 0.05, ]
heatstress.response14.3 <- heatstress.response14.3[order(heatstress.response14.3$p_val_adj), ]

heatstress.response14.3$grouping <- "Full Bleaching Marker"

heatstress.response14 <- rbind(heatstress.response14.1, heatstress.response14.2, heatstress.response14.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T0-T", "Neurons 3_Ofav-T3-T"))
heatstress.response15.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response15.1 <- heatstress.response15.1[heatstress.response15.1$p_val_adj <= 0.05, ]
heatstress.response15.1 <- heatstress.response15.1[order(heatstress.response15.1$p_val_adj), ]

heatstress.response15.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T0-T", "Neurons 3_Ofav-T5-T"))
heatstress.response15.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response15.2 <- heatstress.response15.2[heatstress.response15.2$p_val_adj <= 0.05, ]
heatstress.response15.2 <- heatstress.response15.2[order(heatstress.response15.2$p_val_adj), ]

heatstress.response15.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons 3_Ofav-T0-T", "Neurons 3_Ofav-T7-T"))
heatstress.response15.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response15.3 <- heatstress.response15.3[heatstress.response15.3$p_val_adj <= 0.05, ]
heatstress.response15.3 <- heatstress.response15.3[order(heatstress.response15.3$p_val_adj), ]

heatstress.response15.3$grouping <- "Full Bleaching Marker"

heatstress.response15 <- rbind(heatstress.response15.1, heatstress.response15.2, heatstress.response15.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T3-T"))
heatstress.response16.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response16.1 <- heatstress.response16.1[heatstress.response16.1$p_val_adj <= 0.05, ]
heatstress.response16.1 <- heatstress.response16.1[order(heatstress.response16.1$p_val_adj), ]

heatstress.response16.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T5-T"))
heatstress.response16.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response16.2 <- heatstress.response16.2[heatstress.response16.2$p_val_adj <= 0.05, ]
heatstress.response16.2 <- heatstress.response16.2[order(heatstress.response16.2$p_val_adj), ]

heatstress.response16.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T7-T"))
heatstress.response16.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response16.3 <- heatstress.response16.3[heatstress.response16.3$p_val_adj <= 0.05, ]
heatstress.response16.3 <- heatstress.response16.3[order(heatstress.response16.3$p_val_adj), ]

heatstress.response16.3$grouping <- "Full Bleaching Marker"

heatstress.response16 <- rbind(heatstress.response16.1, heatstress.response16.2, heatstress.response16.3)

heatstress.response <- rbind(heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6, heatstress.response7, heatstress.response8, heatstress.response9, heatstress.response10, heatstress.response11, heatstress.response12, heatstress.response13, heatstress.response14, heatstress.response15, heatstress.response16)


heatstress.response_early <- heatstress.response[heatstress.response$grouping== "Early Bleaching Marker", ]
write.csv(heatstress.response_early, "figures/Bleaching_DEGS/DE_CoralALL_earlymarkers.csv")

heatstress.response_late <- heatstress.response[heatstress.response$grouping== "Secondary Early Bleaching Marker", ]
write.csv(heatstress.response_late, "figures/Bleaching_DEGS/DE_CoralALL_secondarymarkers.csv")

heatstress.response_full <- heatstress.response[heatstress.response$grouping== "Full Bleaching Marker", ]
write.csv(heatstress.response_full, "figures/Bleaching_DEGS/DE_CoralALL_fullbleachingmarkers.csv")


## For symbiont I need a minmum value in pct.2
heatstress.response2 <- rbind(heatstress.response2.1, heatstress.response2.2, heatstress.response2.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T3-T"))
heatstress.response2.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response2.1 <- heatstress.response2.1[heatstress.response2.1$p_val_adj <= 0.05, ]
heatstress.response2.1 <- heatstress.response2.1[heatstress.response2.1$pct.2 >= 0.01, ]
heatstress.response2.1 <- heatstress.response2.1[order(heatstress.response2.1$p_val_adj), ]
heatstress.response2.1 <- head(heatstress.response2.1, 20)
heatstress.response2.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T5-T"))
heatstress.response2.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response2.2 <- heatstress.response2.2[heatstress.response2.2$p_val_adj <= 0.05, ]
heatstress.response2.2 <- heatstress.response2.2[heatstress.response2.2$pct.2 >= 0.01, ]
heatstress.response2.2 <- heatstress.response2.2[order(heatstress.response2.2$p_val_adj), ]
heatstress.response2.2 <- head(heatstress.response2.2, 20)
heatstress.response2.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Expelled Breviolum_Ofav-T0-T", "Expelled Breviolum_Ofav-T7-T"))
heatstress.response2.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response2.3 <- heatstress.response2.3[heatstress.response2.3$p_val_adj <= 0.05, ]
heatstress.response2.3 <- heatstress.response2.3[heatstress.response2.3$pct.2 >= 0.01, ]
heatstress.response2.3 <- heatstress.response2.3[order(heatstress.response2.3$p_val_adj), ]
heatstress.response2.3 <- head(heatstress.response2.3, 20)
heatstress.response2.3$grouping <- "Full Bleaching Marker"

heatstress.response2 <- rbind(heatstress.response2.1, heatstress.response2.2, heatstress.response2.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T3-T"))
heatstress.response3.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response3.1 <- heatstress.response3.1[heatstress.response3.1$p_val_adj <= 0.05, ]
heatstress.response3.1 <- heatstress.response3.1[heatstress.response3.1$pct.2 >= 0.01, ]
heatstress.response3.1 <- heatstress.response3.1[order(heatstress.response3.1$p_val_adj), ]
heatstress.response3.1 <- head(heatstress.response3.1, 20)
heatstress.response3.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T5-T"))
heatstress.response3.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response3.2 <- heatstress.response3.2[heatstress.response3.2$p_val_adj <= 0.05, ]
heatstress.response3.2 <- heatstress.response3.2[heatstress.response3.2$pct.2 >= 0.01, ]
heatstress.response3.2 <- heatstress.response3.2[order(heatstress.response3.2$p_val_adj), ]
heatstress.response3.2 <- head(heatstress.response3.2, 20)
heatstress.response3.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T7-T"))
heatstress.response3.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response3.3 <- heatstress.response3.3[heatstress.response3.3$p_val_adj <= 0.05, ]
heatstress.response3.3 <- heatstress.response3.3[heatstress.response3.3$pct.2 >= 0.01, ]
heatstress.response3.3 <- heatstress.response3.3[order(heatstress.response3.3$p_val_adj), ]
heatstress.response3.3 <- head(heatstress.response3.3, 20)
heatstress.response3.3$grouping <- "Full Bleaching Marker"

heatstress.response3 <- rbind(heatstress.response3.1, heatstress.response3.2, heatstress.response3.3)

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T3-T"))
heatstress.response4.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response4.1 <- heatstress.response4.1[heatstress.response4.1$p_val_adj <= 0.05, ]
heatstress.response4.1 <- heatstress.response4.1[heatstress.response4.1$pct.2 >= 0.01, ]
heatstress.response4.1 <- heatstress.response4.1[order(heatstress.response4.1$p_val_adj), ]
heatstress.response4.1 <- head(heatstress.response4.1, 20)
heatstress.response4.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T5-T"))
heatstress.response4.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response4.2 <- heatstress.response4.2[heatstress.response4.2$p_val_adj <= 0.05, ]
heatstress.response4.2 <- heatstress.response4.2[heatstress.response4.2$pct.2 >= 0.01, ]
heatstress.response4.2 <- heatstress.response4.2[order(heatstress.response4.2$p_val_adj), ]
heatstress.response4.2 <- head(heatstress.response4.2, 20)
heatstress.response4.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Durusdinium_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T7-T"))
heatstress.response4.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response4.3 <- heatstress.response4.3[heatstress.response4.3$p_val_adj <= 0.05, ]
heatstress.response4.3 <- heatstress.response4.3[heatstress.response4.3$pct.2 >= 0.01, ]
heatstress.response4.3 <- heatstress.response4.3[order(heatstress.response4.3$p_val_adj), ]
heatstress.response4.3 <- head(heatstress.response4.3, 20)
heatstress.response4.3$grouping <- "Full Bleaching Marker"

heatstress.response4 <- rbind(heatstress.response4.1, heatstress.response4.2, heatstress.response4.3)

heatstress.response <- rbind(heatstress.response2, heatstress.response3, heatstress.response4)

write.csv(heatstress.response, "figures/Bleaching_DEGS/DE_SymbiontsALL_Table2.csv")
```

# DE Super Clusters All Together for Table Analysis
```{r}
levels(ofav.combined.labeled)
ofav.combined.labeled.SC <- RenameIdents(ofav.combined.labeled, "Unidentified"="Unidentified", "Calicoblastic Cells"="Calicoblastic Cells", "Cnidocytes"="Cnidocytes", "Digestive Filaments"="Digestive Filaments", "Epidermis"="Epidermis", "Gastrodermis (Muscle-like)"="Gastrodermis (Muscle-like)", "Gastrodermis 1"="Gastrodermis", "Gastrodermis 2"="Gastrodermis", "Gland 1"="Gland", "Gland 2"="Gland", "Gland 3"="Gland", "Immune"="Immune", "Neurons 1"="Neurons", "Neurons 2"="Neurons", "Neurons 3"="Neurons", "Precursors"="Precursors", "Extracellular Breviolum"="Breviolum", "Breviolum-hosting Cells"="Breviolum", "Durusdinium-hosting Cells"=	"Durusdinium")
levels(ofav.combined.labeled.SC)

ofav.combined.labeled.SC$super_clusters <- paste(Idents(ofav.combined.labeled.SC))
unique(ofav.combined.labeled.SC$super_clusters)
Idents(ofav.combined.labeled.SC) <- "super_clusters"
levels(Idents(ofav.combined.labeled.SC))

ofav.combined.labeled.SC$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.SC), ofav.combined.labeled.SC$orig.ident, sep = "_")
unique(ofav.combined.labeled.SC$seurat_clusters.Timepoints)
Idents(ofav.combined.labeled.SC) <- "seurat_clusters.Timepoints"

ofav.combined.labeled.DE <- ofav.combined.labeled.SC
levels(ofav.combined.labeled.DE)

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T3-T"))
heatstress.response2.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response2.1 <- heatstress.response2.1[heatstress.response2.1$p_val_adj <= 0.05, ]
heatstress.response2.1 <- heatstress.response2.1[order(heatstress.response2.1$p_val_adj), ]

heatstress.response2.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T5-T"))
heatstress.response2.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response2.2 <- heatstress.response2.2[heatstress.response2.2$p_val_adj <= 0.05, ]
heatstress.response2.2 <- heatstress.response2.2[order(heatstress.response2.2$p_val_adj), ]

heatstress.response2.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Calicoblastic Cells_Ofav-T0-T", "Calicoblastic Cells_Ofav-T7-T"))
heatstress.response2.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response2.3 <- heatstress.response2.3[heatstress.response2.3$p_val_adj <= 0.05, ]
heatstress.response2.3 <- heatstress.response2.3[order(heatstress.response2.3$p_val_adj), ]

heatstress.response2.3$grouping <- "Full Bleaching Marker"

heatstress.response2 <- rbind(heatstress.response2.1, heatstress.response2.2, heatstress.response2.3)

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T3-T"))
heatstress.response3.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response3.1 <- heatstress.response3.1[heatstress.response3.1$p_val_adj <= 0.05, ]
heatstress.response3.1 <- heatstress.response3.1[order(heatstress.response3.1$p_val_adj), ]

heatstress.response3.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T5-T"))
heatstress.response3.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response3.2 <- heatstress.response3.2[heatstress.response3.2$p_val_adj <= 0.05, ]
heatstress.response3.2 <- heatstress.response3.2[order(heatstress.response3.2$p_val_adj), ]

heatstress.response3.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Cnidocytes_Ofav-T0-T", "Cnidocytes_Ofav-T7-T"))
heatstress.response3.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response3.3 <- heatstress.response3.3[heatstress.response3.3$p_val_adj <= 0.05, ]
heatstress.response3.3 <- heatstress.response3.3[order(heatstress.response3.3$p_val_adj), ]

heatstress.response3.3$grouping <- "Full Bleaching Marker"

heatstress.response3 <- rbind(heatstress.response3.1, heatstress.response3.2, heatstress.response3.3)

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments_Ofav-T0-T", "Digestive Filaments_Ofav-T3-T"))
heatstress.response4.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response4.1 <- heatstress.response4.1[heatstress.response4.1$p_val_adj <= 0.05, ]
heatstress.response4.1 <- heatstress.response4.1[order(heatstress.response4.1$p_val_adj), ]

heatstress.response4.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments_Ofav-T0-T", "Digestive Filaments_Ofav-T5-T"))
heatstress.response4.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response4.2 <- heatstress.response4.2[heatstress.response4.2$p_val_adj <= 0.05, ]
heatstress.response4.2 <- heatstress.response4.2[order(heatstress.response4.2$p_val_adj), ]

heatstress.response4.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Digestive Filaments_Ofav-T0-T", "Digestive Filaments_Ofav-T7-T"))
heatstress.response4.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response4.3 <- heatstress.response4.3[heatstress.response4.3$p_val_adj <= 0.05, ]
heatstress.response4.3 <- heatstress.response4.3[order(heatstress.response4.3$p_val_adj), ]

heatstress.response4.3$grouping <- "Full Bleaching Marker"

heatstress.response4 <- rbind(heatstress.response4.1, heatstress.response4.2, heatstress.response4.3)


ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T3-T"))
heatstress.response6.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response6.1 <- heatstress.response6.1[heatstress.response6.1$p_val_adj <= 0.05, ]
heatstress.response6.1 <- heatstress.response6.1[order(heatstress.response6.1$p_val_adj), ]

heatstress.response6.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T5-T"))
heatstress.response6.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response6.2 <- heatstress.response6.2[heatstress.response6.2$p_val_adj <= 0.05, ]
heatstress.response6.2 <- heatstress.response6.2[order(heatstress.response6.2$p_val_adj), ]

heatstress.response6.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T7-T"))
heatstress.response6.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response6.3 <- heatstress.response6.3[heatstress.response6.3$p_val_adj <= 0.05, ]
heatstress.response6.3 <- heatstress.response6.3[order(heatstress.response6.3$p_val_adj), ]

heatstress.response6.3$grouping <- "Full Bleaching Marker"

heatstress.response6 <- rbind(heatstress.response6.1, heatstress.response6.2, heatstress.response6.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis_Ofav-T0-T", "Gastrodermis_Ofav-T3-T"))
heatstress.response7.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response7.1 <- heatstress.response7.1[heatstress.response7.1$p_val_adj <= 0.05, ]
heatstress.response7.1 <- heatstress.response7.1[order(heatstress.response7.1$p_val_adj), ]

heatstress.response7.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis_Ofav-T0-T", "Gastrodermis_Ofav-T5-T"))
heatstress.response7.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response7.2 <- heatstress.response7.2[heatstress.response7.2$p_val_adj <= 0.05, ]
heatstress.response7.2 <- heatstress.response7.2[order(heatstress.response7.2$p_val_adj), ]

heatstress.response7.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gastrodermis_Ofav-T0-T", "Gastrodermis_Ofav-T7-T"))
heatstress.response7.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response7.3 <- heatstress.response7.3[heatstress.response7.3$p_val_adj <= 0.05, ]
heatstress.response7.3 <- heatstress.response7.3[order(heatstress.response7.3$p_val_adj), ]

heatstress.response7.3$grouping <- "Full Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland_Ofav-T0-T", "Gland_Ofav-T3-T"))
heatstress.response10.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response10.1 <- heatstress.response10.1[heatstress.response10.1$p_val_adj <= 0.05, ]
heatstress.response10.1 <- heatstress.response10.1[order(heatstress.response10.1$p_val_adj), ]

heatstress.response10.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland_Ofav-T0-T", "Gland_Ofav-T5-T"))
heatstress.response10.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response10.2 <- heatstress.response10.2[heatstress.response10.2$p_val_adj <= 0.05, ]
heatstress.response10.2 <- heatstress.response10.2[order(heatstress.response10.2$p_val_adj), ]

heatstress.response10.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Gland_Ofav-T0-T", "Gland_Ofav-T7-T"))
heatstress.response10.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response10.3 <- heatstress.response10.3[heatstress.response10.3$p_val_adj <= 0.05, ]
heatstress.response10.3 <- heatstress.response10.3[order(heatstress.response10.3$p_val_adj), ]

heatstress.response10.3$grouping <- "Full Bleaching Marker"

heatstress.response10 <- rbind(heatstress.response10.1, heatstress.response10.2, heatstress.response10.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Epidermis_Ofav-T0-T", "Epidermis_Ofav-T3-T"))
heatstress.response11.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response11.1 <- heatstress.response11.1[heatstress.response11.1$p_val_adj <= 0.05, ]
heatstress.response11.1 <- heatstress.response11.1[order(heatstress.response11.1$p_val_adj), ]

heatstress.response11.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Epidermis_Ofav-T0-T", "Epidermis_Ofav-T5-T"))
heatstress.response11.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response11.2 <- heatstress.response11.2[heatstress.response11.2$p_val_adj <= 0.05, ]
heatstress.response11.2 <- heatstress.response11.2[order(heatstress.response11.2$p_val_adj), ]

heatstress.response11.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Epidermis_Ofav-T0-T", "Epidermis_Ofav-T7-T"))
heatstress.response11.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response11.3 <- heatstress.response11.3[heatstress.response11.3$p_val_adj <= 0.05, ]
heatstress.response11.3 <- heatstress.response11.3[order(heatstress.response11.3$p_val_adj), ]

heatstress.response11.3$grouping <- "Full Bleaching Marker"

heatstress.response11 <- rbind(heatstress.response11.1, heatstress.response11.2, heatstress.response11.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T3-T"))
heatstress.response12.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response12.1 <- heatstress.response12.1[heatstress.response12.1$p_val_adj <= 0.05, ]
heatstress.response12.1 <- heatstress.response12.1[order(heatstress.response12.1$p_val_adj), ]

heatstress.response12.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T5-T"))
heatstress.response12.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response12.2 <- heatstress.response12.2[heatstress.response12.2$p_val_adj <= 0.05, ]
heatstress.response12.2 <- heatstress.response12.2[order(heatstress.response12.2$p_val_adj), ]

heatstress.response12.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Immune_Ofav-T0-T", "Immune_Ofav-T7-T"))
heatstress.response12.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response12.3 <- heatstress.response12.3[heatstress.response12.3$p_val_adj <= 0.05, ]
heatstress.response12.3 <- heatstress.response12.3[order(heatstress.response12.3$p_val_adj), ]

heatstress.response12.3$grouping <- "Full Bleaching Marker"

heatstress.response12 <- rbind(heatstress.response12.1, heatstress.response12.2, heatstress.response12.3)
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons_Ofav-T0-T", "Neurons_Ofav-T3-T"))
heatstress.response13.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response13.1 <- heatstress.response13.1[heatstress.response13.1$p_val_adj <= 0.05, ]
heatstress.response13.1 <- heatstress.response13.1[order(heatstress.response13.1$p_val_adj), ]

heatstress.response13.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons_Ofav-T0-T", "Neurons_Ofav-T5-T"))
heatstress.response13.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response13.2 <- heatstress.response13.2[heatstress.response13.2$p_val_adj <= 0.05, ]
heatstress.response13.2 <- heatstress.response13.2[order(heatstress.response13.2$p_val_adj), ]

heatstress.response13.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Neurons_Ofav-T0-T", "Neurons_Ofav-T7-T"))
heatstress.response13.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response13.3 <- heatstress.response13.3[heatstress.response13.3$p_val_adj <= 0.05, ]
heatstress.response13.3 <- heatstress.response13.3[order(heatstress.response13.3$p_val_adj), ]

heatstress.response13.3$grouping <- "Full Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T3-T"))
heatstress.response16.1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response16.1 <- heatstress.response16.1[heatstress.response16.1$p_val_adj <= 0.05, ]
heatstress.response16.1 <- heatstress.response16.1[order(heatstress.response16.1$p_val_adj), ]

heatstress.response16.1$grouping <- "Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T5-T"))
heatstress.response16.2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response16.2 <- heatstress.response16.2[heatstress.response16.2$p_val_adj <= 0.05, ]
heatstress.response16.2 <- heatstress.response16.2[order(heatstress.response16.2$p_val_adj), ]

heatstress.response16.2$grouping <- "Secondary Early Bleaching Marker"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("Precursors_Ofav-T0-T", "Precursors_Ofav-T7-T"))
heatstress.response16.3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .5)
heatstress.response16.3 <- heatstress.response16.3[heatstress.response16.3$p_val_adj <= 0.05, ]
heatstress.response16.3 <- heatstress.response16.3[order(heatstress.response16.3$p_val_adj), ]

heatstress.response16.3$grouping <- "Full Bleaching Marker"

heatstress.response16 <- rbind(heatstress.response16.1, heatstress.response16.2, heatstress.response16.3)

heatstress.response <- rbind(heatstress.response2, heatstress.response4, heatstress.response6, heatstress.response10, heatstress.response11, heatstress.response12, heatstress.response16)


heatstress.response_early <- heatstress.response[heatstress.response$grouping== "Early Bleaching Marker", ]
write.csv(heatstress.response_early, "figures/Bleaching_DEGS/DE_SuperCluster_CoralALL_earlymarkers.csv")

heatstress.response_late <- heatstress.response[heatstress.response$grouping== "Secondary Early Bleaching Marker", ]
write.csv(heatstress.response_late, "figures/Bleaching_DEGS/DE_SuperCluster_CoralALL_secondarymarkers.csv")

heatstress.response_full <- heatstress.response[heatstress.response$grouping== "Full Bleaching Marker", ]
write.csv(heatstress.response_full, "figures/Bleaching_DEGS/DE_SuperCluster_CoralALL_fullbleachingmarkers.csv")
```









#### Dotplot with cell cluster markers
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)

markers.to.plot <- c("QW917-002686", "QW917-010242", "QW917-013627", "QW917-013752", "QW917-030101", "QW917-001525", "QW917-016491", "QW917-002944", "QW917-024580", "QW917-016495", "QW917-018090", "QW917-008368", "QW917-011241", "QW917-014250", "QW917-034318", "QW917-018353", "QW917-016496", "QW917-016501", "QW917-000806", "QW917-030981", "QW917-019333", "QW917-009174", "QW917-008780", "QW917-008044", "QW917-005964", "QW917-002611", "QW917-003505", "QW917-019813", "QW917-003256", "QW917-005475")
dp<- DotPlot(ofav.combined.labeled, features = markers.to.plot, cols = colors_to_use, dot.scale = 8, split.by = "orig.ident") +
    RotatedAxis() + ggtitle("Cell Markers")
dp

pdf("figures/Bleaching_DEGS/dotplot_cell-specific_markers.pdf", width=25, height=20)
plot(dp)
dev.off()

library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-002686", "QW917-010242", "QW917-013627", "QW917-013752", "QW917-030101", "QW917-001525", "QW917-016491", "QW917-002944", "QW917-024580", "QW917-016495", "QW917-018090", "QW917-008368", "QW917-011241", "QW917-014250", "QW917-034318", "QW917-018353", "QW917-016496", "QW917-016501", "QW917-000806", "QW917-030981", "QW917-019333", "QW917-009174", "QW917-008780", "QW917-008044", "QW917-005964", "QW917-002611", "QW917-003505", "QW917-019813", "QW917-003256", "QW917-005475")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Positive Bleaching Markers", subtitle = "Specific to certain clusters")

pdf("figures/Bleaching_DEGS/violin_cell-specific_markers.pdf.pdf", width=15, height=25)
plot(a)
dev.off()
```



#### Quick Hetamap with these
```{r}
genes_to_use <- c("QW917-002686",
"QW917-024580",
"QW917-002944",
"QW917-016491",
"QW917-001525",
"QW917-030101",
"QW917-013752",
"QW917-013627",
"QW917-010242",
"QW917-008368",
"QW917-018090",
"QW917-016495",
"QW917-034318",
"QW917-014250",
"QW917-011241",
"QW917-016501",
"QW917-016496",
"QW917-018353",
"QW917-009174",
"QW917-019333",
"QW917-030981",
"QW917-000806",
"QW917-008044",
"QW917-008780",
"QW917-002611",
"QW917-005964",
"QW917-003505",
"QW917-019813",
"QW917-003256",
"QW917-005475",
"QW917-036332",
"QW917-036331",
"QW917-034316",
"QW917-009678",
"QW917-030986",
"QW917-016488",
"QW917-010131",
"QW917-006427",
"QW917-024964",
"QW917-030646",
"QW917-003971",
"QW917-023502",
"QW917-013386",
"QW917-017006",
"QW917-003567",
"QW917-027832",
"QW917-023065",
"QW917-017942",
"QW917-026114",
"QW917-008288",
"QW917-009852",
"QW917-010145",
"QW917-010146",
"QW917-008341",
"QW917-023565",
"QW917-009324",
"QW917-001863",
"QW917-018305",
"QW917-032268",
"QW917-000347",
"QW917-015206",
"QW917-009778",
"QW917-032489",
"QW917-029254",
"QW917-005586",
"QW917-030285",
"QW917-006360",
"QW917-033575",
"QW917-020799",
"QW917-013974",
"QW917-009336",
"QW917-024190",
"QW917-024765",
"QW917-028699",
"QW917-024271",
"QW917-013452",
"QW917-017802",
"QW917-005571",
"QW917-036231",
"QW917-036230",
"QW917-024491",
"QW917-015093",
"QW917-021056",
"QW917-036205",
"QW917-036204",
"QW917-010171",
"QW917-029111",
"QW917-002688",
"QW917-028062",
"QW917-008226",
"QW917-026948")


levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")

ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")

ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled, idents = c("Expelled Breviolum"), invert = TRUE)
ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled.hm1, idents = c("In-hospite Breviolum"), invert = TRUE)
ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled.hm1, idents = c("In-hospite Durusdinium"), invert = TRUE)

unique(ofav.combined.labeled.hm1$seurat_clusters.Timepoints)
ofav.combined.labeled.hm <- ofav.combined.labeled.hm1
rm(ofav.combined.labeled.hm1)
Idents(ofav.combined.labeled.hm) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.hm)
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)
ofav.combined.labeled.hm<- ScaleData(ofav.combined.labeled.hm, features = all.genes)

my_levels <- c("Calicoblastic Cells_Ofav-T0-T", "Cnidocytes_Ofav-T0-T", "Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 2_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 2_Ofav-T0-T", "Gastrodermis 3_Ofav-T0-T", "Gland 1_Ofav-T0-T", "Gland 2_Ofav-T0-T", "Immune_Ofav-T0-T", "Neurons 1_Ofav-T0-T", "Neurons 2_Ofav-T0-T", "Neurons 3_Ofav-T0-T", "Precursors_Ofav-T0-T", "Unidentified_Ofav-T0-T", "Calicoblastic Cells_Ofav-T3-T", "Cnidocytes_Ofav-T3-T", "Digestive Filaments 1_Ofav-T3-T", "Digestive Filaments 2_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 2_Ofav-T3-T", "Gastrodermis 3_Ofav-T3-T", "Gland 1_Ofav-T3-T", "Gland 2_Ofav-T3-T", "Immune_Ofav-T3-T", "Neurons 1_Ofav-T3-T", "Neurons 2_Ofav-T3-T", "Neurons 3_Ofav-T3-T", "Precursors_Ofav-T3-T", "Unidentified_Ofav-T3-T", "Calicoblastic Cells_Ofav-T5-T", "Cnidocytes_Ofav-T5-T", "Digestive Filaments 1_Ofav-T5-T", "Digestive Filaments 2_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 2_Ofav-T5-T", "Gastrodermis 3_Ofav-T5-T", "Gland 1_Ofav-T5-T", "Gland 2_Ofav-T5-T", "Immune_Ofav-T5-T", "Neurons 1_Ofav-T5-T", "Neurons 2_Ofav-T5-T", "Neurons 3_Ofav-T5-T", "Precursors_Ofav-T5-T", "Unidentified_Ofav-T5-T", "Calicoblastic Cells_Ofav-T7-T", "Cnidocytes_Ofav-T7-T", "Digestive Filaments 1_Ofav-T7-T", "Digestive Filaments 2_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T7-T", "Gastrodermis 3_Ofav-T7-T", "Gland 1_Ofav-T7-T", "Gland 2_Ofav-T7-T", "Immune_Ofav-T7-T", "Neurons 1_Ofav-T7-T", "Neurons 2_Ofav-T7-T", "Neurons 3_Ofav-T7-T", "Precursors_Ofav-T7-T", "Unidentified_Ofav-T7-T")

colors_hm <- c("darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)

ofav.combined.labeled.hm@meta.data[,"seurat_clusters.Timepoints",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters.Timepoints) %>% slice_sample(n= 20) -> index

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + NoLegend() + scale_fill_gradientn(colors = c("blue", "white", "red"))

pdf("figures/Bleaching_DEGS/DEGS_Coral-clusterspecific-heatmap.pdf", height = 30, width=25)
plot(hm)
dev.off()
```

## Dotplot of Brev vs Durs 
```{r}
library(scCustomize)
library(wesanderson)
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("QW917-002688", "QW917-002686", "QW917-002685", "QW917-030646", "QW917-018353", "QW917-009336", "QW917-028319", "QW917-005606", "QW917-001525", "QW917-023319", "QW917-003256", "QW917-014705", "QW917-005571", "QW917-008443", "QW917-030668", "QW917-018851", "QW917-022673", "QW917-014250", "QW917-007699", "QW917-033195", "QW917-008828", "QW917-033575", "QW917-028232", "QW917-027187", "QW917-020637", "QW917-025759", "QW917-031802", "QW917-023845", "QW917-003054", "QW917-029254", "QW917-011254", "QW917-024271", "QW917-013452", "QW917-032489", "QW917-029992", "QW917-013974", "QW917-013627", "QW917-023565", "QW917-026114", "QW917-030101", "QW917-010381", "QW917-004226", "QW917-030504", "QW917-018168", "QW917-000161", "QW917-034316", "QW917-024967", "QW917-021056", "QW917-009835", "QW917-015093", "QW917-018150", "QW917-027476", "QW917-026128", "QW917-002570", "QW917-005894", "QW917-005425", "QW917-005969", "QW917-029232", "QW917-018090", "QW917-022086", "QW917-035095", "QW917-027832", "QW917-035954", "QW917-002392", "QW917-017896", "QW917-005962", "QW917-013838", "QW917-016501", "QW917-031651", "QW917-014249", "QW917-016496", "QW917-005300", "QW917-016495", "QW917-004899", "QW917-002144", "QW917-020989", "QW917-024396", "QW917-000010")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Cell Markers") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

write.csv(dp$data, "~/Desktop/colors.csv")

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_brevVdur_markers2.pdf", width=20, height=5)
plot(dp)
dev.off()

```



```{r}
genes_to_use <- c("QW917-003080",
"QW917-004830",
"QW917-024689",
"QW917-005606",
"QW917-024580",
"QW917-014249",
"QW917-005962",
"QW917-019333",
"QW917-027013",
"QW917-000806",
"QW917-0056061",
"QW917-0030801",
"QW917-012685",
"QW917-0030802",
"QW917-0056062",
"QW917-008044",
"QW917-0059621",
"QW917-0246891",
"QW917-005425",
"QW917-0270132",
"QW917-0142491",
"QW917-028245",
"QW917-032073",
"QW917-0030803",
"QW917-0059622",
"QW917-0246892",
"QW917-0282451",
"QW917-0048301",
"QW917-0056063",
"QW917-0320731",
"QW917-0142492",
"QW917-016495",
"QW917-0270131",
"QW917-0030804",
"QW917-0246893",
"QW917-0059623",
"QW917-0320732",
"QW917-0048302",
"QW917-0270133",
"QW917-002611",
"QW917-0282452",
"QW917-0245801",
"QW917-0142493",
"QW917-0030805",
"QW917-0059624",
"QW917-0246894",
"QW917-0056064",
"QW917-0282453",
"QW917-0270134",
"QW917-0320733",
"QW917-000091",
"QW917-0048303",
"QW917-005964",
"QW917-0030806",
"QW917-0059625",
"QW917-0246895",
"QW917-0320734",
"QW917-010131",
"QW917-0282454",
"QW917-0059641",
"QW917-0245802",
"QW917-005969",
"QW917-0056065",
"QW917-0030807",
"QW917-0246896",
"QW917-0059626",
"QW917-0270135",
"QW917-0048304",
"QW917-0056066",
"QW917-004807",
"symbB.v1.2.040993",
"QW917-0142494",
"QW917-0320735",
"QW917-0030808",
"QW917-0246897",
"QW917-0059627",
"QW917-0056067",
"QW917-016859",
"QW917-0320736",
"QW917-0142495",
"QW917-0270136",
"QW917-0026111",
"QW917-0282455",
"QW917-0030809",
"QW917-0246898",
"QW917-0059628",
"QW917-0048305",
"QW917-0270137",
"QW917-0056068",
"QW917-0282456",
"QW917-000356",
"QW917-0245803",
"symbB.v1.2.037890",
"QW917-00308010",
"QW917-0246899",
"QW917-0059629",
"QW917-0270138",
"QW917-0320737",
"QW917-0056069",
"QW917-0059691",
"QW917-0000911",
"QW917-0048306",
"QW917-0142496",
"QW917-00308011",
"QW917-00596210",
"QW917-02468910",
"QW917-0270139",
"QW917-0282457",
"QW917-00560610",
"QW917-0245804",
"QW917-0059642",
"QW917-0048307",
"QW917-0320738",
"QW917-00308012",
"QW917-02468911",
"QW917-0054251",
"QW917-00560611",
"QW917-0320739",
"QW917-02701310",
"QW917-00596211",
"QW917-0282458",
"QW917-005426",
"QW917-0142497",
"QW917-00308013",
"QW917-02468912",
"QW917-00596212",
"QW917-03207310",
"QW917-00560612",
"QW917-02701311",
"QW917-0048071",
"QW917-0048308",
"QW917-0245805",
"QW917-0059643",
"QW917-00308014",
"QW917-00596213",
"QW917-02468913",
"QW917-02701312",
"QW917-0048309",
"QW917-03207311",
"QW917-00560613",
"QW917-0168591",
"QW917-0026112",
"QW917-0282459")


levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")

ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")

ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled, idents = c("Expelled Breviolum"), invert = TRUE)
ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled.hm1, idents = c("In-hospite Breviolum"), invert = TRUE)
ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled.hm1, idents = c("In-hospite Durusdinium"), invert = TRUE)

unique(ofav.combined.labeled.hm1$seurat_clusters.Timepoints)
ofav.combined.labeled.hm <- ofav.combined.labeled.hm1
rm(ofav.combined.labeled.hm1)
Idents(ofav.combined.labeled.hm) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.hm)
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)
ofav.combined.labeled.hm<- ScaleData(ofav.combined.labeled.hm, features = all.genes)

my_levels <- c("Calicoblastic Cells_Ofav-T0-T", "Cnidocytes_Ofav-T0-T", "Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 2_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 2_Ofav-T0-T", "Gastrodermis 3_Ofav-T0-T", "Gland 1_Ofav-T0-T", "Gland 2_Ofav-T0-T", "Immune_Ofav-T0-T", "Neurons 1_Ofav-T0-T", "Neurons 2_Ofav-T0-T", "Neurons 3_Ofav-T0-T", "Precursors_Ofav-T0-T", "Unidentified_Ofav-T0-T", "Calicoblastic Cells_Ofav-T3-T", "Cnidocytes_Ofav-T3-T", "Digestive Filaments 1_Ofav-T3-T", "Digestive Filaments 2_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 2_Ofav-T3-T", "Gastrodermis 3_Ofav-T3-T", "Gland 1_Ofav-T3-T", "Gland 2_Ofav-T3-T", "Immune_Ofav-T3-T", "Neurons 1_Ofav-T3-T", "Neurons 2_Ofav-T3-T", "Neurons 3_Ofav-T3-T", "Precursors_Ofav-T3-T", "Unidentified_Ofav-T3-T", "Calicoblastic Cells_Ofav-T5-T", "Cnidocytes_Ofav-T5-T", "Digestive Filaments 1_Ofav-T5-T", "Digestive Filaments 2_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 2_Ofav-T5-T", "Gastrodermis 3_Ofav-T5-T", "Gland 1_Ofav-T5-T", "Gland 2_Ofav-T5-T", "Immune_Ofav-T5-T", "Neurons 1_Ofav-T5-T", "Neurons 2_Ofav-T5-T", "Neurons 3_Ofav-T5-T", "Precursors_Ofav-T5-T", "Unidentified_Ofav-T5-T", "Calicoblastic Cells_Ofav-T7-T", "Cnidocytes_Ofav-T7-T", "Digestive Filaments 1_Ofav-T7-T", "Digestive Filaments 2_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T7-T", "Gastrodermis 3_Ofav-T7-T", "Gland 1_Ofav-T7-T", "Gland 2_Ofav-T7-T", "Immune_Ofav-T7-T", "Neurons 1_Ofav-T7-T", "Neurons 2_Ofav-T7-T", "Neurons 3_Ofav-T7-T", "Precursors_Ofav-T7-T", "Unidentified_Ofav-T7-T")

colors_hm <- c("darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)

ofav.combined.labeled.hm@meta.data[,"seurat_clusters.Timepoints",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters.Timepoints) %>% slice_sample(n= 20) -> index

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + NoLegend() + scale_fill_gradientn(colors = c("blue", "white", "red"))

pdf("figures/Bleaching_DEGS/DEGS_CoralAll-heatmap.pdf", height = 30, width=25)
plot(hm)
dev.off()
```



### DE Analysis (Super Cluster-level)
```{r}
ofav.combined.labeled.DE <- subset(x = ofav.combined.labeled, idents = c("Unidentified"), invert = TRUE)
levels(Idents(ofav.combined.labeled.DE))

ofav.combined.labeled.SC <- RenameIdents(ofav.combined.labeled.DE, "Digestive Filaments 1"= "Digestive Filaments", "In-hospite Breviolum"= "Breviolum", "Expelled Breviolum"= "Breviolum", "Calicoblastic Cells"= "Calicoblastic Cells", "Gastrodermis 1"= "Gastrodermis", "Gastrodermis 2"= "Gastrodermis", "Gastrodermis 3"= "Gastrodermis", "Neurons 1"= "Neurons", "Immune"= "Immune", "Neurons 2"= "Neurons", "Gland 1"= "Gland", "Neurons 3"= "Neurons", "Gastrodermis (Muscle-like)"= "Gastrodermis (Muscle-like)", "In-hospite Durusdinium"= "Durusdinium", "Digestive Filaments 2"= "Digestive Filaments", "Precursors"="Precursors", "Gland 2"= "Gland", "Cnidocytes"="Cnidocytes")

ofav.combined.labeled.SC$super_clusters <- paste(Idents(ofav.combined.labeled.SC), ofav.combined.labeled.SC$orig.ident, sep = "_")
unique(ofav.combined.labeled.SC$super_clusters)
Idents(ofav.combined.labeled.SC) <- "super_clusters"
levels(Idents(ofav.combined.labeled.SC))
```
#### Digestive Filaments
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Digestive Filaments_Ofav-T0-T", "Digestive Filaments_Ofav-T3-T", "Digestive Filaments_Ofav-T5-T", "Digestive Filaments_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Digestive Filaments_Ofav-T0-T", "Digestive Filaments_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Digestive Filaments_Ofav-T0-T", "Digestive Filaments_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Digestive Filaments_Ofav-T0-T", "Digestive Filaments_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Digestive Filaments_Ofav-T3-T", "Digestive Filaments_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Digestive Filaments_Ofav-T5-T", "Digestive Filaments_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Digestive_Filaments_combined.csv")
```
#### Gastrodermis
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gastrodermis_Ofav-T0-T", "Gastrodermis_Ofav-T3-T", "Gastrodermis_Ofav-T5-T", "Gastrodermis_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gastrodermis_Ofav-T0-T", "Gastrodermis_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gastrodermis_Ofav-T0-T", "Gastrodermis_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gastrodermis_Ofav-T0-T", "Gastrodermis_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gastrodermis_Ofav-T3-T", "Gastrodermis_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gastrodermis_Ofav-T5-T", "Gastrodermis_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Gastrodermis_combined.csv")
```

#### Gland
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gland_Ofav-T0-T", "Gland_Ofav-T3-T", "Gland_Ofav-T5-T", "Gland_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gland_Ofav-T0-T", "Gland_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gland_Ofav-T0-T", "Gland_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gland_Ofav-T0-T", "Gland_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gland_Ofav-T3-T", "Gland_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Gland_Ofav-T5-T", "Gland_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Gland_combined.csv")
```

#### Neurons
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Neurons_Ofav-T0-T", "Neurons_Ofav-T3-T", "Neurons_Ofav-T5-T", "Neurons_Ofav-T7-T"))
heatstress.response1 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=.5, min.pct= .25)
heatstress.response1 <- heatstress.response1[heatstress.response1$p_val_adj <= 0.05, ]
heatstress.response1$grouping <- "All Markers"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Neurons_Ofav-T0-T", "Neurons_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
heatstress.response2$grouping <- "Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Neurons_Ofav-T0-T", "Neurons_Ofav-T5-T"))
heatstress.response3 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response3 <- heatstress.response3[heatstress.response3$p_val_adj <= 0.05, ]
heatstress.response3$grouping <- "Secondary Early Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Neurons_Ofav-T0-T", "Neurons_Ofav-T7-T"))
heatstress.response4 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response4 <- heatstress.response4[heatstress.response4$p_val_adj <= 0.05, ]
heatstress.response4$grouping <- "Bleaching"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Neurons_Ofav-T3-T", "Neurons_Ofav-T5-T"))
heatstress.response5 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response5 <- heatstress.response5[heatstress.response5$p_val_adj <= 0.05, ]
heatstress.response5$grouping <- "Mid-bleach Transition"

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Neurons_Ofav-T5-T", "Neurons_Ofav-T7-T"))
heatstress.response6 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, logfc.threshold=0.5, min.pct= .25)
heatstress.response6 <- heatstress.response6[heatstress.response6$p_val_adj <= 0.05, ]
heatstress.response6$grouping <- "Late stage bleaching"

heatstress.response <- rbind(heatstress.response1, heatstress.response2, heatstress.response3, heatstress.response4, heatstress.response5, heatstress.response6)
heatstress.response_nodups <- heatstress.response[!duplicated(heatstress.response[c("gene")]), ]

write.csv(heatstress.response_nodups, "figures/Bleaching_DEGS/DE_Neurons_combined.csv")
```




### Stacked Violins of DEGs
#### TO vs T3 (Early Bleaching Markers)
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-009678", "QW917-016496", "symbB.v1.2.033380", "QW917-030101", "QW917-018353", "QW917-018168", "QW917-013627", "QW917-013752", "QW917-029781", "QW917-010242", "QW917-006349", "QW917-018150", "QW917-018353", "QW917-030101", "QW917-018168", "QW917-029781", "QW917-013752", "QW917-010242", "QW917-003326", "QW917-014250", "QW917-020637", "QW917-028319", "QW917-018090", "QW917-018353", "QW917-025879", "QW917-018090", "QW917-016495", "symbB.v1.2.033380", "QW917-016496", "QW917-018353", "QW917-028232", "QW917-014250", "QW917-018168", "QW917-016496", "QW917-018353", "QW917-016496", "QW917-018353", "QW917-005894", "QW917-016501", "QW917-005894", "QW917-016496", "QW917-018353", "QW917-016501", "QW917-016878", "QW917-005894", "QW917-018353", "QW917-002688")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Early Bleaching Markers", subtitle = "Significant DEG between Healthy and Mid-bleach 1")

pdf("figures/Bleaching_DEGS/early_markers.pdf", width=15, height=25)
plot(a)
dev.off()
```

#### TO vs T3 (Early Bleaching Markers) LFC > 3
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-005606", "QW917-016878", "QW917-016501", "QW917-027542", "QW917-017111", "QW917-025432", "QW917-008368", "QW917-024580", "symbB.v1.2.040993", "QW917-016496")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Early Bleaching Markers", subtitle = "Significant DEG between Healthy and Mid-bleach 1 with LFC>3")

pdf("figures/Bleaching_DEGS/early_markers2.pdf", width=10, height=10)
plot(a)
dev.off()
```


##### T0 vs T5: Secondary Early Bleaching Markers per cell type
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-028062", "QW917-010184", "QW917-023502", "QW917-022832", "QW917-011241", "QW917-014250", "QW917-034318", "QW917-028062", "QW917-023502", "QW917-011241", "QW917-013627", "QW917-034318", "QW917-028062", "QW917-026774", "QW917-027832", "QW917-006349", "QW917-027832", "QW917-025879", "QW917-026114", "QW917-018150", "QW917-018090", "QW917-022832", "QW917-018150", "QW917-018090", "QW917-018150")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Secondary Early Bleaching Markers", subtitle = "Significant DEG between Healthy and Mid-bleach 2")

pdf("figures/Bleaching_DEGS/secondary_markers.pdf", width=15, height=25)
plot(a)
dev.off()
```

##### T0 vs T5: Secondary Early Bleaching Markers per cell type LFC > 3
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-016878", "QW917-005606", "QW917-016501", "QW917-0056061", "QW917-004807", "QW917-008447", "QW917-016495", "symbB.v1.2.040993", "QW917-010131", "QW917-016488", "QW917-031733", "QW917-024580", "QW917-027013",  "QW917-026774", "QW917-017942", "QW917-006349")
DefaultAssay(ofav.combined.labeled) <- "RNA"
a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Secondary Early Bleaching Markers", subtitle = "Significant DEG between Healthy and Mid-bleach 2 with LFC>3")

pdf("figures/Bleaching_DEGS/secondary_markers2.pdf", width=15, height=25)
plot(a)
dev.off()
``` 

##### T0 vs T7: Bleaching Markers per cell type
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-009324", "QW917-013627", "QW917-029254", "QW917-032489", "QW917-015206", "QW917-033575", "QW917-009336", "QW917-013974", "QW917-020799", "QW917-030668", "QW917-005586", "QW917-024271", "QW917-013452", "QW917-028699", "QW917-024765", "QW917-024190", "QW917-001863", "QW917-017799", "QW917-010242", "QW917-030285", "QW917-018353", "QW917-018090", "QW917-007699", "QW917-024321", "QW917-027476", "QW917-000629", "QW917-024318", "QW917-005475", "QW917-027879", "QW917-022587", "QW917-036231", "QW917-018169", "QW917-026657", "QW917-002685", "QW917-006427", "QW917-015093", "QW917-006430", "QW917-023414", "QW917-021056", "QW917-010727", "QW917-029766", "QW917-016496", "QW917-036205", "QW917-036204", "QW917-013386", "QW917-010171", "QW917-031915", "QW917-025879", "QW917-002686", "QW917-002688", "QW917-025087", "QW917-022832", "QW917-023825", "QW917-017006", "QW917-016501", "QW917-023565", "QW917-028062", "QW917-023536", "QW917-029781", "QW917-017008", "QW917-018305", "QW917-010184", "QW917-032411", "QW917-033692", "QW917-019830", "QW917-009835", "QW917-030646", "QW917-006349", "QW917-002392", "QW917-012971", "QW917-003256", "QW917-002944", "QW917-026948", "QW917-014316", "QW917-023314")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Bleaching Markers", subtitle = "Significant DEG between Healthy and Fully Bleached")

pdf("figures/Bleaching_DEGS/fullbleach_markers.pdf", width=15, height=50)
plot(a)
dev.off()
```

##### T0 vs T7: Bleaching Markers per cell type w LFC >3
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-005606", "QW917-003971", "QW917-024580", "QW917-010145", "QW917-035634", "QW917-004231", "QW917-009174", "QW917-024689", "QW917-003080", "QW917-027013", "QW917-008044", "QW917-004807", "QW917-005962", "QW917-016495", "QW917-028245", "QW917-014249", "QW917-004830", "QW917-017006", "QW917-008447", "QW917-024491", "QW917-030646", "QW917-006349", "QW917-010131", "QW917-032073", "symbB.v1.2.040993", "QW917-016859", "QW917-013386")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Bleaching Markers", subtitle = "Significant DEG between Healthy and Fully Bleached LFC >3")

pdf("figures/Bleaching_DEGS/fullbleach_markers2.pdf", width=15, height=20)
plot(a)
dev.off()
```

##### Violin Summary Fig
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c( "Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-005606", "QW917-016878", "QW917-016501", "QW917-008368", "QW917-024580", "symbB.v1.2.040993", "QW917-016496", "QW917-004807", "QW917-008447", "QW917-016495", "QW917-017942", "QW917-006349", "QW917-010145", "QW917-024689", "QW917-003080", "QW917-027013", "QW917-005962", "QW917-030646")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Bleaching Markers", subtitle = "LFC > 3")

pdf("figures/Bleaching_DEGS/Violin_summary2.pdf", width=15, height=20)
plot(a)
dev.off()
```




##### T3 vs T5: Transitions in Bleaching Response as intensity rises
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-000592")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = F, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Mid-bleach Transition Markers", subtitle = "Significant DEG between both mid-bleached samples")
a

pdf("figures/Bleaching_DEGS/bleachingtransition_markers.pdf", width=15, height=10)
plot(a)
dev.off()
```
##### T5 vs T7: Late Stage Bleaching/death markers
```{r}
library(scCustomize)
library(wesanderson)
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)

my_levels <- c("Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Unidentified", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-009336", "QW917-002685", "QW917-016878", "QW917-035714", "QW917-009632", "QW917-010171", "QW917-015093", "QW917-026114", "QW917-002686")

a <- VlnPlot_scCustom(ofav.combined.labeled, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Late-stage Bleaching Markers", subtitle = "Significant DEG between last mid-bleached and fully bleached")
a
pdf("figures/Bleaching_DEGS/latebleaching_markers.pdf", width=15, height=15)
plot(a)
dev.off()
```

# Symbiont Analysis
## Peripheral Analysis
```{r}
features <- c("symbB.v1.2.037890", "symbB.v1.2.040993", "QW917-005962", "QW917-005894")
a <- FeaturePlot(ofav.combined.labeled.symb, features = features)
pdf("figures/peripheral_symbionts.pdf", width=10, height=7)
plot(a)
dev.off()
```


## Violin DEGs
```{r}
library(scCustomize)
library(wesanderson)

my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)
ofav.combined.labeled.symb <- subset(x = ofav.combined.labeled, idents = c("Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium"), invert = F)
my_levels <- c("Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled.symb@active.ident <- factor(x = ofav.combined.labeled.symb@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")

features <- c("QW917-007699", "QW917-010381", "symbB.v1.2.040993", "QW917-006349", "QW917-001525", "QW917-031802", "QW917-030101", "QW917-015206", "QW917-008443", "QW917-026114", "QW917-018150", "QW917-018353", "QW917-027187", "QW917-022673", "symbB.v1.2.041403", "QW917-002686", "QW917-030668", "symbB.v1.2.001035", "QW917-029992", "QW917-005894", "QW917-017490", "QW917-028697", "QW917-004770", "QW917-018305", "QW917-028792", "QW917-008828", "QW917-025087", "QW917-000347", "QW917-022587", "QW917-023502", "QW917-002688", "QW917-013501", "QW917-001647", "QW917-016794", "QW917-024967", "QW917-027982", "QW917-018355", "QW917-023825", "QW917-028683", "QW917-024689", "QW917-027780", "QW917-016495", "QW917-035955", "QW917-026128", "QW917-004807", "QW917-003080", "QW917-009835", "QW917-011523", "QW917-021056", "QW917-009852", "QW917-002144", "QW917-017896", "QW917-003907", "QW917-015220", "QW917-013974", "QW917-018090", "QW917-027476", "symbB.v1.2.025478", "symbB.v1.2.000589", "symbB.v1.2.014626", "QW917-003256", "QW917-002685", "symbB.v1.2.039120", "symbB.v1.2.033305", "QW917-022832", "QW917-009336", "QW917-028062", "QW917-010184", "QW917-027832", "QW917-009678", "QW917-002392", "g20949", "QW917-010073", "QW917-016878", "QW917-016501", "QW917-024580", "QW917-014249", "g9043", "QW917-025829", "QW917-016496", "QW917-030749", "QW917-018169")

a <- VlnPlot_scCustom(ofav.combined.labeled.symb, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Bleaching Markers", subtitle = "LFC > 3")

pdf("figures/Bleaching_DEGS/Violin_symbionts.pdf", width=10, height=40)
plot(a)
dev.off()
```

## Symbiont UMAP
```{r}
#ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")
#ofav.combined.labeled$seurat_clusters <- Idents(ofav.combined.labeled)
#Idents(ofav.combined.labeled) <- "seurat_clusters.Timepoints"
Idents(ofav.combined.labeled)

ofav.combined.labeled.symb <- subset(x = ofav.combined.labeled, idents = c("Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium"))
my_levels <- c("Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")
ofav.combined.labeled.symb@active.ident <- factor(x = ofav.combined.labeled.symb@active.ident, levels = my_levels)

umap_symb_labeled <- DimPlot(ofav.combined.labeled.symb, reduction = "umap", split.by = "orig.ident", label = F, pt.size = 0.35, ncol = 2, cols=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) + labs(title = "UMAP Clustering of Symbiodiniaceae Cells")
umap_symb_labeled
pdf("figures/UMAP-Symbionts-temps.pdf", width=12, height=10)
plot(umap_symb_labeled)
dev.off()
```

## Symbiont Violin Plots
```{r}
ofav.combined.labeled.symb$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.symb), ofav.combined.labeled.symb$orig.ident, sep = "_")
ofav.combined.labeled.symb$seurat_clusters <- Idents(ofav.combined.labeled.symb)
Idents(ofav.combined.labeled.symb) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.symb)

my_levels <- c("Expelled Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T0-T", "Expelled Breviolum_Ofav-T3-T", "In-hospite Breviolum_Ofav-T3-T", "In-hospite Durusdinium_Ofav-T3-T", "Expelled Breviolum_Ofav-T5-T", "In-hospite Breviolum_Ofav-T5-T", "In-hospite Durusdinium_Ofav-T5-T", "Expelled Breviolum_Ofav-T7-T", "In-hospite Breviolum_Ofav-T7-T", "In-hospite Durusdinium_Ofav-T7-T")
ofav.combined.labeled.symb@active.ident <- factor(x = ofav.combined.labeled.symb@active.ident, levels = my_levels)
levels(ofav.combined.labeled.symb)
```

```{r}
gene1 <- "QW917-007699"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Cofilin; COG = Z:(Z) Cytoskeleton", subtitle="Genes: QW917-007699") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_Coflin.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "QW917-010381"
gene2 <- "symbB.v1.2.018628"
gene3 <- "g7946"
ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene; COG = P:(P) Inorganic ion transport and metabolism", subtitle="Genes: QW917-010381 + symbB.v1.2.018628 + g7946") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_IonTransport.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "symbB.v1.2.040993"
ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: symbB.v1.2.040993") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_uncharacterized.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-006349"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene; COG = I:(I) Lipid transport and metabolism", subtitle="Genes: QW917-006349") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_LipidTransport.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "QW917-001525"
gene2 <- "symbB.v1.2.029302"
gene3 <- "g20134"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene; COG = T:(T) Signal transduction mechanisms", subtitle="Genes: QW917-001525 + symbB.v1.2.029302 + g20134") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_Signalling.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-030101"
gene2 <- "symbB.v1.2.008653"
gene3 <- "g16404"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "F-box/WD repeat-containing protein 7", subtitle="Genes: QW917-030101 + symbB.v1.2.008653 + g16404") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_FBox.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "symbB.v1.2.025478"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: symbB.v1.2.025478") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_UC2.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "symbB.v1.2.000589"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: symbB.v1.2.000589") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_UC3.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "symbB.v1.2.014626"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: symbB.v1.2.014626") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_UC4.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "symbB.v1.2.039120"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: symbB.v1.2.039120") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_UC5.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "g20949"
gene2 <- "symbB.v1.2.027536"
gene3 <- "QW917-018256"
ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: g20949 + symbB.v1.2.027536 + QW917-018256") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_UC6.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "g9043"
ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: g20949 + symbB.v1.2.027536 + QW917-018256") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_UC6.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

#### Find out What else to plot
```{r}
library(scCustomize)
library(wesanderson)

colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("QW917-031802", "QW917-030101", "QW917-015206", "QW917-008443", "QW917-026114", "QW917-018150", "QW917-018353", "QW917-027187", "QW917-022673", "symbB.v1.2.041403", "QW917-002686", "QW917-030668", "symbB.v1.2.001035", "QW917-029992", "QW917-005894", "QW917-017490", "QW917-028697", "QW917-004770", "QW917-018305", "QW917-028792", "QW917-008828", "QW917-025087", "QW917-000347", "QW917-022587", "QW917-023502", "QW917-002688", "QW917-013501", "QW917-001647", "QW917-016794", "QW917-024967", "QW917-027982", "QW917-018355", "QW917-023825", "QW917-028683", "QW917-024689", "QW917-027780", "QW917-016495", "QW917-035955", "QW917-026128", "QW917-004807", "QW917-003080", "QW917-009835", "QW917-011523", "QW917-021056", "QW917-009852", "QW917-002144", "QW917-017896", "QW917-003907", "QW917-015220", "QW917-013974", "QW917-018090", "QW917-027476", "symbB.v1.2.025478", "symbB.v1.2.000589", "symbB.v1.2.014626", "QW917-003256", "QW917-002685", "symbB.v1.2.039120", "symbB.v1.2.033305", "QW917-022832", "QW917-009336", "QW917-028062", "QW917-010184", "QW917-027832", "QW917-009678", "QW917-002392", "g20949", "QW917-010073", "QW917-016878", "QW917-016501", "QW917-024580", "QW917-014249", "g9043", "QW917-025829", "QW917-016496", "QW917-030749", "QW917-018169")

a <- VlnPlot_scCustom(ofav.combined.labeled.symb, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Symbiont Markers to Plot", subtitle = "Significant DEG between Healthy and Mid-bleach 1")

pdf("figures/Bleaching_Symbiont_DEGs/what2plot.pdf", width=15, height=25)
plot(a)
dev.off()
```

```{r}
gene1 <- "QW917-026114"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: QW917-026114") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_UC7.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-018150"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "fts3-like protein; COG = P:(P) Inorganic ion transport and metabolism", subtitle="Genes: QW917-018150") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_IonTransport2.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-005894"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "BTB (POZ) domain-containing protein", subtitle="Genes: QW917-005894") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_EarlyBleachingMarker_BTB.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-004770"
gene2 <- "g24293"
gene3 <- "symbB.v1.2.033044"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Netrin receptor unc5c: COG = T:(T) Signal transduction mechanisms", subtitle="Genes: QW917-004770 + g24293 + symbB.v1.2.033044") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_SecondaryEarlyBleachingMarker_Netrin.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-008828"
gene2 <- "g2741"
gene3 <- "symbB.v1.2.001083"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "RAD52 motif-containing protein", subtitle="Genes: QW917-008828 + g24293 + symbB.v1.2.033044") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_SecondaryEarlyBleachingMarker_RAD52.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-002688"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]

# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Methyltransferase-like protein 13; COG = E:(E) Amino acid transport and metabolism", subtitle="Genes: QW917-002688 + g24293 + symbB.v1.2.033044") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_SecondaryEarlyBleachingMarker_METHYL.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-016495"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]

# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "ndufa8; COG = C:(C) Energy production and conversion", subtitle="Genes: QW917-016495") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_METHYL.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-004807"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]

# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Lysophosphatidylcholine acyltransferase 3; COG = I:(I) Lipid transport and metabolism", subtitle="Genes: QW917-004807") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_LPCAT3.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "QW917-013974"
gene2<- "g20480"
gene3 <- "symbB.v1.2.007591"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]

# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "TOM1-like protein 2; COG = U:(U) Intracellular trafficking, secretion, and vesicular transport", subtitle="Genes: QW917-013974") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_TOM1.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-018090"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]


# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Radial spoke head 1; GO:0043227 = membrane-bounded organelle", subtitle="Genes: QW917-018090") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_RSPH.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "QW917-003256"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]


# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Dynamin-related GTPase protein; COG = U:(U) Intracellular trafficking, secretion, and vesicular transport", subtitle="Genes: QW917-003256") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_DNM1.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-002685"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]

# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: QW917-002685") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_UC1.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-009336"
gene2 <- "symbB.v1.2.010962"
gene3 <- "g21104"

ofav.combined.labeled.tm <- ofav.combined.labeled.symb
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Gene", subtitle="Genes: QW917-009336 + symbB.v1.2.010962 + g21104") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_UC2.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "QW917-009678"
gene2 <- "symbB.v1.2.010170"
gene3 <- "g33744"

expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Signal transducing adapter molecule 2; COG = T:(T) Signal transduction mechanisms", subtitle="Genes: QW917-009678 + symbB.v1.2.010170 + g33744") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_STAM2.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-016878"

expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Myosin-Id; COG = Z:(Z) Cytoskeleton", subtitle="Genes: QW917-016878") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_MYO1D.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
here
```{r}
gene1 <- "QW917-024580"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "IPR027417 P-loop containing nucleoside triphosphate hydrolase", subtitle="Genes: QW917-024580") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_PLoop.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "QW917-025829"
gene2 <- "symbB.v1.2.019320"
gene3 <- "g3672"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Glycosyltransferase 28 C-terminal domain, variant 2; COG = C:(C) Energy production and conversion", subtitle="Genes: QW917-025829 + symbB.v1.2.019320 + g3672") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_GLYCTRANS.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
library(scCustomize)
library(wesanderson)

colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")
features <- c("symbB.v1.2.040993", "symbB.v1.2.025478", "symbB.v1.2.001983", "symbB.v1.2.000589", "symbB.v1.2.014626", "symbB.v1.2.011981", "symbB.v1.2.039120", "symbB.v1.2.031786", "symbB.v1.2.030622", "symbB.v1.2.012929", "symbB.v1.2.033305", "symbB.v1.2.015891", "symbB.v1.2.009560", "symbB.v1.2.043650", "symbB.v1.2.032431", "symbB.v1.2.037890", "symbB.v1.2.041290", "symbB.v1.2.038881", "symbB.v1.2.037117", "symbB.v1.2.030903", "symbB.v1.2.037835", "symbB.v1.2.024311", "symbB.v1.2.037202", "symbB.v1.2.009723", "symbB.v1.2.004622", "symbB.v1.2.020207", "symbB.v1.2.037115", "symbB.v1.2.020350", "symbB.v1.2.008065", "symbB.v1.2.033151", "symbB.v1.2.031068", "symbB.v1.2.020543", "symbB.v1.2.016082", "symbB.v1.2.035240", "symbB.v1.2.027854", "symbB.v1.2.033722", "symbB.v1.2.040363", "symbB.v1.2.023765", "symbB.v1.2.040942", "symbB.v1.2.044196", "symbB.v1.2.015660", "symbB.v1.2.040296", "symbB.v1.2.016697", "symbB.v1.2.035896", "symbB.v1.2.003777", "symbB.v1.2.041360", "symbB.v1.2.010895", "symbB.v1.2.012137", "symbB.v1.2.014585", "symbB.v1.2.024463", "symbB.v1.2.033751", "symbB.v1.2.013127", "symbB.v1.2.034300", "symbB.v1.2.010759", "symbB.v1.2.003212", "symbB.v1.2.008803", "symbB.v1.2.033336", "symbB.v1.2.031619", "symbB.v1.2.008514", "symbB.v1.2.036299", "symbB.v1.2.017480", "symbB.v1.2.021284", "symbB.v1.2.028113", "symbB.v1.2.031543", "symbB.v1.2.041618", "symbB.v1.2.027536", "symbB.v1.2.034866", "symbB.v1.2.009814", "symbB.v1.2.005036", "symbB.v1.2.026552", "symbB.v1.2.006093", "symbB.v1.2.028077", "symbB.v1.2.019632", "symbB.v1.2.019156", "symbB.v1.2.017107", "symbB.v1.2.019924", "symbB.v1.2.013537", "symbB.v1.2.014853", "symbB.v1.2.035347", "symbB.v1.2.002509", "symbB.v1.2.027194", "symbB.v1.2.012928", "symbB.v1.2.029080", "symbB.v1.2.036874", "symbB.v1.2.016556", "symbB.v1.2.011075", "symbB.v1.2.042135", "symbB.v1.2.012472", "symbB.v1.2.019489", "symbB.v1.2.020046", "symbB.v1.2.036979", "symbB.v1.2.026609", "symbB.v1.2.034220", "symbB.v1.2.037146", "symbB.v1.2.027537", "symbB.v1.2.025480", "symbB.v1.2.019895", "symbB.v1.2.020562", "symbB.v1.2.015754", "symbB.v1.2.006324", "symbB.v1.2.011088", "symbB.v1.2.040681", "symbB.v1.2.037191", "symbB.v1.2.007494", "symbB.v1.2.024924", "symbB.v1.2.032716", "symbB.v1.2.027626", "symbB.v1.2.033271", "symbB.v1.2.006444", "symbB.v1.2.000471", "symbB.v1.2.023419", "symbB.v1.2.010300", "symbB.v1.2.031787", "symbB.v1.2.002740", "symbB.v1.2.005415", "symbB.v1.2.011430", "symbB.v1.2.006894", "symbB.v1.2.005898", "g3039", "g25968", "g21490", "g9043", "g20949", "symbB.v1.2.041403", "symbB.v1.2.001035", "g18888")

a <- VlnPlot_scCustom(ofav.combined.labeled.symb, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Symbiont Markers to Plot", subtitle = "Significant DEG between Healthy and Mid-bleach 1")

pdf("figures/Bleaching_Symbiont_DEGs/brevwhat2plot.pdf", width=15, height=45)
plot(a)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.040993"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
combined_expression <- expression_gene1
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Symbiont Gene", subtitle="Genes: symbB.v1.2.040993") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont1.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "symbB.v1.2.000589"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
combined_expression <- expression_gene1
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Peridinin-chlorophyll A binding protein", subtitle="Genes: symbB.v1.2.000589") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont2.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "symbB.v1.2.032431"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
combined_expression <- expression_gene1
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "glyceraldehyde 3-phosphate dehydrogenase (phosphorylating)", subtitle="Genes: symbB.v1.2.032431") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont3.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "symbB.v1.2.037890"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
combined_expression <- expression_gene1
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "ferredoxin", subtitle="Genes: symbB.v1.2.037890") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont4.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "symbB.v1.2.033151"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
combined_expression <- expression_gene1
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Symbiont Gene", subtitle="Genes: symbB.v1.2.033151") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont5.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "symbB.v1.2.041360"
gene2 <- "g30271"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
combined_expression <- expression_gene1 + expression_gene2
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "RuBisCO (Large)", subtitle="Genes: symbB.v1.2.041360 + g30271") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont6.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "symbB.v1.2.017107"
gene2 <- "g3952"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
combined_expression <- expression_gene1 + expression_gene2
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "phosphoglycerate kinase", subtitle="Genes: symbB.v1.2.017107 + g30271") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont7.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
gene1 <- "g21490"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
combined_expression <- expression_gene1
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Uncharacterized Symbiont Gene", subtitle="Genes: g21490") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont8.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "g9043"
gene2 <- "symbB.v1.2.039345"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
combined_expression <- expression_gene1
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5), axis.text.x = element_text(angle = 90)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "hondroitin N-acetylgalactosaminyltransferase", subtitle="Genes: g9043 + symbB.v1.2.039345") + 
  theme(legend.position = "none") + scale_fill_manual(values=c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")) 
p1

pdf("figures/Bleaching_Symbiont_DEGs/Violin_FullBleachingMarker_symbiont9.pdf", width=7, height=5)
plot(p1)
dev.off() 
```



# GO Between timpeoints
```{r}
library(Seurat)
library(tidyverse)
library(magrittr)
library(ReactomePA)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(DOSE)
library("topGO")
ofav.combined.labeled.DE <- subset(x = ofav.combined.labeled, idents = c("Unidentified"), invert = TRUE)
levels(Idents(ofav.combined.labeled.DE))
ofav.combined.labeled.SC <- RenameIdents(ofav.combined.labeled.DE, "Digestive Filaments 1"= "Digestive Filaments", "In-hospite Breviolum"= "Breviolum", "Expelled Breviolum"= "Breviolum", "Calicoblastic Cells"= "Calicoblastic Cells", "Gastrodermis 1"= "Gastrodermis", "Gastrodermis 2"= "Gastrodermis", "Gastrodermis 3"= "Gastrodermis", "Neurons 1"= "Neurons", "Immune"= "Immune", "Neurons 2"= "Neurons", "Gland 1"= "Gland", "Neurons 3"= "Neurons", "Gastrodermis (Muscle-like)"= "Gastrodermis (Muscle-like)", "In-hospite Durusdinium"= "Durusdinium", "Digestive Filaments 2"= "Digestive Filaments", "Precursors"="Precursors", "Gland 2"= "Gland", "Cnidocytes"="Cnidocytes")
ofav.combined.labeled.SC$super_clusters <- paste(Idents(ofav.combined.labeled.SC), ofav.combined.labeled.SC$orig.ident, sep = "_")
unique(ofav.combined.labeled.SC$super_clusters)
Idents(ofav.combined.labeled.SC) <- "super_clusters"
levels(Idents(ofav.combined.labeled.SC))
```

## Digestive Filaments Cells
```{r}
## Bleaching
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.SC, idents = c("Digestive Filaments_Ofav-T0-T", "Digestive Filaments_Ofav-T7-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["Digestive Filaments_Ofav-T7-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "red") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO Digestive Filaments: Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_Digestive_Filaments_bleaching.pdf", width=15, height=10)
plot(GO_plot)
dev.off()
```

## Symbiosomes ("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T3-T", "In-hospite Breviolum_Ofav-T5-T", "In-hospite Breviolum_Ofav-T7-T")
### Breviolum
```{r}
ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T7-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["In-hospite Breviolum_Ofav-T7-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkgoldenrod3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO In-hospite Breviolum Symbiosomes: Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_In-hospite-Breviolum_bleaching.pdf", width=25, height=10)
plot(GO_plot)
dev.off()

ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T3-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["In-hospite Breviolum_Ofav-T3-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkgoldenrod3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO In-hospite Breviolum Symbiosomes: Early Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_In-hospite-Breviolum_earlybleaching.pdf", width=25, height=10)
plot(GO_plot)
dev.off()


ofav.combined.labeled.DE.subset <- subset(x = ofav.combined.labeled.DE, idents = c("In-hospite Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T5-T"))
heatstress.response2 <- FindAllMarkers(ofav.combined.labeled.DE.subset,only.pos=T, return.thresh=0.05, min.pct= .01)
heatstress.response2 <- heatstress.response2[heatstress.response2$p_val_adj <= 0.05, ]
formatted_data <- list()
for (cluster_id in unique(heatstress.response2$cluster)) {
  cluster_genes <- heatstress.response2$gene[heatstress.response2$cluster == cluster_id]
  cluster_pvalues <- heatstress.response2$p_val_adj[heatstress.response2$cluster == cluster_id]
  cluster_data <- data.frame(genes = cluster_genes, pvalues = as.numeric(cluster_pvalues))
  rownames(cluster_data) <- cluster_genes
  formatted_data[[as.character(cluster_id)]] <- cluster_data
}
genes_per_cluster <- formatted_data
gocluster1 <- genes_per_cluster[["In-hospite Breviolum_Ofav-T5-T"]]
# Convert genes and pvalues columns to a named list
text_data <- read.csv(file = "id2symbol2.txt", sep="\t", header = TRUE, stringsAsFactors = FALSE)
merged_data <- merge(gocluster1, text_data, by.x = "genes", by.y = "ID", all.x = TRUE)
merged_data$genes <- ifelse(!is.na(merged_data$symbol), merged_data$symbol, merged_data$genes)
merged_data <- merged_data[complete.cases(merged_data$symbol), ]
genes_list <- setNames(merged_data$pvalues, merged_data$genes)
range(genes_list)
selection <- function(allScore){ return(allScore < 0.05)}
allGO2genes <- annFUN.org(whichOnto="BP", feasibleGenes=NULL, mapping="org.Hs.eg.db", ID="symbol")
GOdata <- new("topGOdata",
  ontology="BP",
  allGenes=genes_list,
  annot=annFUN.GO2genes,
  GO2genes=allGO2genes,
  geneSel=selection,
  nodeSize=10)
results.ks <- runTest(GOdata, algorithm="classic", statistic="ks")
goEnrichment <- GenTable(GOdata, KS=results.ks, orderBy="KS", topNodes=20)
goEnrichment$KS <- as.numeric(goEnrichment$KS)
goEnrichment <- goEnrichment[goEnrichment$KS<0.05,]
goEnrichment <- goEnrichment[,c("GO.ID","Term","KS")]
goEnrichment$Term <- gsub(" [a-z]*\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- gsub("\\.\\.\\.$", "", goEnrichment$Term)
goEnrichment$Term <- paste(goEnrichment$GO.ID, goEnrichment$Term, sep=", ")
goEnrichment$Term <- factor(goEnrichment$Term, levels=rev(goEnrichment$Term))
GO_plot <- ggplot(goEnrichment, aes(x=Term, y=-log10(KS))) +
    stat_summary(geom = "bar", fun.y = mean, position = "dodge", fill = "darkgoldenrod3") +
    xlab("Biological process") +
    ylab("Enrichment (-log10 of P-value)") +
    ggtitle("GO In-hospite Breviolum Symbiosomes: Late Bleaching Response") +
    scale_y_continuous(limits = c(0,5), expand = c(0,0)) +
    theme_bw(base_size=24) +
    theme(
        legend.position='none',
        legend.background=element_rect(),
        plot.title=element_text(angle=0, size=24, face="bold", vjust=1),
        axis.text.x=element_text(angle=0, size=18, face="bold", hjust=1.10),
        axis.text.y=element_text(angle=0, size=18, face="bold", vjust=0.5),
        axis.title=element_text(size=24, face="bold"),
        legend.key=element_blank(),     #removes the border
        legend.key.size=unit(1, "cm"),      #Sets overall area/size of the legend
        legend.text=element_text(size=18),  #Text size
        title=element_text(size=18)) +
    guides(colour=guide_legend(override.aes=list(size=2.5))) +
    coord_flip()

pdf("figures/Bleaching_GOs/GO_In-hospite-Breviolum_latebleaching.pdf", width=25, height=10)
plot(GO_plot)
dev.off()
```



## Giant Heat-Map with All Markers across timepoints (https://bioinformatics.stackexchange.com/questions/4851/how-i-can-reproduce-this-heat-map)
```{r}
genes_to_use <- c("QW917-005606",
"QW917-027542",
"QW917-008447",
"QW917-003971",
"QW917-004231",
"QW917-026433",
"QW917-006427",
"QW917-027545",
"QW917-036332",
"QW917-017799",
"QW917-031915",
"QW917-030986",
"QW917-035634",
"QW917-016016",
"QW917-036331",
"QW917-035635",
"QW917-026903",
"QW917-019191",
"QW917-009324",
"QW917-025786",
"QW917-014742",
"QW917-029172",
"QW917-010146",
"QW917-034907",
"QW917-001863",
"QW917-021056",
"QW917-030646",
"QW917-008341",
"QW917-015206",
"QW917-031586",
"QW917-013627",
"QW917-032489",
"QW917-029254",
"QW917-010145",
"QW917-005586",
"QW917-030285",
"QW917-005300",
"QW917-033575",
"QW917-018305",
"QW917-009678",
"QW917-023565",
"QW917-020799",
"QW917-013974",
"QW917-009336",
"QW917-030668",
"QW917-024190",
"QW917-024765",
"QW917-028699",
"QW917-009778",
"QW917-024271",
"QW917-032268",
"QW917-006360",
"QW917-010242",
"QW917-013452",
"QW917-000347",
"QW917-025432",
"QW917-017111",
"QW917-001602",
"QW917-018265",
"QW917-007874",
"QW917-023065",
"QW917-027476",
"QW917-001645",
"symbB.v1.2.033380",
"QW917-003902",
"QW917-001600",
"QW917-008749",
"QW917-007699",
"QW917-023502",
"QW917-002808",
"QW917-013386",
"QW917-031908",
"QW917-017802",
"QW917-024321",
"QW917-036231",
"QW917-004379",
"QW917-015173",
"QW917-036230",
"QW917-011121",
"QW917-029766",
"QW917-023414",
"QW917-010727",
"symbB.v1.2.040993",
"QW917-006430",
"QW917-015093",
"QW917-005903",
"QW917-017006",
"QW917-021090",
"QW917-002685",
"QW917-030399",
"QW917-015074",
"QW917-022246",
"QW917-015057",
"QW917-024491",
"QW917-036205",
"QW917-036204",
"QW917-010171",
"QW917-029646",
"QW917-016488",
"QW917-031733",
"QW917-010131",
"QW917-003749",
"QW917-017291",
"QW917-010212",
"QW917-029192",
"QW917-024964",
"QW917-025659",
"QW917-027201",
"QW917-010456",
"QW917-027442",
"QW917-024490",
"QW917-003567",
"QW917-017502",
"QW917-028556",
"QW917-027832",
"QW917-008245",
"QW917-033447",
"QW917-002688",
"QW917-002686",
"QW917-022625",
"QW917-016021",
"QW917-024605",
"QW917-014367",
"QW917-033785",
"QW917-029111",
"QW917-028062",
"QW917-017942",
"QW917-002399",
"QW917-008226",
"symbB.v1.2.036431",
"symbB.v1.2.037117",
"symbB.v1.2.037890",
"QW917-033692",
"QW917-029903",
"QW917-033413",
"QW917-031495",
"QW917-026114",
"QW917-002944",
"QW917-005451",
"QW917-002182",
"QW917-026948",
"QW917-005453",
"QW917-013256",
"QW917-002736",
"QW917-021098",
"QW917-008288",
"QW917-009852",
"QW917-014316",
"QW917-016496",
"QW917-025130",
"QW917-016878",
"QW917-005894",
"QW917-003972",
"QW917-016491",
"QW917-018168",
"QW917-035941",
"QW917-029781",
"QW917-001171",
"QW917-030101",
"QW917-018353",
"QW917-016214",
"QW917-005571",
"QW917-001525",
"QW917-013752",
"QW917-008368",
"QW917-006349",
"QW917-018150",
"QW917-028319",
"QW917-014250",
"QW917-020637",
"QW917-003326",
"QW917-022132",
"QW917-033275",
"QW917-007406",
"QW917-018090",
"QW917-025879",
"QW917-016501",
"QW917-016495",
"QW917-0133861",
"QW917-011148",
"QW917-034906",
"QW917-016123",
"QW917-028232",
"QW917-010184",
"QW917-0026851",
"QW917-0026881",
"QW917-0026861",
"QW917-035714",
"QW917-001227",
"QW917-024580",
"QW917-004807",
"QW917-0181691",
"QW917-022832",
"QW917-007767",
"QW917-034318",
"QW917-011241",
"QW917-0168781",
"QW917-026774",
"QW917-009632",
"QW917-031657",
"QW917-0165011",
"QW917-015587",
"QW917-000592",
"QW917-0095991",
"QW917-001282",
"QW917-000085",
"QW917-018979",
"QW917-032073",
"QW917-005962",
"QW917-028245",
"QW917-005969",
"QW917-024689",
"QW917-003080",
"QW917-019813",
"QW917-002611",
"QW917-027013",
"QW917-019333",
"QW917-010925",
"QW917-004830",
"QW917-030749",
"QW917-010422",
"QW917-014249",
"QW917-013995",
"QW917-029175",
"QW917-025186",
"QW917-032678",
"QW917-028405",
"QW917-030460",
"QW917-032411",
"QW917-009174",
"QW917-000806",
"QW917-025829",
"QW917-027780",
"QW917-025087",
"QW917-030466",
"QW917-018167",
"QW917-017896",
"QW917-024967",
"QW917-008484",
"QW917-028328",
"QW917-021960",
"QW917-033002",
"QW917-030981",
"QW917-012971",
"QW917-014631",
"QW917-018169",
"QW917-024396",
"QW917-024395",
"QW917-009835",
"QW917-013711",
"QW917-034316",
"QW917-033337",
"QW917-023536",
"QW917-012685",
"QW917-005425",
"QW917-005426",
"QW917-000356",
"QW917-005964",
"QW917-016859",
"QW917-008044",
"QW917-031830",
"QW917-026897",
"QW917-014552",
"QW917-013054",
"QW917-022605",
"QW917-002392",
"QW917-029780",
"QW917-002953",
"QW917-018152",
"QW917-000638",
"QW917-000629",
"QW917-023314",
"QW917-008780",
"QW917-024318",
"QW917-005475",
"QW917-027879",
"QW917-022587",
"QW917-017008",
"QW917-006331",
"QW917-0164961",
"QW917-032688",
"QW917-0180901",
"QW917-036526",
"QW917-002570",
"QW917-023024",
"QW917-000091",
"QW917-031543",
"QW917-026657",
"QW917-000010",
"QW917-026605",
"QW917-029435",
"QW917-031289",
"QW917-015447",
"QW917-019842",
"QW917-018166",
"QW917-0270131",
"QW917-029715",
"QW917-018981",
"QW917-0245801",
"QW917-027857",
"QW917-0048071",
"QW917-017710",
"QW917-021127",
"QW917-022831",
"QW917-019683",
"QW917-029232",
"QW917-029370",
"QW917-003517",
"QW917-017486",
"QW917-003512",
"QW917-0324111",
"QW917-003505",
"QW917-0181521",
"QW917-022895",
"QW917-029427",
"QW917-015399",
"QW917-024417",
"QW917-022142",
"QW917-035095",
"QW917-024960",
"QW917-001823",
"QW917-031651",
"QW917-022897",
"QW917-0063491",
"QW917-023825",
"QW917-033167",
"QW917-003400",
"QW917-0183531",
"QW917-031968",
"QW917-034154",
"QW917-027520",
"QW917-028683",
"QW917-024171",
"QW917-000002",
"QW917-031955",
"QW917-016270",
"QW917-0277801",
"QW917-0063492",
"QW917-019830",
"QW917-0250871",
"QW917-0101841",
"QW917-020885",
"QW917-033454",
"QW917-033406",
"QW917-018034",
"QW917-035384",
"QW917-020909",
"QW917-015443",
"QW917-001199",
"QW917-009143",
"QW917-003256",
"QW917-0164951",
"QW917-004075",
"QW917-0228321",
"QW917-005500")


levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")

ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")

ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled, idents = c("Expelled Breviolum"), invert = TRUE)
ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled.hm1, idents = c("In-hospite Breviolum"), invert = TRUE)
ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled.hm1, idents = c("In-hospite Durusdinium"), invert = TRUE)

unique(ofav.combined.labeled.hm1$seurat_clusters.Timepoints)
ofav.combined.labeled.hm <- ofav.combined.labeled.hm1
rm(ofav.combined.labeled.hm1)
Idents(ofav.combined.labeled.hm) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.hm)
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)
ofav.combined.labeled.hm<- ScaleData(ofav.combined.labeled.hm, features = all.genes)

my_levels <- c("Calicoblastic Cells_Ofav-T0-T", "Cnidocytes_Ofav-T0-T", "Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 2_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 2_Ofav-T0-T", "Gastrodermis 3_Ofav-T0-T", "Gland 1_Ofav-T0-T", "Gland 2_Ofav-T0-T", "Immune_Ofav-T0-T", "Neurons 1_Ofav-T0-T", "Neurons 2_Ofav-T0-T", "Neurons 3_Ofav-T0-T", "Precursors_Ofav-T0-T", "Unidentified_Ofav-T0-T", "Calicoblastic Cells_Ofav-T3-T", "Cnidocytes_Ofav-T3-T", "Digestive Filaments 1_Ofav-T3-T", "Digestive Filaments 2_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 2_Ofav-T3-T", "Gastrodermis 3_Ofav-T3-T", "Gland 1_Ofav-T3-T", "Gland 2_Ofav-T3-T", "Immune_Ofav-T3-T", "Neurons 1_Ofav-T3-T", "Neurons 2_Ofav-T3-T", "Neurons 3_Ofav-T3-T", "Precursors_Ofav-T3-T", "Unidentified_Ofav-T3-T", "Calicoblastic Cells_Ofav-T5-T", "Cnidocytes_Ofav-T5-T", "Digestive Filaments 1_Ofav-T5-T", "Digestive Filaments 2_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 2_Ofav-T5-T", "Gastrodermis 3_Ofav-T5-T", "Gland 1_Ofav-T5-T", "Gland 2_Ofav-T5-T", "Immune_Ofav-T5-T", "Neurons 1_Ofav-T5-T", "Neurons 2_Ofav-T5-T", "Neurons 3_Ofav-T5-T", "Precursors_Ofav-T5-T", "Unidentified_Ofav-T5-T", "Calicoblastic Cells_Ofav-T7-T", "Cnidocytes_Ofav-T7-T", "Digestive Filaments 1_Ofav-T7-T", "Digestive Filaments 2_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T7-T", "Gastrodermis 3_Ofav-T7-T", "Gland 1_Ofav-T7-T", "Gland 2_Ofav-T7-T", "Immune_Ofav-T7-T", "Neurons 1_Ofav-T7-T", "Neurons 2_Ofav-T7-T", "Neurons 3_Ofav-T7-T", "Precursors_Ofav-T7-T", "Unidentified_Ofav-T7-T")

colors_hm <- c("darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)

ofav.combined.labeled.hm@meta.data[,"seurat_clusters.Timepoints",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters.Timepoints) %>% slice_sample(n= 20) -> index

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + NoLegend() + scale_fill_gradientn(colors = c("blue", "white", "red"))

pdf("figures/Bleaching_DEGS/Bleaching-heatmap.pdf", height = 50, width=25)
plot(hm)
dev.off()
```

### Symbiosome Violin
```{r}
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)
ofav.combined.labeled.symb <- subset(x = ofav.combined.labeled, idents = c("Breviolum-hosting Cells", "Durusdinium-hosting Cells"), invert = F)
my_levels <- c("Breviolum-hosting Cells", "Durusdinium-hosting Cells")
ofav.combined.labeled.symb@active.ident <- factor(x = ofav.combined.labeled.symb@active.ident, levels = my_levels)
colors_to_use <- wes_palette("Zissou1", 4, type = "continuous")

features <- c("QW917-002688", "QW917-002686", "QW917-002685", "QW917-030646", "QW917-018353", "QW917-009336", "QW917-028319", "QW917-005606", "QW917-001525", "QW917-023319", "QW917-003256", "QW917-014705", "QW917-005571", "QW917-008443", "QW917-030668", "QW917-018851", "QW917-022673", "QW917-014250", "QW917-007699", "QW917-033195", "QW917-008828", "QW917-033575", "QW917-028232", "QW917-027187", "QW917-020637", "QW917-025759", "QW917-031802", "QW917-023845", "QW917-003054", "QW917-029254", "QW917-011254", "QW917-024271", "QW917-013452", "QW917-032489", "QW917-029992", "QW917-013974", "QW917-013627", "QW917-023565", "QW917-026114", "QW917-030101", "QW917-010381", "QW917-018150", "QW917-027476", "QW917-005894", "QW917-018090", "QW917-017896", "QW917-016501", "QW917-016495")

a <- VlnPlot_scCustom(ofav.combined.labeled.symb, features = features, stack = TRUE, flip = TRUE, split.by="orig.ident", colors_use = colors_to_use) +
        theme(legend.position = "none") + ggtitle("Symbiosome Symbiont-specific Markers")

pdf("figures/Bleaching_Symbiont_DEGS/Violin_DurVsBrev.pdf", width=10, height=40)
plot(a)
dev.off()

```


### Heatmap with Symbionts-only
```{r}
genes_to_use <- c("symbB.v1.2.027854",
"symbB.v1.2.037202",
"symbB.v1.2.032431",
"symbB.v1.2.000589",
"symbB.v1.2.037117",
"QW917-005606",
"QW917-006349",
"QW917-001525",
"QW917-018353",
"symbB.v1.2.041403",
"QW917-018150",
"QW917-024916",
"QW917-004226",
"g25968",
"g20949",
"QW917-010073",
"QW917-014572",
"QW917-016878",
"QW917-011566",
"QW917-006021",
"QW917-016963",
"QW917-009608",
"QW917-017710",
"QW917-034320",
"QW917-018090",
"QW917-026789",
"QW917-029903",
"QW917-026774",
"QW917-028328",
"QW917-026917",
"QW917-017815",
"QW917-0058941",
"QW917-017800",
"QW917-013501",
"QW917-001647",
"QW917-035955",
"QW917-024244",
"QW917-007767",
"QW917-016794",
"QW917-027780",
"QW917-032620",
"QW917-012295",
"QW917-024547",
"QW917-024520",
"QW917-027982",
"QW917-017490",
"QW917-005296",
"QW917-006360",
"QW917-024967",
"QW917-022587",
"QW917-006331",
"QW917-027013",
"QW917-018355",
"QW917-003428",
"QW917-018305",
"QW917-024580",
"QW917-025087",
"QW917-024396",
"QW917-028697",
"QW917-011148",
"QW917-009678",
"QW917-0026851",
"QW917-004770",
"QW917-0183251",
"QW917-0026881",
"QW917-023502",
"QW917-034907",
"QW917-0134521",
"QW917-0331951",
"QW917-0257591",
"QW917-0088281",
"QW917-0003471",
"QW917-033575",
"QW917-015220",
"QW917-028792",
"QW917-010184",
"QW917-012688",
"QW917-033337",
"QW917-002293",
"QW917-002484",
"QW917-016501",
"QW917-013512",
"QW917-003080",
"QW917-024468",
"QW917-032678",
"QW917-028683",
"QW917-024689",
"QW917-023825",
"QW917-025055",
"QW917-015399",
"QW917-026128",
"QW917-034255",
"QW917-005100",
"QW917-025879",
"QW917-015477",
"QW917-004807",
"QW917-018169",
"QW917-031065",
"QW917-029252",
"QW917-001013",
"QW917-026160",
"QW917-027086",
"QW917-024501",
"QW917-005431",
"QW917-016519",
"QW917-013053",
"QW917-028783",
"QW917-016462",
"QW917-024485",
"QW917-035803",
"QW917-018818",
"QW917-034023",
"QW917-030863",
"QW917-028845",
"QW917-009835",
"QW917-0178961",
"QW917-029570",
"QW917-009852",
"QW917-029773",
"QW917-006393",
"QW917-0210561",
"QW917-005301",
"QW917-017714",
"QW917-0164951",
"QW917-032411",
"QW917-002144",
"QW917-011523",
"QW917-0039071",
"QW917-0139741",
"QW917-002392",
"QW917-000926",
"QW917-005962",
"QW917-002570",
"QW917-005969",
"QW917-035095",
"QW917-005964",
"QW917-032073",
"QW917-028245",
"QW917-016859",
"QW917-000010",
"QW917-018167",
"QW917-027857",
"QW917-002611",
"QW917-0245801",
"QW917-030749")


levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")

ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")

ofav.combined.labeled.hm1 <- subset(x = ofav.combined.labeled, idents = c("Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium"), invert = F)

unique(ofav.combined.labeled.hm1$seurat_clusters.Timepoints)
ofav.combined.labeled.hm <- ofav.combined.labeled.hm1
rm(ofav.combined.labeled.hm1)
Idents(ofav.combined.labeled.hm) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.hm)
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)
ofav.combined.labeled.hm<- ScaleData(ofav.combined.labeled.hm, features = all.genes)


my_levels <- c("Expelled Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T0-T","Expelled Breviolum_Ofav-T3-T", "In-hospite Breviolum_Ofav-T3-T", "In-hospite Durusdinium_Ofav-T3-T", "Expelled Breviolum_Ofav-T5-T", "In-hospite Breviolum_Ofav-T5-T", "In-hospite Durusdinium_Ofav-T5-T", "Expelled Breviolum_Ofav-T7-T", "In-hospite Breviolum_Ofav-T7-T", "In-hospite Durusdinium_Ofav-T7-T")

colors_hm <- c("darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)

ofav.combined.labeled.hm@meta.data[,"seurat_clusters.Timepoints",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters.Timepoints) %>% slice_sample(n= 50) -> index

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + NoLegend() + scale_fill_gradientn(colors = c("blue", "white", "red"))

pdf("figures/Bleaching_DEGS/Bleaching-Symbionts-heatmap.pdf", height = 30, width=25)
plot(hm)
dev.off()
```

### Combined Heatmap of these Genes
```{r}
genes_to_use <- c("QW917-005606",
"QW917-036332",
"QW917-036331",
"QW917-027542",
"QW917-017111",
"QW917-025432",
"QW917-030986",
"QW917-008447",
"symbB.v1.2.040993",
"QW917-026433",
"QW917-010131",
"QW917-016488",
"QW917-023065",
"QW917-003749",
"QW917-001602",
"QW917-017006",
"QW917-031733",
"QW917-017942",
"QW917-026114",
"QW917-008288",
"QW917-003971",
"QW917-010145",
"QW917-035634",
"QW917-004231",
"QW917-024491",
"QW917-030646",
"QW917-013386",
"QW917-016878",
"QW917-016501",
"QW917-016496",
"QW917-006349",
"QW917-025130",
"QW917-005894",
"QW917-024580",
"QW917-018150",
"QW917-008368",
"QW917-033275",
"QW917-018353",
"QW917-004807",
"QW917-016495",
"QW917-002688",
"QW917-002686",
"QW917-018169",
"QW917-003080",
"QW917-026774",
"QW917-027013",
"QW917-018090",
"QW917-024689",
"QW917-009174",
"QW917-004830",
"QW917-005962",
"QW917-008044",
"QW917-014249",
"QW917-028245",
"QW917-032073",
"QW917-016859",
"QW917-026114")


levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")

ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")

ofav.combined.labeled.hm1 <- ofav.combined.labeled

unique(ofav.combined.labeled.hm1$seurat_clusters.Timepoints)
ofav.combined.labeled.hm <- ofav.combined.labeled.hm1
rm(ofav.combined.labeled.hm1)
Idents(ofav.combined.labeled.hm) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.hm)
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)
ofav.combined.labeled.hm<- ScaleData(ofav.combined.labeled.hm, features = all.genes)

my_levels <- c("Unidentified_Ofav-T0-T", "Calicoblastic Cells_Ofav-T0-T", "Cnidocytes_Ofav-T0-T", "Digestive Filaments_Ofav-T0-T", "Epidermis_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 2_Ofav-T0-T", "Gland 1_Ofav-T0-T", "Gland 2_Ofav-T0-T", "Gland 3_Ofav-T0-T", "Immune_Ofav-T0-T", "Neurons 1_Ofav-T0-T", "Neurons 2_Ofav-T0-T", "Neurons 3_Ofav-T0-T", "Precursors_Ofav-T0-T", "Extracellular Breviolum_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Unidentified_Ofav-T3-T", "Calicoblastic Cells_Ofav-T3-T", "Cnidocytes_Ofav-T3-T", "Digestive Filaments_Ofav-T3-T", "Epidermis_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 2_Ofav-T3-T", "Gland 1_Ofav-T3-T", "Gland 2_Ofav-T3-T", "Gland 3_Ofav-T3-T", "Immune_Ofav-T3-T", "Neurons 1_Ofav-T3-T", "Neurons 2_Ofav-T3-T", "Neurons 3_Ofav-T3-T", "Precursors_Ofav-T3-T", "Extracellular Breviolum_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Unidentified_Ofav-T5-T", "Calicoblastic Cells_Ofav-T5-T", "Cnidocytes_Ofav-T5-T", "Digestive Filaments_Ofav-T5-T", "Epidermis_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 2_Ofav-T5-T", "Gland 1_Ofav-T5-T", "Gland 2_Ofav-T5-T", "Gland 3_Ofav-T5-T", "Immune_Ofav-T5-T", "Neurons 1_Ofav-T5-T", "Neurons 2_Ofav-T5-T", "Neurons 3_Ofav-T5-T", "Precursors_Ofav-T5-T", "Extracellular Breviolum_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Unidentified_Ofav-T7-T", "Calicoblastic Cells_Ofav-T7-T", "Cnidocytes_Ofav-T7-T", "Digestive Filaments_Ofav-T7-T", "Epidermis_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T7-T", "Gland 1_Ofav-T7-T", "Gland 2_Ofav-T7-T", "Gland 3_Ofav-T7-T", "Immune_Ofav-T7-T", "Neurons 1_Ofav-T7-T", "Neurons 2_Ofav-T7-T", "Neurons 3_Ofav-T7-T", "Precursors_Ofav-T7-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T7-T")

colors_hm <- c("azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)

ofav.combined.labeled.hm@meta.data[,"seurat_clusters.Timepoints",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters.Timepoints) %>% slice_sample(n= 25) -> index

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + NoLegend() + scale_fill_gradientn(colors = c("blue", "#fffff2", "red"))

pdf("figures/Bleaching_DEGS/Bleaching-combined-heatmap.pdf", height = 25, width=30)
plot(hm)
dev.off()

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + scale_fill_gradientn(colors = c("blue", "#fffff2", "red"))

pdf("figures/Bleaching_DEGS/Bleaching-combined-heatmap_wlegend.pdf", height = 25, width=50)
plot(hm)
dev.off()
```

### Combined Heatmap of these Genes
```{r}
genes_to_use <- c("QW917-005606",
"QW917-027542",
"QW917-008447",
"QW917-025432",
"QW917-017111",
"QW917-001602",
"QW917-030986",
"QW917-036231",
"QW917-004379",
"QW917-015173",
"QW917-036230",
"symbB.v1.2.027854",
"symbB.v1.2.037202",
"QW917-029766",
"QW917-023414",
"QW917-010727",
"QW917-006427",
"symbB.v1.2.040993",
"QW917-006430",
"QW917-015093",
"QW917-005903",
"QW917-017006",
"QW917-021090",
"QW917-021056",
"QW917-002685",
"QW917-030399",
"QW917-015074",
"QW917-036205",
"QW917-036204",
"QW917-010171",
"QW917-013386",
"QW917-026433",
"QW917-031915",
"QW917-016488",
"QW917-031733",
"QW917-010131",
"QW917-003749",
"QW917-017291",
"QW917-010212",
"QW917-029192",
"QW917-024964",
"QW917-025659",
"QW917-027201",
"QW917-010456",
"QW917-027442",
"QW917-024490",
"QW917-023065",
"QW917-003567",
"QW917-029646",
"QW917-017942",
"QW917-002399",
"symbB.v1.2.036431",
"symbB.v1.2.037117",
"symbB.v1.2.037890",
"QW917-033692",
"QW917-017799",
"QW917-030646",
"symbB.v1.2.032431",
"symbB.v1.2.000589",
"QW917-002944",
"QW917-005451",
"QW917-002182",
"QW917-026948",
"QW917-016496",
"QW917-025130",
"QW917-008368",
"QW917-006349",
"QW917-001525",
"QW917-018353",
"symbB.v1.2.041403",
"QW917-018150",
"QW917-022132",
"QW917-033275",
"QW917-007406",
"QW917-025879",
"QW917-024916",
"QW917-004226",
"g25968",
"g20949",
"QW917-010073",
"QW917-016501",
"QW917-016878",
"QW917-0168781",
"QW917-014572",
"QW917-011566",
"QW917-006021",
"QW917-016963",
"QW917-009608",
"QW917-017710",
"QW917-034320",
"QW917-018090",
"QW917-026789",
"QW917-029903",
"QW917-026774",
"QW917-028328",
"QW917-026917",
"QW917-017815",
"QW917-0058941",
"QW917-017800",
"QW917-013501",
"QW917-001647",
"QW917-035955",
"QW917-024244",
"QW917-007767",
"QW917-016794",
"QW917-027780",
"QW917-032620",
"QW917-012295",
"QW917-024547",
"QW917-024520",
"QW917-027982",
"QW917-017490",
"QW917-005296",
"QW917-006360",
"QW917-024967",
"QW917-022587",
"QW917-006331",
"QW917-027013",
"QW917-018355",
"QW917-003428",
"QW917-018305",
"QW917-033337",
"QW917-024580",
"QW917-025087",
"QW917-024396",
"QW917-028697",
"QW917-011148",
"QW917-009678",
"QW917-0026851",
"QW917-004770",
"QW917-0183251",
"QW917-0026881",
"QW917-023502",
"QW917-034907",
"QW917-002293",
"QW917-0134521",
"QW917-0331951",
"QW917-0257591",
"QW917-002484",
"QW917-0088281",
"QW917-0003471",
"QW917-033575",
"QW917-015220",
"QW917-028792",
"QW917-010184",
"QW917-012688",
"QW917-009632",
"QW917-015587",
"QW917-000592",
"QW917-000085",
"QW917-018979",
"QW917-032073",
"QW917-005962",
"QW917-028245",
"QW917-005969",
"QW917-024689",
"QW917-003080",
"QW917-019813",
"QW917-002611",
"QW917-019333",
"QW917-010925",
"QW917-004830",
"QW917-030749",
"QW917-010422",
"QW917-014249",
"QW917-013995",
"QW917-029175",
"QW917-012685",
"QW917-005425",
"QW917-005426",
"QW917-000356",
"QW917-005964",
"QW917-016859",
"QW917-008044",
"QW917-031830",
"QW917-009174",
"QW917-026897",
"QW917-028405",
"QW917-014552",
"QW917-032411",
"QW917-016495",
"QW917-036526",
"QW917-002570",
"QW917-023024",
"QW917-000091",
"QW917-000806",
"QW917-031543",
"QW917-026657",
"QW917-000010",
"QW917-026605",
"QW917-029435",
"QW917-031289",
"QW917-015447",
"QW917-019842",
"QW917-018166",
"QW917-0270131",
"QW917-029715",
"QW917-018169",
"QW917-018981",
"QW917-0245801",
"QW917-017896",
"QW917-027857",
"QW917-0048071",
"QW917-013512",
"QW917-024468",
"QW917-032678",
"QW917-028683",
"QW917-023825",
"QW917-025055",
"QW917-002392",
"QW917-015399",
"QW917-026128",
"QW917-034255",
"QW917-005100",
"QW917-015477",
"QW917-004807",
"QW917-031065",
"QW917-029252",
"QW917-001013",
"QW917-026160",
"QW917-027086",
"QW917-024501",
"QW917-005431",
"QW917-016519",
"QW917-013053",
"QW917-028783",
"QW917-016462",
"QW917-024485",
"QW917-035803",
"QW917-018818",
"QW917-000926",
"QW917-034023",
"QW917-030863",
"QW917-028845",
"QW917-009835",
"QW917-0178961",
"QW917-029570",
"QW917-009852",
"QW917-029773",
"QW917-006393",
"QW917-0210561",
"QW917-005301",
"QW917-017714",
"QW917-0164951",
"QW917-002144",
"QW917-011523",
"QW917-0039071",
"QW917-0139741",
"QW917-018167",
"QW917-022831",
"QW917-019683",
"QW917-029232",
"QW917-029370",
"QW917-003517",
"QW917-030981",
"QW917-017486",
"QW917-003512",
"QW917-024417",
"QW917-022142",
"QW917-035095",
"QW917-024960",
"QW917-001823",
"QW917-031651",
"QW917-018152",
"QW917-031955",
"QW917-016270",
"QW917-019830",
"QW917-033454",
"QW917-033406",
"QW917-018034",
"QW917-012971",
"QW917-005500")


levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments 1", "Digestive Filaments 2", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gastrodermis 3", "Gland 1", "Gland 2", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Expelled Breviolum", "In-hospite Breviolum", "In-hospite Durusdinium")

ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")

ofav.combined.labeled.hm1 <- ofav.combined.labeled

unique(ofav.combined.labeled.hm1$seurat_clusters.Timepoints)
ofav.combined.labeled.hm <- ofav.combined.labeled.hm1
rm(ofav.combined.labeled.hm1)
Idents(ofav.combined.labeled.hm) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.hm)
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)
ofav.combined.labeled.hm<- ScaleData(ofav.combined.labeled.hm, features = all.genes)

my_levels <- c("Calicoblastic Cells_Ofav-T0-T", "Cnidocytes_Ofav-T0-T", "Digestive Filaments 1_Ofav-T0-T", "Digestive Filaments 2_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 2_Ofav-T0-T", "Gastrodermis 3_Ofav-T0-T", "Gland 1_Ofav-T0-T", "Gland 2_Ofav-T0-T", "Immune_Ofav-T0-T", "Neurons 1_Ofav-T0-T", "Neurons 2_Ofav-T0-T", "Neurons 3_Ofav-T0-T", "Precursors_Ofav-T0-T", "Unidentified_Ofav-T0-T", "Expelled Breviolum_Ofav-T0-T", "In-hospite Breviolum_Ofav-T0-T", "In-hospite Durusdinium_Ofav-T0-T", "Calicoblastic Cells_Ofav-T3-T", "Cnidocytes_Ofav-T3-T", "Digestive Filaments 1_Ofav-T3-T", "Digestive Filaments 2_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 2_Ofav-T3-T", "Gastrodermis 3_Ofav-T3-T", "Gland 1_Ofav-T3-T", "Gland 2_Ofav-T3-T", "Immune_Ofav-T3-T", "Neurons 1_Ofav-T3-T", "Neurons 2_Ofav-T3-T", "Neurons 3_Ofav-T3-T", "Precursors_Ofav-T3-T", "Unidentified_Ofav-T3-T", "Expelled Breviolum_Ofav-T3-T", "In-hospite Breviolum_Ofav-T3-T", "In-hospite Durusdinium_Ofav-T3-T", "Calicoblastic Cells_Ofav-T5-T", "Cnidocytes_Ofav-T5-T", "Digestive Filaments 1_Ofav-T5-T", "Digestive Filaments 2_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 2_Ofav-T5-T", "Gastrodermis 3_Ofav-T5-T", "Gland 1_Ofav-T5-T", "Gland 2_Ofav-T5-T", "Immune_Ofav-T5-T", "Neurons 1_Ofav-T5-T", "Neurons 2_Ofav-T5-T", "Neurons 3_Ofav-T5-T", "Precursors_Ofav-T5-T", "Unidentified_Ofav-T5-T", "Expelled Breviolum_Ofav-T5-T", "In-hospite Breviolum_Ofav-T5-T", "In-hospite Durusdinium_Ofav-T5-T", "Calicoblastic Cells_Ofav-T7-T", "Cnidocytes_Ofav-T7-T", "Digestive Filaments 1_Ofav-T7-T", "Digestive Filaments 2_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T7-T", "Gastrodermis 3_Ofav-T7-T", "Gland 1_Ofav-T7-T", "Gland 2_Ofav-T7-T", "Immune_Ofav-T7-T", "Neurons 1_Ofav-T7-T", "Neurons 2_Ofav-T7-T", "Neurons 3_Ofav-T7-T", "Precursors_Ofav-T7-T", "Unidentified_Ofav-T7-T", "Expelled Breviolum_Ofav-T7-T", "In-hospite Breviolum_Ofav-T7-T", "In-hospite Durusdinium_Ofav-T7-T")

colors_hm <- c("darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "darkorchid4", "plum1", "red1", "red3", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkolivegreen3", "darkorange1", "darkorange3", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "azure2", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)

ofav.combined.labeled.hm@meta.data[,"seurat_clusters.Timepoints",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters.Timepoints) %>% slice_sample(n= 25) -> index

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + NoLegend() + scale_fill_gradientn(colors = c("blue", "white", "red"))

pdf("figures/Bleaching_DEGS/Bleaching-combined-heatmap_LFC2.pdf", height = 75, width=25)
plot(hm)
dev.off()
```


# Heatmap of Marker Genes
```{r}
ofav.markers<- FindAllMarkers(ofav.combined.labeled, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.1, return.thresh=0.05)
ofav.markers <- ofav.markers[ofav.markers$p_val_adj<0.05,]
ofav.markers <- ofav.markers[ofav.markers$gene != "QW917-005894", ]
ofav.markers <- ofav.markers[ofav.markers$gene != "QW917-006349", ]
ofav.markers <- ofav.markers[ofav.markers$gene != "QW917-007699", ]
ofav.markers %>% group_by(cluster) %>% top_n(5, avg_log2FC) -> top10

levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")
ofav.combined.labeled.hm <- ofav.combined.labeled
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)

colors_hm <- c("azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)
ofav.combined.labeled.hm@meta.data[,"seurat_clusters",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters) %>% sample_n(40) -> index
hm <- DoHeatmap(ofav.combined.labeled.hm, slot="data", features = top10$gene,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + scale_fill_gradientn(colors = c("white", "red"), na.value = "white")

pdf("figures/marker-genes-heatmap2.pdf", height = 30, width=25)
plot(hm)
dev.off()


##Cluster by Big Markers


```


```{r}

levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")
ofav.combined.labeled.hm <- ofav.combined.labeled
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)

colors_hm <- c("azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)
ofav.combined.labeled.hm@meta.data[,"seurat_clusters",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters) %>% sample_n(40) -> index
hm <- DoHeatmap(ofav.combined.labeled.hm, slot= "data", features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + scale_fill_gradientn(colors = c("white", "red"), na.value = "white")

pdf("figures/marker-genes-heatmap2.pdf", height = 30, width=25)
plot(hm)
dev.off()
```



# Heatmap of Marker Genes (Super-clusters)
```{r}
ofav.combined.labeled.SC <- RenameIdents(ofav.combined.labeled, "Unidentified"="Unidentified", "Calicoblastic Cells"="Calicoblastic Cells", "Cnidocytes"="Cnidocytes", "Digestive Filaments"="Digestive Filaments", "Epidermis"="Epidermis", "Gastrodermis (Muscle-like)"="Gastrodermis (Muscle-like)", "Gastrodermis 1"="Gastrodermis", "Gastrodermis 2"="Gastrodermis", "Gland 1"="Gland", "Gland 2"="Gland", "Gland 3"="Gland", "Immune"="Immune", "Neurons 1"="Neurons", "Neurons 2"="Neurons", "Neurons 3"="Neurons", "Precursors"="Precursors", "Extracellular Breviolum"="Breviolum", "Breviolum-hosting Cells"="Breviolum", "Durusdinium-hosting Cells"=	"Durusdinium")
levels(ofav.combined.labeled.SC)

ofav.combined.labeled.SC$super_clusters <- paste(Idents(ofav.combined.labeled.SC))
unique(ofav.combined.labeled.SC$super_clusters)
Idents(ofav.combined.labeled.SC) <- "super_clusters"
levels(Idents(ofav.combined.labeled.SC))
ofav.combined.labeled.SC@active.ident <- factor(x = ofav.combined.labeled.SC@active.ident, levels = my_levels)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis", "Gland", "Immune", "Neurons", "Precursors", "Breviolum", "Durusdinium")

ofav.markers<- FindAllMarkers(ofav.combined.labeled.SC, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25, return.thresh=0.05)
ofav.markers <- ofav.markers[ofav.markers$p_val_adj<0.05,]
ofav.markers %>% group_by(cluster) %>% top_n(7, avg_log2FC) -> top10

levels(ofav.combined.labeled.SC)

DefaultAssay(ofav.combined.labeled.SC) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.SC)

colors_hm <- c("azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkorange", "darkkhaki", "deepskyblue", "bisque3", "darkgoldenrod1", "darkgoldenrod4")

ofav.combined.labeled.SC@active.ident <- factor(x = ofav.combined.labeled.SC@active.ident, levels = my_levels)

ofav.combined.labeled.SC@meta.data[,"seurat_clusters",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters) %>% sample_n(20) -> index

hm <- DoHeatmap(ofav.combined.labeled.SC, slot="data", features = top10$gene,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + scale_fill_gradientn(colors = c("#fffff2", "red"))

pdf("figures/marker-super-cluster-genes-heatmap3.pdf", height = 30, width=25)
plot(hm)
dev.off()
```


# Symbiont Analysis 
## Intra-Breviolum Differences
```{r}
#ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")
#ofav.combined.labeled$seurat_clusters <- Idents(ofav.combined.labeled)
#Idents(ofav.combined.labeled) <- "seurat_clusters.Timepoints"
Idents(ofav.combined.labeled)

ofav.combined.labeled.brev <- subset(x = ofav.combined.labeled, idents = c("Breviolum 2", "Breviolum 1"))
Idents(ofav.combined.labeled.brev)

levels(ofav.combined.labeled.brev)
my_levels <- c("Breviolum 1", "Breviolum 2")

brev.diff <- FindAllMarkers(ofav.combined.labeled.brev,only.pos=T, return.thresh=0.05, logfc.threshold=0.25, min.pct = 0.25)
write.csv(brev.diff, "markers/Brev_subpop_markers.csv")


ofav.combined.labeled.brev@active.ident <- factor(x = ofav.combined.labeled.brev@active.ident, levels = my_levels)

umap_brev_labeled <- DimPlot(ofav.combined.labeled.brev, reduction = "umap", split.by = "orig.ident", label = F, pt.size = 0.35, ncol = 2, cols=c("aquamarine1", "aquamarine3")) + labs(title = "UMAP Clustering of Breviolum Cells")
umap_brev_labeled
pdf("figures/UMAP-Brev-temps.pdf", width=10, height=10)
plot(umap_brev_labeled)
dev.off()
```

### Based off this, I want to see what is DE between Symbionts, but also between the time treatments...
```{r}
ofav.combined.labeled.brev$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.brev), ofav.combined.labeled.brev$orig.ident, sep = "_")

unique(ofav.combined.labeled.brev$seurat_clusters.Timepoints)

ofav.combined.labeled.brev$seurat_clusters <- Idents(ofav.combined.labeled.brev)
Idents(ofav.combined.labeled.brev) <- "seurat_clusters.Timepoints"
heatstress.response <- FindAllMarkers(ofav.combined.labeled.brev, only.pos=T, return.thresh=0.05, logfc.threshold=1)
write.csv(heatstress.response, "markers/DE_Brev_Timepoint_markers.csv")
levels(ofav.combined.labeled.brev)
ofav.combined.labeled.brev1 <- subset(x = ofav.combined.labeled.brev, idents = c("Breviolum 1_Ofav-T0-T", "Breviolum 1_Ofav-T3-T", "Breviolum 1_Ofav-T5-T", "Breviolum 1_Ofav-T7-T"))
heatstress.response <- FindAllMarkers(ofav.combined.labeled.brev1,only.pos=T, return.thresh=0.05, logfc.threshold=1)
write.csv(heatstress.response, "markers/DE_Brev1_Timepoint_markers.csv")

ofav.combined.labeled.brev2 <- subset(x = ofav.combined.labeled.brev, idents = c("Breviolum 2_Ofav-T0-T", "Breviolum 2_Ofav-T3-T"))
heatstress.response <- FindAllMarkers(ofav.combined.labeled.brev2,only.pos=T, return.thresh=0.05, logfc.threshold=1)
write.csv(heatstress.response, "markers/DE_Brev2_Timepoint_markers.csv")
```

```{r}
my_levels <- c("Breviolum 1_Ofav-T0-T","Breviolum 2_Ofav-T0-T","Breviolum 1_Ofav-T3-T","Breviolum 2_Ofav-T3-T", "Breviolum 1_Ofav-T5-T", "Breviolum 2_Ofav-T5-T", "Breviolum 1_Ofav-T7-T", "Breviolum 2_Ofav-T7-T")
ofav.combined.labeled.brev@active.ident <- factor(x = ofav.combined.labeled.brev@active.ident, levels = my_levels)
```

```{r}
DefaultAssay(ofav.combined.labeled.brev) <- "RNA"
p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-005962", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-005962", subtitle="SERTA domain-containing protein") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker1_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-003080", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-003080", subtitle="IPR043504 Peptidase S1, PA clan, chymotrypsin-like fold") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker2_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-034892", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-034892", subtitle="IPR035983 HECT, E3 ligase catalytic domain") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker3_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-022422", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-022422", subtitle="N-lysine methyltransferase setd6, variant 3") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker4_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-005894", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-005894", subtitle="BTB (POZ) domain-containing protein") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker5_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-018090", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-018090", subtitle="Radial spoke head 1") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker6_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-031802", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-031802", subtitle="Unknown") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker7_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="symbB.v1.2.035240", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: symbB.v1.2.035240", subtitle="calmodulin") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker8_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-007699", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-007699", subtitle="cofilin") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker9_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()


p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-035714", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-035714", subtitle="IPR005195 Glycoside hydrolase, family 65, central catalytic") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker10_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-008443", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-008443", subtitle="Lipid transport and metabolism") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker11_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="QW917-003256", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-003256", subtitle="Dynamin-related GTPase protein\nU:(U) Intracellular trafficking, secretion, and vesicular transport") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker12_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()


p1 <- VlnPlot(ofav.combined.labeled.brev, features ="symbB.v1.2.037890", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: symbB.v1.2.037890", subtitle="ferredoxin") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker13_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="symbB.v1.2.018894", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: symbB.v1.2.018894", subtitle="heat shock protein 90kDa beta") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker14_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.brev, features ="symbB.v1.2.032431", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: symbB.v1.2.032431", subtitle="glyceraldehyde 3-phosphate dehydrogenase (phosphorylating) ") + 
  theme(legend.position = "none") 
p1

pdf("figures/BrevTP_Marker15_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
levels(x = ofav.combined.labeled)
ofav.combined.labeled.symbs <- subset(x = ofav.combined.labeled, idents = c("Breviolum 2", "Breviolum 1", "Durusdinium"))
levels(ofav.combined.labeled.symbs)
my_levels <- c("Breviolum 1", "Breviolum 2", "Durusdinium")
ofav.combined.labeled.symbs@active.ident <- factor(x = ofav.combined.labeled.symbs@active.ident, levels = my_levels)
ofav.combined.labeled.symbs <- DimPlot(ofav.combined.labeled.symbs, reduction = "umap", split.by = "orig.ident", label = F, pt.size = 0.35, ncol = 2, cols=c("aquamarine1", "aquamarine3", "darkred")) + labs(title = "UMAP Clustering of Symbiodiniaceae Cells")
ofav.combined.labeled.symbs
pdf("figures/UMAP-Symbs-temps.pdf", width=10, height=7)
plot(ofav.combined.labeled.symbs)
dev.off()
```

```{r}
ofav.combined.labeled.dur <- subset(x = ofav.combined.labeled, idents = c("Durusdinium"))
ofav.combined.labeled.dur$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.dur), ofav.combined.labeled.dur$orig.ident, sep = "_")

unique(ofav.combined.labeled.dur$seurat_clusters.Timepoints)

ofav.combined.labeled.dur$seurat_clusters <- Idents(ofav.combined.labeled.dur)
Idents(ofav.combined.labeled.dur) <- "seurat_clusters.Timepoints"
DefaultAssay(ofav.combined.labeled.dur) <- "RNA"

p1 <- VlnPlot(ofav.combined.labeled.dur, features ="symbB.v1.2.040993", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("darkred", "darkred", "darkred", "darkred"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: symbB.v1.2.040993", subtitle="NA") + 
  theme(legend.position = "none") 
p1

pdf("figures/DurTP_Marker1_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.dur, features ="QW917-018353", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("darkred", "darkred", "darkred", "darkred"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-018353", subtitle="IPR003598 Immunoglobulin subtype 2") + 
  theme(legend.position = "none") 
p1

pdf("figures/DurTP_Marker2_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.dur, features ="QW917-016495", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("darkred", "darkred", "darkred", "darkred"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-016495", subtitle="ndufa8, NADH-ubiquinone oxidoreductase complex I 19kd subunit\nGO:0098803 - respiratory chain complex") + 
  theme(legend.position = "none") 
p1

pdf("figures/DurTP_Marker3_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.dur, features ="QW917-009835", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("darkred", "darkred", "darkred", "darkred"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Gene: QW917-009835", subtitle="IPR001245 Serine-threonine/tyrosine-protein kinase, catalytic domain") + 
  theme(legend.position = "none") 
p1

pdf("figures/DurTP_Marker4_Violin.pdf", width=7, height=5)
plot(p1)
dev.off()
```

### Ortholog Analysis
```{r}
ofav.combined.labeled.symbs <- subset(x = ofav.combined.labeled, idents = c("Breviolum 2", "Breviolum 1", "Durusdinium"))
levels(ofav.combined.labeled.symbs)
my_levels <- c("Breviolum 1", "Breviolum 2", "Durusdinium")

levels(ofav.combined.labeled.symbs)

ofav.combined.labeled.symbs$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.symbs), ofav.combined.labeled.symbs$orig.ident, sep = "_")

unique(ofav.combined.labeled.symbs$seurat_clusters.Timepoints)

ofav.combined.labeled.symbs$seurat_clusters <- Idents(ofav.combined.labeled.symbs)
Idents(ofav.combined.labeled.symbs) <- "seurat_clusters.Timepoints"
heatstress.response <- FindAllMarkers(ofav.combined.labeled.symbs, only.pos=T, return.thresh=0.05, logfc.threshold=1)
write.csv(heatstress.response, "markers/DE_AllSymbs_Timepoint_markers.csv")
levels(ofav.combined.labeled.symbs)
ofav.combined.labeled.dur <- subset(x = ofav.combined.labeled.symbs, idents = c("Durusdinium_Ofav-T0-T", "Durusdinium_Ofav-T3-T", "Durusdinium_Ofav-T5-T", "Durusdinium_Ofav-T7-T"))
heatstress.response <- FindAllMarkers(ofav.combined.labeled.dur,only.pos=T, return.thresh=0.05, logfc.threshold=1)
write.csv(heatstress.response, "markers/DE_Dur_Timepoint_markers.csv")
```

```{r}
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="QW917-003256", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3", "darkred", "darkred", "darkred", "darkred"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Dynamin-related GTPase protein\nU:(U) Intracellular trafficking, secretion, and vesicular transport") + 
  theme(legend.position = "none")
p1
pdf("figures/Symb_Marker1.pdf", width=7, height=5)
plot(p1)
dev.off()
```


```{r}
# Specify the genes you want to plot
gene1 <- "symbB.v1.2.006870"
gene2 <- "symbB.v1.2.028674"
gene3 <- "symbB.v1.2.037218"
gene5 <- "QW917-008443"
gene6 <- "g24501"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
expression_gene5 <- ofav.combined.labeled.tm[["RNA"]]@data[gene5, ]
expression_gene6 <- ofav.combined.labeled.tm[["RNA"]]@data[gene6, ]


# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3 + expression_gene4 + expression_gene5 + expression_gene6

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3", "darkred", "darkred", "darkred", "darkred"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Lipid Transport and Metabolism") + 
  theme(legend.position = "none")
p1
pdf("figures/Symb_ortholog_Marker1.pdf", width=7, height=5)
plot(p1)
dev.off()


# Specify the genes you want to plot
gene1 <- "g17767"
gene3 <- "g27975"
gene4 <- "symbB.v1.2.008661"
gene5 <- "symbB.v1.2.014291"
gene6 <- "symbB.v1.2.018354"
gene7 <- "QW917-027780"


ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
expression_gene4 <- ofav.combined.labeled.tm[["RNA"]]@data[gene4, ]
expression_gene5 <- ofav.combined.labeled.tm[["RNA"]]@data[gene5, ]
expression_gene6 <- ofav.combined.labeled.tm[["RNA"]]@data[gene6, ]
expression_gene7 <- ofav.combined.labeled.tm[["RNA"]]@data[gene7, ]


# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3 + expression_gene4 + expression_gene5 + expression_gene6 + expression_gene7

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, group.by= "seurat_clusters.Timepoints", cols=c("aquamarine1", "aquamarine1", "aquamarine1", "aquamarine1", "aquamarine3", "aquamarine3", "aquamarine3", "aquamarine3", "darkred", "darkred", "darkred", "darkred"), fill.by=ident) +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "IPR005821 Ion transport domain") + 
  theme(legend.position = "none")
p1
pdf("figures/Symb_ortholog_Marker2.pdf", width=7, height=5)
plot(p1)
dev.off()
```

### Breviolum (general): Time-point markers
```{r}
ofav.combined.labeled.symbs$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.symbs), ofav.combined.labeled.symbs$orig.ident, sep = "_")
ofav.combined.labeled.symbs$seurat_clusters <- Idents(ofav.combined.labeled.symbs)
Idents(ofav.combined.labeled.symbs) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.symbs)

my_levels <- c("Breviolum 1_Ofav-T0-T", "Breviolum 2_Ofav-T0-T", "Breviolum 3_Ofav-T0-T", "Durusdinium_Ofav-T0-T", "Breviolum 1_Ofav-T3-T", "Breviolum 2_Ofav-T3-T", "Breviolum 3_Ofav-T3-T", "Durusdinium_Ofav-T3-T", "Breviolum 1_Ofav-T5-T", "Durusdinium_Ofav-T5-T", "Breviolum 1_Ofav-T7-T", "Durusdinium_Ofav-T7-T")
ofav.combined.labeled.symbs@active.ident <- factor(x = ofav.combined.labeled.symbs@active.ident, levels = my_levels)
levels(ofav.combined.labeled.symbs)
```

```{r}
p1 <- VlnPlot(ofav.combined.labeled.symbs, features ="LOC110042424", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "LOC110042424: Unidentified", subtitle="Genes: LOC110042424") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev_timepoint_marker1.pdf", width=7, height=5)
plot(p1)
dev.off()

p1 <- VlnPlot(ofav.combined.labeled.symbs, features ="LOC110057958", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "ETS translocation variant 1-like", subtitle="Genes: LOC110057958") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev_timepoint_marker2.pdf", width=7, height=5)
plot(p1)
dev.off()

gene1 <- "LOC110064494"
gene2 <- "symbB.v1.2.014291"
gene3 <- "g17767"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Calreticulin-like", subtitle="Genes: LOC110064494 + symbB.v1.2.014291 + g17767") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev_timepoint_marker3.pdf", width=7, height=5)
plot(p1)
dev.off()


gene1 <- "LOC110055160"
gene2 <- "g31140"
gene3 <- "symbB.v1.2.007663"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "YTH domain-containing protein 1-like", subtitle="Genes: LOC110055160 + g31140 + symbB.v1.2.007663") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev_timepoint_marker4.pdf", width=7, height=5)
plot(p1)
dev.off()

gene1 <- "symbB.v1.2.015892"
gene2 <- "g33609"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "HSP90AB1", subtitle="Genes: symbB.v1.2.015892 + g33609") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev_timepoint_marker5.pdf", width=7, height=5)
plot(p1)
dev.off()
```

### Breviolum 1: Time-point markers
```{r}
gene1 <- "symbB.v1.2.031786"
gene2 <- "g2274"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "PetJ: functions as an electron carrier in oxygenic photosynthesis", subtitle="Genes: symbB.v1.2.031786 + g2274") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev1_timepoint_marker1.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.020046"
gene2 <- "symbB.v1.2.031120"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Unidentified", subtitle="Genes: symbB.v1.2.020046 + symbB.v1.2.031120") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev1_timepoint_marker2.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.040296"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "NRT2.2: cellular response to nitrate", subtitle="Gene: symbB.v1.2.040296") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev1_timepoint_marker3.pdf", width=7, height=5)
plot(p1)
dev.off()
```

### Breviolum 2: Time-point markers
```{r}
gene1 <- "LOC110060875"
gene2 <- "g4480"
gene3 <- "symbB.v1.2.027715"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Cathepsin B-like", subtitle="Genes: LOC110060875 + symbB.v1.2.027715 + g4480") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev2_timepoint_marker1.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "LOC110052299"
gene2 <- "symbB.v1.2.000231"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "soma ferritin-like", subtitle="Genes: LOC110052299 + symbB.v1.2.000231") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev2_timepoint_marker2.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.001035"
gene2 <- "g11006"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "KatG: Bifunctional enzyme with both catalase and broad-spectrum peroxidase activity", subtitle="Genes: symbB.v1.2.001035 + g11006") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev2_timepoint_marker3.pdf", width=7, height=5)
plot(p1)
dev.off()
```

### Breviolum 3: Time-point markers
```{r}
p1 <- VlnPlot(ofav.combined.labeled.symbs, features ="symbB.v1.2.037202", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "NRT2.2: cellular response to nitrate", subtitle="Genes: symbB.v1.2.037202") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev3_timepoint_marker1.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.043650"
gene2 <- "g27800"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Cytochrome b-245", subtitle="Genes: symbB.v1.2.043650 + g27800") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev3_timepoint_marker2.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.039610"
gene2 <- "g6851"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "BFD-like [2Fe-2S] binding domain", subtitle="Genes: symbB.v1.2.039610 + g6851") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev3_timepoint_marker3.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.042032"
gene2 <- "g19457"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "AmtB: ammonium transporter, marine subtype", subtitle="Genes: symbB.v1.2.042032 + g19457") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev3_timepoint_marker4.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.019932"
gene2 <- "g1711"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Lhcf1: fucoxanthin chlorophyll a c protein", subtitle="Genes: symbB.v1.2.019932 + g1711") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Brev3_timepoint_marker5.pdf", width=7, height=5)
plot(p1)
dev.off()
```
```{r}
gene1 <- "symbB.v1.2.043650"
gene2 <- "g27800"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Cytochrome b-245", subtitle="Genes: symbB.v1.2.043650 + g27800") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Symb_timepoints_marker6.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "symbB.v1.2.042032"
gene2 <- "g19457"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Ammonium Transporter Family", subtitle="Genes: symbB.v1.2.042032 + g19457") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Symb_timepoints_marker7.pdf", width=7, height=5)
plot(p1)
dev.off()
```
```{r}
gene1 <- "g24999"
gene2 <- "symbB.v1.2.017939"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "No ID", subtitle="Genes: g24999 + symbB.v1.2.017939") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Symb_timepoints_marker9.pdf", width=7, height=5)
plot(p1)
dev.off()
```
```{r}
gene1 <- "symbB.v1.2.032431"
gene2 <- "g23558"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Glyceraldehyde 3-phosphate dehydrogenase (phosphorylating); Glycolysis", subtitle="Genes: symbB.v1.2.032431 + g25457") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Symb_timepoints_marker10.pdf", width=7, height=5)
plot(p1)
dev.off()
```

```{r}
gene1 <- "g24036"
gene2 <- "LOC110041304"
gene3 <- "symbB.v1.2.031175"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2 + expression_gene3

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Ion Channel", subtitle="Genes: g24036 + LOC110041304 + symbB.v1.2.031175") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Symb_timepoints_marker13.pdf", width=7, height=5)
plot(p1)
dev.off()
```
### Durusdinium: Time-point Markers
```{r}
gene1 <- "LOC110049464"
gene2 <- "symbB.v1.2.017207"

ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "Tubulin polymerization-promoting protein family member 2-like", subtitle="Genes: LOC110049464 + symbB.v1.2.017207") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Dur_timepoint_marker1.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
gene1 <- "LOC110068597"
gene2 <- "symbB.v1.2.005751"
ofav.combined.labeled.tm <- ofav.combined.labeled.symbs
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
# Calculate the combined expression by adding the expression values of the two genes
combined_expression <- expression_gene1 + expression_gene2

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["combined_expression"]] <- combined_expression

# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.tm, features ="combined_expression", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "17-beta-hydroxysteroid dehydrogenase type 6-like", subtitle="Genes: LOC110068597 + symbB.v1.2.005751") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Dur_timepoint_marker2.pdf", width=7, height=5)
plot(p1)
dev.off() 
```
```{r}
# Create a single violin plot of the combined expression
p1 <- VlnPlot(ofav.combined.labeled.symbs, features ="symbB.v1.2.025478", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "HSP70", subtitle="Gene: symbB.v1.2.025478") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Dur_timepoint_marker3.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

```{r}
p1 <- VlnPlot(ofav.combined.labeled.symbs, features ="LOC110056367", pt.size=0, split.by = "orig.ident") +
  theme(plot.title = element_text(hjust = 0.5),plot.subtitle = element_text(hjust = 0.5)) +
  geom_boxplot(width=0.1, fill="white") + 
  labs(title = "stromelysin-1-like", subtitle="Gene: LOC110056367") + 
  theme(legend.position = "none") + scale_fill_viridis_d() 
p1

pdf("figures/orthologs/Dur_timepoint_marker4.pdf", width=7, height=5)
plot(p1)
dev.off() 
```

# ID DE genes through heat-stress (need to first subset out the symbionts to get better signal), then run to ID DE genes between symbionts during heat-stress
```{r}
DefaultAssay(ofav.combined.labeled) <- "RNA"
ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")
ofav.combined.labeled$seurat_clusters <- Idents(ofav.combined.labeled)
Idents(ofav.combined.labeled) <- "seurat_clusters.Timepoints"
heatstress.response <- FindAllMarkers(ofav.combined.labeled, only.pos=T, return.thresh=0.05)
head(heatstress.response, n = 15)
write.csv(heatstress.response, "markers/Timepoint_DE_markers.csv")
```


```{r}
markers.to.plot <- c("LOC110065941", "LOC110043167", "LOC110057483", "LOC110061104", "LOC110047412", "LOC110068916", "tRNA-Trp", "LOC110059405", "LOC110043735",
    "symbB.v1.2.014815", "LOC110057749", "LOC110068534", "LOC110039961", "LOC110064112", "LOC110063077", "symbB.v1.2.032431", "g22298", "LOC110058653")
dp<- DotPlot(ofav.combined.labeled, features = markers.to.plot, cols = c("darkblue", "yellow","orange", "red" ), dot.scale = 8, split.by = "orig.ident") +
    RotatedAxis()
pdf("figures/dotplot_TopGene_timepoints.pdf", width=10, height=15)
plot(dp)
dev.off()
```


# Figure Fixes for paper
## Heatmap of Marker Genes (Super-clusters) but better
```{r}
ofav.markers<- FindAllMarkers(ofav.combined.labeled, only.pos = TRUE, min.pct = 0.25, min.diff.pct = 0.20, logfc.threshold = 0.25, return.thresh=0.05)
ofav.markers <- ofav.markers[ofav.markers$p_val_adj<0.05,]
ofav.markers <- ofav.markers[ofav.markers$gene != "QW917-005894", ]
ofav.markers <- ofav.markers[ofav.markers$gene != "QW917-006349", ]
ofav.markers <- ofav.markers[ofav.markers$gene != "QW917-007699", ]
ofav.markers %>% group_by(cluster) %>% top_n(5, avg_log2FC) -> top10

levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")
ofav.combined.labeled.hm <- ofav.combined.labeled
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)

colors_hm <- c("azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)
ofav.combined.labeled.hm@meta.data[,"seurat_clusters",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters) %>% sample_n(60) -> index
hm <- DoHeatmap(ofav.combined.labeled.hm, slot="data", features = top10$gene,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + scale_fill_gradientn(colors = c("#fffff2", "red"), na.value = "#fffff2")

pdf("figures/marker-genes-heatmap_better.pdf", height = 20, width=25)
plot(hm)
dev.off()
```
###  Heatmap of Marker Genes (Super-clusters)
```{r}
ofav.combined.labeled.SC <- RenameIdents(ofav.combined.labeled, "Unidentified"="Unidentified", "Calicoblastic Cells"="Calicoblastic Cells", "Cnidocytes"="Cnidocytes", "Digestive Filaments"="Digestive Filaments", "Epidermis"="Epidermis", "Gastrodermis (Muscle-like)"="Gastrodermis (Muscle-like)", "Gastrodermis 1"="Gastrodermis", "Gastrodermis 2"="Gastrodermis", "Gland 1"="Gland", "Gland 2"="Gland", "Gland 3"="Gland", "Immune"="Immune", "Neurons 1"="Neurons", "Neurons 2"="Neurons", "Neurons 3"="Neurons", "Precursors"="Precursors", "Extracellular Breviolum"="Breviolum", "Breviolum-hosting Cells"="Breviolum", "Durusdinium-hosting Cells"=	"Durusdinium")
levels(ofav.combined.labeled.SC)

ofav.combined.labeled.SC$super_clusters <- paste(Idents(ofav.combined.labeled.SC))
unique(ofav.combined.labeled.SC$super_clusters)
Idents(ofav.combined.labeled.SC) <- "super_clusters"
levels(Idents(ofav.combined.labeled.SC))
ofav.combined.labeled.SC@active.ident <- factor(x = ofav.combined.labeled.SC@active.ident, levels = my_levels)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis", "Gland", "Immune", "Neurons", "Precursors", "Breviolum", "Durusdinium")

ofav.markers<- FindAllMarkers(ofav.combined.labeled.SC, only.pos = TRUE, min.pct = 0.25, min.diff.pct = 0.25, logfc.threshold = 0.25, return.thresh=0.05)
ofav.markers <- ofav.markers[ofav.markers$p_val_adj<0.05,]
ofav.markers %>% group_by(cluster) %>% top_n(7, avg_log2FC) -> top10

levels(ofav.combined.labeled.SC)

DefaultAssay(ofav.combined.labeled.SC) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.SC)

colors_hm <- c("azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkorange", "darkkhaki", "deepskyblue", "bisque3", "darkgoldenrod1", "darkgoldenrod4")

ofav.combined.labeled.SC@active.ident <- factor(x = ofav.combined.labeled.SC@active.ident, levels = my_levels)

ofav.combined.labeled.SC@meta.data[,"seurat_clusters",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters) %>% sample_n(20) -> index

hm <- DoHeatmap(ofav.combined.labeled.SC, slot="data", features = top10$gene,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + scale_fill_gradientn(colors = c("#fffff2", "red"))

pdf("figures/marker-super-cluster-genes-heatmap_better.pdf", height = 30, width=25)
plot(hm)
dev.off()
```
## Dot Plot of Symbiont Front Loading + Healthy Breviolum and Gatsrodermal Cells
### I'll first see the DE genes between Gastrodermal cells and see if they effect the Symbiosome cells too
Took all genes with pct.1 above 50%, order by COG
```{r}
library(scCustomize)
library(wesanderson)
ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Breviolum-hosting Cells", "Durusdinium-hosting Cells", "Gastrodermis (Muscle-like)", "Gastrodermis 1",  "Gastrodermis 2"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T3-T", "Gastrodermis 2_Ofav-T5-T", "Gastrodermis 2_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("QW917-025087", "QW917-028232", "QW917-016495", "QW917-025829", "QW917-008780", "QW917-002688", "QW917-035714", "QW917-018169", "QW917-004807", "QW917-006349", "QW917-024580", "QW917-005606", "QW917-003080", "QW917-024689", "QW917-002686", "QW917-018150", "QW917-023065", "QW917-030646", "QW917-002685", "QW917-028245", "QW917-017896", "QW917-034316", "QW917-007767", "QW917-033002", "QW917-010184", "QW917-014250", "QW917-023565", "QW917-026114", "QW917-005894", "QW917-005962", "QW917-027013", "QW917-004830", "QW917-014249", "QW917-026657", "QW917-016496", "QW917-028062", "QW917-032073", "QW917-009678", "QW917-009835", "QW917-018353", "QW917-027780", "QW917-022832", "QW917-007699", "QW917-016501")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Cell Markers") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_gastrodermal_symbionts__heatmarkers_pct50.pdf", width=15, height=10)
plot(dp)
dev.off()

```


## Remvoe gastros
```{r}
library(scCustomize)
library(wesanderson)
ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("QW917-025087", "QW917-028232", "QW917-016495", "QW917-025829", "QW917-008780", "QW917-002688", "QW917-035714", "QW917-018169", "QW917-004807", "QW917-006349", "QW917-024580", "QW917-005606", "QW917-003080", "QW917-024689", "QW917-002686", "QW917-018150", "QW917-023065", "QW917-030646", "QW917-002685", "QW917-028245", "QW917-017896", "QW917-034316", "QW917-007767", "QW917-033002", "QW917-010184", "QW917-014250", "QW917-023565", "QW917-026114", "QW917-005894", "QW917-005962", "QW917-027013", "QW917-004830", "QW917-014249", "QW917-026657", "QW917-016496", "QW917-028062", "QW917-032073", "QW917-009678", "QW917-009835", "QW917-018353", "QW917-027780", "QW917-022832", "QW917-007699", "QW917-016501")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Cell Markers") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_justsymbionts__heatmarkers_pct50.pdf", width=15, height=5)
plot(dp)
dev.off()
```

### Dot plot (do one of just brev and dur, then one of just gastrodermal cells for supplemental)
```{r}
library(scCustomize)
library(wesanderson)
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Breviolum-hosting Cells", "Durusdinium-hosting Cells", "Gastrodermis (Muscle-like)", "Gastrodermis 1",  "Gastrodermis 2"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T0-T", "Gastrodermis 2_Ofav-T3-T", "Gastrodermis 2_Ofav-T5-T", "Gastrodermis 2_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("QW917-028232", "QW917-035095", "QW917-024396", "QW917-029232", "QW917-025759", "QW917-016495", "QW917-002688", "QW917-018168", "QW917-008443", "QW917-020637", "QW917-021056", "QW917-026128", "QW917-005606", "QW917-005571", "QW917-030668", "QW917-024271", "QW917-002686", "QW917-018851", "QW917-029254", "QW917-027832", "QW917-010381", "QW917-018150", "QW917-002685", "QW917-030646", "QW917-009336", "QW917-028319", "QW917-023319", "QW917-022673", "QW917-014250", "QW917-033195", "QW917-031802", "QW917-023845", "QW917-003054", "QW917-011254", "QW917-013452", "QW917-032489", "QW917-029992", "QW917-013627", "QW917-023565", "QW917-026114", "QW917-030504", "QW917-034316", "QW917-024967", "QW917-005425", "QW917-035954", "QW917-017896", "QW917-005300", "QW917-004899", "QW917-008828", "QW917-033575", "QW917-027187", "QW917-030101", "QW917-015093", "QW917-005894", "QW917-005969", "QW917-018090", "QW917-002392", "QW917-005962", "QW917-013838", "QW917-014249", "QW917-016496", "QW917-002144", "QW917-020989", "QW917-018353", "QW917-001525", "QW917-014705", "QW917-000161", "QW917-009835", "QW917-027476", "QW917-022086", "QW917-031651", "QW917-000010", "QW917-003256", "QW917-013974", "QW917-004226", "QW917-002570", "QW917-007699", "QW917-016501")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Cell Markers") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_brevVdur_markers4.pdf", width=20, height=10)
plot(dp)
dev.off()

```

## New Heat-stress Heatmap
```{r}
genes_to_use <- c("QW917-005606",
"QW917-036332",
"QW917-036331",
"QW917-003971",
"QW917-010145",
"QW917-035634",
"QW917-004231",
"QW917-010146",
"QW917-036230",
"QW917-036231",
"QW917-030986",
"QW917-008447",
"QW917-001602",
"QW917-017006",
"QW917-024491",
"QW917-021056",
"QW917-026433",
"QW917-010131",
"QW917-016488",
"QW917-023065",
"QW917-003749",
"QW917-030646",
"QW917-031733",
"QW917-017942",
"QW917-008288",
"QW917-026114",
"symbB.v1.2.040993",
"QW917-016878",
"QW917-015093",
"QW917-025130",
"QW917-002944",
"QW917-016491",
"QW917-013386",
"QW917-017802",
"QW917-031915",
"QW917-033275",
"QW917-008368",
"QW917-016501",
"QW917-016496",
"QW917-016495",
"QW917-005894",
"QW917-005571",
"QW917-002685",
"QW917-024580",
"QW917-006349",
"QW917-018090",
"QW917-018150",
"QW917-004807",
"QW917-002688",
"QW917-018169",
"QW917-002686",
"QW917-011241",
"QW917-014250",
"QW917-034318",
"QW917-010184",
"QW917-031802",
"QW917-003080",
"QW917-005964",
"QW917-004830",
"QW917-009174",
"QW917-014249",
"QW917-019333",
"QW917-000806",
"QW917-030981",
"QW917-008044",
"QW917-028245",
"QW917-032073",
"QW917-016859",
"QW917-002611",
"QW917-033337",
"QW917-008780",
"QW917-027013",
"QW917-024689",
"QW917-005962",
"QW917-007767",
"QW917-023314",
"QW917-033002",
"QW917-026774")

levels(ofav.combined.labeled)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")

ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")

ofav.combined.labeled.hm1 <- ofav.combined.labeled

unique(ofav.combined.labeled.hm1$seurat_clusters.Timepoints)
ofav.combined.labeled.hm <- ofav.combined.labeled.hm1
rm(ofav.combined.labeled.hm1)
Idents(ofav.combined.labeled.hm) <- "seurat_clusters.Timepoints"
levels(ofav.combined.labeled.hm)
DefaultAssay(ofav.combined.labeled.hm) <- "RNA"
all.genes <- rownames(ofav.combined.labeled.hm)
ofav.combined.labeled.hm<- ScaleData(ofav.combined.labeled.hm, features = all.genes)

my_levels <- c("Unidentified_Ofav-T0-T", "Calicoblastic Cells_Ofav-T0-T", "Cnidocytes_Ofav-T0-T", "Digestive Filaments_Ofav-T0-T", "Epidermis_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 2_Ofav-T0-T", "Gland 1_Ofav-T0-T", "Gland 2_Ofav-T0-T", "Gland 3_Ofav-T0-T", "Immune_Ofav-T0-T", "Neurons 1_Ofav-T0-T", "Neurons 2_Ofav-T0-T", "Neurons 3_Ofav-T0-T", "Precursors_Ofav-T0-T", "Extracellular Breviolum_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Unidentified_Ofav-T3-T", "Calicoblastic Cells_Ofav-T3-T", "Cnidocytes_Ofav-T3-T", "Digestive Filaments_Ofav-T3-T", "Epidermis_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 2_Ofav-T3-T", "Gland 1_Ofav-T3-T", "Gland 2_Ofav-T3-T", "Gland 3_Ofav-T3-T", "Immune_Ofav-T3-T", "Neurons 1_Ofav-T3-T", "Neurons 2_Ofav-T3-T", "Neurons 3_Ofav-T3-T", "Precursors_Ofav-T3-T", "Extracellular Breviolum_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Unidentified_Ofav-T5-T", "Calicoblastic Cells_Ofav-T5-T", "Cnidocytes_Ofav-T5-T", "Digestive Filaments_Ofav-T5-T", "Epidermis_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 2_Ofav-T5-T", "Gland 1_Ofav-T5-T", "Gland 2_Ofav-T5-T", "Gland 3_Ofav-T5-T", "Immune_Ofav-T5-T", "Neurons 1_Ofav-T5-T", "Neurons 2_Ofav-T5-T", "Neurons 3_Ofav-T5-T", "Precursors_Ofav-T5-T", "Extracellular Breviolum_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Unidentified_Ofav-T7-T", "Calicoblastic Cells_Ofav-T7-T", "Cnidocytes_Ofav-T7-T", "Digestive Filaments_Ofav-T7-T", "Epidermis_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T7-T", "Gland 1_Ofav-T7-T", "Gland 2_Ofav-T7-T", "Gland 3_Ofav-T7-T", "Immune_Ofav-T7-T", "Neurons 1_Ofav-T7-T", "Neurons 2_Ofav-T7-T", "Neurons 3_Ofav-T7-T", "Precursors_Ofav-T7-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T7-T")

colors_hm <- c("azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4", "azure2", "darkorchid4", "plum1", "red1", "seagreen1", "darkolivegreen4", "darkolivegreen1", "darkolivegreen2", "darkorange1", "darkorange2", "darkorange", "darkkhaki", "deepskyblue3", "deepskyblue4", "deepskyblue2", "bisque3", "darkgoldenrod1", "darkgoldenrod3", "darkgoldenrod4")

ofav.combined.labeled.hm@active.ident <- factor(x = ofav.combined.labeled.hm@active.ident, levels = my_levels)

ofav.combined.labeled.hm@meta.data[,"seurat_clusters.Timepoints",drop=FALSE] %>% add_rownames() %>% group_by(seurat_clusters.Timepoints) %>% slice_sample(n= 25) -> index

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + NoLegend() + scale_fill_gradientn(colors = c("blue", "#fffff2", "red"))

pdf("figures/Bleaching_DEGS/Bleaching-combined-heatmap_larger.pdf", height = 25, width=30)
plot(hm)
dev.off()

hm <- DoHeatmap(ofav.combined.labeled.hm, features = genes_to_use,label=T,cells=index$rowname, draw.lines=T, lines.width=2, group.colors=colors_hm, angle=90) + scale_fill_gradientn(colors = c("blue", "#fffff2", "red"))

pdf("figures/Bleaching_DEGS/Bleaching-combined-heatmaplarger_wlegend.pdf", height = 25, width=50)
plot(hm)
dev.off()
```


## Dotplots of Nitrogen Cycle Related Genes determined by Ortholog Matches (for supplemtal, part of Radecker story)
```{r}
library(scCustomize)
library(wesanderson)
ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled@meta.data$orig.ident <- factor(x = ofav.combined.labeled@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled@meta.data$orig.ident)
my_levels <- c("Unidentified", "Calicoblastic Cells", "Cnidocytes", "Digestive Filaments", "Epidermis", "Gastrodermis (Muscle-like)", "Gastrodermis 1", "Gastrodermis 2", "Gland 1", "Gland 2", "Gland 3", "Immune", "Neurons 1", "Neurons 2", "Neurons 3", "Precursors", "Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells")
ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)

levels(ofav.combined.labeled)
DefaultAssay(ofav.combined.labeled) <- "RNA"
ofav.combined.labeled$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled), ofav.combined.labeled$orig.ident, sep = "_")
ofav.combined.labeled$seurat_clusters <- Idents(ofav.combined.labeled)
Idents(ofav.combined.labeled) <- "seurat_clusters.Timepoints"

my_levels <- c("Unidentified_Ofav-T0-T", "Calicoblastic Cells_Ofav-T0-T", "Cnidocytes_Ofav-T0-T", "Digestive Filaments_Ofav-T0-T", "Epidermis_Ofav-T0-T", "Gastrodermis (Muscle-like)_Ofav-T0-T", "Gastrodermis 1_Ofav-T0-T", "Gastrodermis 2_Ofav-T0-T", "Gland 1_Ofav-T0-T", "Gland 2_Ofav-T0-T", "Gland 3_Ofav-T0-T", "Immune_Ofav-T0-T", "Neurons 1_Ofav-T0-T", "Neurons 2_Ofav-T0-T", "Neurons 3_Ofav-T0-T", "Precursors_Ofav-T0-T", "Extracellular Breviolum_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Unidentified_Ofav-T3-T", "Calicoblastic Cells_Ofav-T3-T", "Cnidocytes_Ofav-T3-T", "Digestive Filaments_Ofav-T3-T", "Epidermis_Ofav-T3-T", "Gastrodermis (Muscle-like)_Ofav-T3-T", "Gastrodermis 1_Ofav-T3-T", "Gastrodermis 2_Ofav-T3-T", "Gland 1_Ofav-T3-T", "Gland 2_Ofav-T3-T", "Gland 3_Ofav-T3-T", "Immune_Ofav-T3-T", "Neurons 1_Ofav-T3-T", "Neurons 2_Ofav-T3-T", "Neurons 3_Ofav-T3-T", "Precursors_Ofav-T3-T", "Extracellular Breviolum_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Unidentified_Ofav-T5-T", "Calicoblastic Cells_Ofav-T5-T", "Cnidocytes_Ofav-T5-T", "Digestive Filaments_Ofav-T5-T", "Epidermis_Ofav-T5-T", "Gastrodermis (Muscle-like)_Ofav-T5-T", "Gastrodermis 1_Ofav-T5-T", "Gastrodermis 2_Ofav-T5-T", "Gland 1_Ofav-T5-T", "Gland 2_Ofav-T5-T", "Gland 3_Ofav-T5-T", "Immune_Ofav-T5-T", "Neurons 1_Ofav-T5-T", "Neurons 2_Ofav-T5-T", "Neurons 3_Ofav-T5-T", "Precursors_Ofav-T5-T", "Extracellular Breviolum_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Unidentified_Ofav-T7-T", "Calicoblastic Cells_Ofav-T7-T", "Cnidocytes_Ofav-T7-T", "Digestive Filaments_Ofav-T7-T", "Epidermis_Ofav-T7-T", "Gastrodermis (Muscle-like)_Ofav-T7-T", "Gastrodermis 1_Ofav-T7-T", "Gastrodermis 2_Ofav-T7-T", "Gland 1_Ofav-T7-T", "Gland 2_Ofav-T7-T", "Gland 3_Ofav-T7-T", "Immune_Ofav-T7-T", "Neurons 1_Ofav-T7-T", "Neurons 2_Ofav-T7-T", "Neurons 3_Ofav-T7-T", "Precursors_Ofav-T7-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T7-T")

ofav.combined.labeled@active.ident <- factor(x = ofav.combined.labeled@active.ident, levels = my_levels)

markers.to.plot <- c("QW917-027442", "QW917-027445", "QW917-004470", "QW917-034161", "QW917-030278", "QW917-001682")
dp<- DotPlot(ofav.combined.labeled, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Nitrogen Cycle Gene Orthologs: GS, GOGAT, GDH") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 0.5)
dp


pdf("figures/Bleaching_DEGS/dotplot_CoralHostNitrogen_markers_orthlogs.pdf", width=15, height=20)
plot(dp)
dev.off()
```

  
- Algae
    - nitrate transporters (NRT), nitrate reductases (NR), NH4 +transporters, and nitrite reductases (NIR) down-regulation, this indicated that algal symbionts invested fewer resources into the uptake of nitrogen from their environment.
    
symbB.v1.2.037202, symbB.v1.2.039609, symbB.v1.2.040296, symbB.v1.2.043343, symbB.v1.2.006166, symbB.v1.2.036678, g10243, g10915, g10916, g25261, g29518, g11468, g13537, g14556, g14852, g18426, g18427, g18428, g21014, g23259, g25447, g25634, g28784, g29439, g31083, g31879, g33461, g4771, symbB.v1.2.004989, symbB.v1.2.012582, symbB.v1.2.014853, symbB.v1.2.016011, symbB.v1.2.016012, symbB.v1.2.016830, symbB.v1.2.021581, symbB.v1.2.021693, symbB.v1.2.025234, symbB.v1.2.027768, symbB.v1.2.029965, symbB.v1.2.032386, symbB.v1.2.032387, symbB.v1.2.032388, symbB.v1.2.032389, symbB.v1.2.032390, symbB.v1.2.032391, symbB.v1.2.037139, symbB.v1.2.041251, symbB.v1.2.042032, symbB.v1.2.042128, symbB.v1.2.042454, symbB.v1.2.042712	

Combine genes.... and add in orthologs too
```{r}
library(scCustomize)
library(wesanderson)
ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("symbB.v1.2.037202", "symbB.v1.2.039609", "symbB.v1.2.040296", "symbB.v1.2.043343", "symbB.v1.2.006166", "symbB.v1.2.036678", "g10243", "g10915", "g10916", "g25261", "g29518", "g11468", "g13537", "g14556", "g14852", "g18426", "g18427", "g18428", "g21014", "g23259", "g25447", "g25634", "g28784", "g29439", "g31083", "g31879", "g33461", "g4771", "symbB.v1.2.004989", "symbB.v1.2.012582", "symbB.v1.2.014853", "symbB.v1.2.016011", "symbB.v1.2.016012", "symbB.v1.2.016830", "symbB.v1.2.021581", "symbB.v1.2.021693", "symbB.v1.2.025234", "symbB.v1.2.027768", "symbB.v1.2.029965", "symbB.v1.2.032386", "symbB.v1.2.032387", "symbB.v1.2.032388", "symbB.v1.2.032389", "symbB.v1.2.032390", "symbB.v1.2.032391", "symbB.v1.2.037139", "symbB.v1.2.041251", "symbB.v1.2.042032", "symbB.v1.2.042128", "symbB.v1.2.042454", "symbB.v1.2.042712")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Cell Markers") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle.pdf", width=20, height=10)
plot(dp)
dev.off()

```

### AMT-1
```{r}
library(scCustomize)
library(wesanderson)
#ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("g11468", "g14556", "g18427", "g25447", "g25634", "g28784", "g29439", "g31879", "symbB.v1.2.012582", "symbB.v1.2.014853", "symbB.v1.2.016012", "symbB.v1.2.016830", "symbB.v1.2.037139", "g4771", "g18426", "g23259", "g31083", "symbB.v1.2.021693", "symbB.v1.2.027768", "symbB.v1.2.042128", "symbB.v1.2.042454", "g10877", "symbB.v1.2.042983", "symbB.v1.2.025233", "g12717", "g27206", "symbB.v1.2.041306", "symbB.v1.2.017320", "symbB.v1.2.032386", "symbB.v1.2.032388", "g19457", "symbB.v1.2.005796", "g26542", "g26690", "g7649", "g5277", "symbB.v1.2.010336", "g33001", "g22042", "g26611", "g10269", "g21885", "g26628", "symbB.v1.2.003119", "g5491", "symbB.v1.2.026927", "g6575", "symbB.v1.2.026928", "g12554", "g21936", "g5435", "symbB.v1.2.004093", "g4719", "g3891", "g12594", "g21417", "g26663", "g29992", "g3897", "g5301", "g5417", "g5709", "g7631", "g9902", "symbB.v1.2.026643", "symbB.v1.2.031958", "symbB.v1.2.011960", "symbB.v1.2.032387", "symbB.v1.2.032389", "symbB.v1.2.032391", "symbB.v1.2.042712")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("AMT1") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle_AMT1.pdf", width=20, height=10)
plot(dp)
dev.off()

```

### AMTB
```{r}
library(scCustomize)
library(wesanderson)
#ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("g13537", "g14852", "g18428", "symbB.v1.2.016011", "symbB.v1.2.029965", "symbB.v1.2.032390", "symbB.v1.2.042032", "symbB.v1.2.025234")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("amtB") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle_amtB.pdf", width=20, height=10)
plot(dp)
dev.off()

```

### NIR
```{r}
library(scCustomize)
library(wesanderson)
#ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("g10243", "symbB.v1.2.035471", "symbB.v1.2.037201", "g29739", "symbB.v1.2.016992", "symbB.v1.2.036677", "symbB.v1.2.040142", "symbB.v1.2.040297", "symbB.v1.2.039611", "symbB.v1.2.037200")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("NIR") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle_NIR.pdf", width=20, height=10)
plot(dp)
dev.off()

```
### nirB
```{r}
library(scCustomize)
library(wesanderson)
#ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("symbB.v1.2.006166", "symbB.v1.2.036678", "g10915", "g10916", "g22668", "g6735", "symbB.v1.2.040143", "symbB.v1.2.039610", "g6851", "g4633", "g14244", "g7738", "g25261", "g8293", "g20011", "g5490", "symbB.v1.2.015591", "g29518")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("nirB") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle_nirB.pdf", width=20, height=10)
plot(dp)
dev.off()

```

### NRT2.2
```{r}
library(scCustomize)
library(wesanderson)
#ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)
markers.to.plot <- c("symbB.v1.2.037202", "symbB.v1.2.039609", "symbB.v1.2.040296")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("NRT2.2") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle_NRT2_2.pdf", width=20, height=10)
plot(dp)
dev.off()

```




### True Algal Nitrogen Plot
```{r}
library(scCustomize)
library(wesanderson)
ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)

gene1 <- "g11468"
gene2 <- "g25447"
gene3 <- "symbB.v1.2.014853"
gene4 <- "symbB.v1.2.042454"
gene5 <- "g10877"
gene6 <- "g12717"
gene7 <- "g27206"


ofav.combined.labeled.tm <- ofav.combined.labeled.subset
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
expression_gene4 <- ofav.combined.labeled.tm[["RNA"]]@data[gene4, ]
expression_gene5 <- ofav.combined.labeled.tm[["RNA"]]@data[gene5, ]
expression_gene6 <- ofav.combined.labeled.tm[["RNA"]]@data[gene6, ]
expression_gene7 <- ofav.combined.labeled.tm[["RNA"]]@data[gene7, ]

# Calculate the combined expression by adding the expression values of the two genes
AMT1 <- expression_gene1 + expression_gene2 + expression_gene3 + expression_gene4 + expression_gene5 + expression_gene6 + expression_gene7

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["AMT1"]] <- AMT1

gene1 <- "symbB.v1.2.042032"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
amtB <- expression_gene1
ofav.combined.labeled.tm[["amtB"]] <- amtB

gene1 <- "symbB.v1.2.016992"
gene2 <- "symbB.v1.2.036677"
gene3 <- "symbB.v1.2.037200"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
NIR <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["NIR"]] <- NIR

gene1 <- "symbB.v1.2.036678"
gene2 <- "symbB.v1.2.039610"
gene3 <- "g6851"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
nirB <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["nirB"]] <- nirB

gene1 <- "symbB.v1.2.037202"
gene2 <- "symbB.v1.2.039609"
gene3 <- "symbB.v1.2.040296"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
NRT2_2 <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["NRT2"]] <- NRT2_2

markers.to.plot <- c("AMT1", "amtB", "NIR", "nirB", "NRT2") 

dp<- DotPlot(ofav.combined.labeled.tm, features = markers.to.plot, cols = c("#FFFFFF", "#daa520"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Nitrogen Uptake Genes") + scale_color_gradient2(low="#FFFFFF", mid= "#f4d195", high = "#daa520", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle.pdf", width=10, height=7)
plot(dp)
dev.off()

## BLUE HEATMAP
dp<- DotPlot(ofav.combined.labeled.tm, features = markers.to.plot, cols = c("#FFFFFF", "#3e5fcc"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Nitrogen Uptake Genes") + scale_color_gradient2(low="#FFFFFF", mid= "#bfcbee", high = "#3e5fcc", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle_blue.pdf", width=10, height=7)
plot(dp)
dev.off()

```


#### Convert Symbiont Violin to Dotplot
Add in Nitrate Genes here as well. Organize According to Function
```{r}
library(scCustomize)
library(wesanderson)
ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)

gene1 <- "g11468"
gene2 <- "g25447"
gene3 <- "symbB.v1.2.014853"
gene4 <- "symbB.v1.2.042454"
gene5 <- "g10877"
gene6 <- "g12717"
gene7 <- "g27206"


ofav.combined.labeled.tm <- ofav.combined.labeled.subset
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
expression_gene4 <- ofav.combined.labeled.tm[["RNA"]]@data[gene4, ]
expression_gene5 <- ofav.combined.labeled.tm[["RNA"]]@data[gene5, ]
expression_gene6 <- ofav.combined.labeled.tm[["RNA"]]@data[gene6, ]
expression_gene7 <- ofav.combined.labeled.tm[["RNA"]]@data[gene7, ]

# Calculate the combined expression by adding the expression values of the two genes
AMT1 <- expression_gene1 + expression_gene2 + expression_gene3 + expression_gene4 + expression_gene5 + expression_gene6 + expression_gene7

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["AMT1"]] <- AMT1

gene1 <- "symbB.v1.2.042032"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
amtB <- expression_gene1
ofav.combined.labeled.tm[["amtB"]] <- amtB

gene1 <- "symbB.v1.2.016992"
gene2 <- "symbB.v1.2.036677"
gene3 <- "symbB.v1.2.037200"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
NIR <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["NIR"]] <- NIR

gene1 <- "symbB.v1.2.036678"
gene2 <- "symbB.v1.2.039610"
gene3 <- "g6851"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
nirB <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["nirB"]] <- nirB

gene1 <- "symbB.v1.2.037202"
gene2 <- "symbB.v1.2.039609"
gene3 <- "symbB.v1.2.040296"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
NRT2_2 <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["NRT2"]] <- NRT2_2

gene1 <- "symbB.v1.2.017107"
gene2 <- "g30271"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
PGK <- expression_gene1 + expression_gene2 
ofav.combined.labeled.tm[["PGK"]] <- PGK

gene1 <- "symbB.v1.2.041360"
gene2 <- "g30271"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
rbcL <- expression_gene1 + expression_gene2 
ofav.combined.labeled.tm[["rbcL"]] <- rbcL

gene1 <- "QW917-030101"
gene2 <- "symbB.v1.2.008653"
gene3 <- "g16404"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
FBXW7 <- expression_gene1 + expression_gene2 + expression_gene3
ofav.combined.labeled.tm[["FBXW7"]] <- FBXW7

gene1 <- "QW917-009678"
gene2 <- "symbB.v1.2.010170"
gene3 <- "g33744"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
STAM2 <- expression_gene1 + expression_gene2 + expression_gene3
ofav.combined.labeled.tm[["STAM2"]] <- STAM2

gene1 <- "QW917-010381"
gene2 <- "symbB.v1.2.018628"
gene3 <- "g7946"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
RVT_2 <- expression_gene1 + expression_gene2 + expression_gene3
ofav.combined.labeled.tm[["RVT_2"]] <- RVT_2

gene1 <- "QW917-008828"
gene2 <- "g24293"
gene3 <- "symbB.v1.2.033044"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
RAD52 <- expression_gene1 + expression_gene2 + expression_gene3
ofav.combined.labeled.tm[["RAD52"]] <- RAD52

markers.to.plot <- c("AMT1", "amtB", "NIR", "nirB", "NRT2", "rbcL", "PGK", "symbB.v1.2.000589", "symbB.v1.2.032431", "FBXW7", "RAD52", "QW917-002688", "STAM2", "symbB.v1.2.040993", "RVT_2") 

dp<- DotPlot(ofav.combined.labeled.tm, features = markers.to.plot, cols = c("#FFFFFF", "#daa520"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Significant Symbiont Orthologs") + scale_color_gradient2(low="#FFFFFF", mid= "#f4d195", high = "#daa520", midpoint = 6)
dp

#pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_expanded.pdf", width=10, height=7)
#plot(dp)
#dev.off()

dp<- DotPlot(ofav.combined.labeled.tm, features = markers.to.plot, cols = c("#FFFFFF", "#3e5fcc"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Significant Symbiont Orthologs") + scale_color_gradient2(low="#FFFFFF", mid= "#bfcbee", high = "#3e5fcc", midpoint = 1)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_expanded_blue.pdf", width=10, height=7)
plot(dp)
dev.off()

```

### GOGAT For symbionts (GS–GOGAT)
```{r}
library(scCustomize)
library(wesanderson)
#ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)

markers.to.plot <- c("symbB.v1.2.011890","symbB.v1.2.013029","symbB.v1.2.039505","g27978","g28635","symbB.v1.2.025752","symbB.v1.2.025753","g6545","g20632","symbB.v1.2.013028","g1225","g10600","g32839","symbB.v1.2.011892","symbB.v1.2.020111","symbB.v1.2.002328","g6546","symbB.v1.2.002740","g20634","g20633","g19719","g10276","g563","g3273","symbB.v1.2.007875","symbB.v1.2.013031","g28634","symbB.v1.2.017142","symbB.v1.2.003123","symbB.v1.2.032260","g4496","g25031","g25805","g6753","symbB.v1.2.015312","symbB.v1.2.017545","g23160","symbB.v1.2.003922","symbB.v1.2.038374","symbB.v1.2.028312","symbB.v1.2.021909","symbB.v1.2.040496","symbB.v1.2.025732","symbB.v1.2.017166","symbB.v1.2.025243","symbB.v1.2.018766","symbB.v1.2.026464","g27321","g3978","g14945","g19066","symbB.v1.2.016681","symbB.v1.2.000363","symbB.v1.2.040826","symbB.v1.2.035694","symbB.v1.2.017146","symbB.v1.2.030031","symbB.v1.2.027618","symbB.v1.2.005472","g29039","g6630","g22563","g4578","g33295","g15601")

dp<- DotPlot(ofav.combined.labeled.subset, features = markers.to.plot, cols = c("#FFFFFF", "#FF0000"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("GOGAT") + scale_color_gradient2(low="#FFFFFF", mid= "#FFCCCC", high = "#FF0000", midpoint = 2)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_NitrogenCycle_GOGAT.pdf", width=15, height=7)
plot(dp)
dev.off()

```

glutamate dehydrogenase (GDH) - symbB.v1.2.025480, symbB.v1.2.025479, symbB.v1.2.002431, g17254, g23085, symbB.v1.2.028589, g15039, symbB.v1.2.034171, symbB.v1.2.009726

Glutamine Synthetase (GS) - symbB.v1.2.024833, symbB.v1.2.036231, g17356, g23621,

Glutamate Synthase (GOGAT) - g6546, symbB.v1.2.002740, symbB.v1.2.013028, g32839, symbB.v1.2.017142, symbB.v1.2.011890

Now add these to the figure above... with new blue heat scale

```{r}
library(scCustomize)
library(wesanderson)
#ofav.combined.labeled <- readRDS("/Volumes/AMB_SSD1/molabits_06042024/Ofav_Exp_Spring2022/ofav.combined.labeled.rds")
ofav.combined.labeled.subset <- subset(x = ofav.combined.labeled, idents = c("Extracellular Breviolum", "Breviolum-hosting Cells", "Durusdinium-hosting Cells"))
my_levels <- c("Ofav-T0-T", "Ofav-T3-T", "Ofav-T5-T", "Ofav-T7-T")
ofav.combined.labeled.subset@meta.data$orig.ident <- factor(x = ofav.combined.labeled.subset@meta.data$orig.ident, levels = my_levels)
levels(ofav.combined.labeled.subset@meta.data$orig.ident)

DefaultAssay(ofav.combined.labeled.subset) <- "RNA"
ofav.combined.labeled.subset$seurat_clusters.Timepoints <- paste(Idents(ofav.combined.labeled.subset), ofav.combined.labeled.subset$orig.ident, sep = "_")
ofav.combined.labeled.subset$seurat_clusters <- Idents(ofav.combined.labeled.subset)
Idents(ofav.combined.labeled.subset) <- "seurat_clusters.Timepoints"
my_levels <- c("Extracellular Breviolum_Ofav-T0-T", "Extracellular Breviolum_Ofav-T3-T", "Extracellular Breviolum_Ofav-T5-T", "Extracellular Breviolum_Ofav-T7-T", "Breviolum-hosting Cells_Ofav-T0-T", "Breviolum-hosting Cells_Ofav-T3-T", "Breviolum-hosting Cells_Ofav-T5-T", "Breviolum-hosting Cells_Ofav-T7-T", "Durusdinium-hosting Cells_Ofav-T0-T", "Durusdinium-hosting Cells_Ofav-T3-T", "Durusdinium-hosting Cells_Ofav-T5-T", "Durusdinium-hosting Cells_Ofav-T7-T")
ofav.combined.labeled.subset@active.ident <- factor(x = ofav.combined.labeled.subset@active.ident, levels = my_levels)

gene1 <- "g11468"
gene2 <- "g25447"
gene3 <- "symbB.v1.2.014853"
gene4 <- "symbB.v1.2.042454"
gene5 <- "g10877"
gene6 <- "g12717"
gene7 <- "g27206"


ofav.combined.labeled.tm <- ofav.combined.labeled.subset
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
expression_gene4 <- ofav.combined.labeled.tm[["RNA"]]@data[gene4, ]
expression_gene5 <- ofav.combined.labeled.tm[["RNA"]]@data[gene5, ]
expression_gene6 <- ofav.combined.labeled.tm[["RNA"]]@data[gene6, ]
expression_gene7 <- ofav.combined.labeled.tm[["RNA"]]@data[gene7, ]

# Calculate the combined expression by adding the expression values of the two genes
AMT1 <- expression_gene1 + expression_gene2 + expression_gene3 + expression_gene4 + expression_gene5 + expression_gene6 + expression_gene7

# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["AMT1"]] <- AMT1

gene1 <- "symbB.v1.2.042032"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
amtB <- expression_gene1
ofav.combined.labeled.tm[["amtB"]] <- amtB

gene1 <- "symbB.v1.2.016992"
gene2 <- "symbB.v1.2.036677"
gene3 <- "symbB.v1.2.037200"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
NIR <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["NIR"]] <- NIR

gene1 <- "symbB.v1.2.036678"
gene2 <- "symbB.v1.2.039610"
gene3 <- "g6851"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
nirB <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["nirB"]] <- nirB

gene1 <- "symbB.v1.2.037202"
gene2 <- "symbB.v1.2.039609"
gene3 <- "symbB.v1.2.040296"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
# Calculate the combined expression by adding the expression values of the two genes
NRT2_2 <- expression_gene1 + expression_gene2 + expression_gene3
# Create a new metadata column for the combined expression
ofav.combined.labeled.tm[["NRT2"]] <- NRT2_2

gene1 <- "symbB.v1.2.017107"
gene2 <- "g30271"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
PGK <- expression_gene1 + expression_gene2 
ofav.combined.labeled.tm[["PGK"]] <- PGK

gene1 <- "symbB.v1.2.041360"
gene2 <- "g30271"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
rbcL <- expression_gene1 + expression_gene2 
ofav.combined.labeled.tm[["rbcL"]] <- rbcL

gene1 <- "QW917-030101"
gene2 <- "symbB.v1.2.008653"
gene3 <- "g16404"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
FBXW7 <- expression_gene1 + expression_gene2 + expression_gene3
ofav.combined.labeled.tm[["FBXW7"]] <- FBXW7

gene1 <- "QW917-009678"
gene2 <- "symbB.v1.2.010170"
gene3 <- "g33744"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
STAM2 <- expression_gene1 + expression_gene2 + expression_gene3
ofav.combined.labeled.tm[["STAM2"]] <- STAM2

gene1 <- "QW917-010381"
gene2 <- "symbB.v1.2.018628"
gene3 <- "g7946"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
RVT_2 <- expression_gene1 + expression_gene2 + expression_gene3
ofav.combined.labeled.tm[["RVT_2"]] <- RVT_2

gene1 <- "QW917-008828"
gene2 <- "g24293"
gene3 <- "symbB.v1.2.033044"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
RAD52 <- expression_gene1 + expression_gene2 + expression_gene3
ofav.combined.labeled.tm[["RAD52"]] <- RAD52

gene1 <- "symbB.v1.2.025480"
gene2 <- "symbB.v1.2.025479"
gene3 <- "symbB.v1.2.002431"
gene4 <- "g17254"
gene5 <- "g23085"
gene6 <- "symbB.v1.2.028589"
gene7 <- "g15039"
gene8 <- "symbB.v1.2.034171"
gene9 <- "symbB.v1.2.009726"

expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
expression_gene4 <- ofav.combined.labeled.tm[["RNA"]]@data[gene4, ]
expression_gene5 <- ofav.combined.labeled.tm[["RNA"]]@data[gene5, ]
expression_gene6 <- ofav.combined.labeled.tm[["RNA"]]@data[gene6, ]
expression_gene7 <- ofav.combined.labeled.tm[["RNA"]]@data[gene7, ]
expression_gene8 <- ofav.combined.labeled.tm[["RNA"]]@data[gene8, ]
expression_gene9 <- ofav.combined.labeled.tm[["RNA"]]@data[gene9, ]
GDH <- expression_gene1 + expression_gene2 + expression_gene3 + expression_gene4 + expression_gene5 + expression_gene6 + expression_gene7 + expression_gene8 + expression_gene9
ofav.combined.labeled.tm[["GDH"]] <- GDH


gene1 <- "symbB.v1.2.024833"
gene2 <- "symbB.v1.2.036231"
gene3 <- "g17356"
gene4 <- "g23621"
expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
expression_gene4 <- ofav.combined.labeled.tm[["RNA"]]@data[gene4, ]
GS <- expression_gene1 + expression_gene2 + expression_gene3 + expression_gene4
ofav.combined.labeled.tm[["GS"]] <- GS

gene1 <- "g6546"
gene2 <- "symbB.v1.2.002740"
gene3 <- "symbB.v1.2.013028"
gene4 <- "g32839"
gene5 <- "symbB.v1.2.017142"
gene6 <- "symbB.v1.2.011890"

expression_gene1 <- ofav.combined.labeled.tm[["RNA"]]@data[gene1, ]
expression_gene2 <- ofav.combined.labeled.tm[["RNA"]]@data[gene2, ]
expression_gene3 <- ofav.combined.labeled.tm[["RNA"]]@data[gene3, ]
expression_gene4 <- ofav.combined.labeled.tm[["RNA"]]@data[gene4, ]
expression_gene5 <- ofav.combined.labeled.tm[["RNA"]]@data[gene5, ]
expression_gene6 <- ofav.combined.labeled.tm[["RNA"]]@data[gene6, ]
GOGAT <- expression_gene1 + expression_gene2 + expression_gene3 + expression_gene4 + expression_gene5 + expression_gene6

ofav.combined.labeled.tm[["GOGAT"]] <- GOGAT


markers.to.plot <- c("AMT1", "amtB", "NIR", "nirB", "NRT2", "GS", "GOGAT", "GDH", "rbcL", "PGK", "symbB.v1.2.000589", "symbB.v1.2.032431", "FBXW7", "RAD52", "QW917-002688", "STAM2", "symbB.v1.2.040993", "RVT_2") 


dp<- DotPlot(ofav.combined.labeled.tm, features = markers.to.plot, cols = c("#FFFFFF", "#3e5fcc"), dot.scale = 8, scale=FALSE) +
    RotatedAxis() + ggtitle("Significant Symbiont Orthologs") + scale_color_gradient2(low="#FFFFFF", mid= "#bfcbee", high = "#3e5fcc", midpoint = 1)
dp

pdf("figures/Bleaching_Symbiont_DEGS/dotplot_symbionts_expanded_blue.pdf", width=10, height=7)
plot(dp)
dev.off()

```